// @progress/kendo-angular-inputs/dist/es/switch/switch.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-inputs/dist/es/switch/switch.component';
import * as import2 from '@progress/kendo-angular-inputs/dist/es/common/draggable';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import4 from '@angular/forms';
const styles_SwitchComponent:any[] = ([] as any[]);
export const RenderType_SwitchComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SwitchComponent,
  data: {}
}
);
export function View_SwitchComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{dragHandle: 0}),
    import0.ɵqud(201326592,2,{wrapper: 0}),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'k-switch-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-switch-background'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'wrapper',
        1
      ]

    ]
    ,(null as any),11,'span',[
      [
        'class',
        'k-switch-container'
      ]
      ,
      [
        'kendoDraggable',
        ''
      ]

    ]
      ,[[
        1,
        'tabindex',
        0
      ]
    ],[
      [
        (null as any),
        'kendoDrag'
      ]
      ,
      [
        (null as any),
        'kendoPress'
      ]
      ,
      [
        (null as any),
        'kendoRelease'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.SwitchComponent = v.component;
      if (('kendoDrag' === en)) {
        const pd_0:any = ((<any>co.onHandleDrag($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('kendoPress' === en)) {
        const pd_1:any = ((<any>co.onHandlePress($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('kendoRelease' === en)) {
        const pd_2:any = ((<any>co.onHandleRelease($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('blur' === en)) {
        const pd_3:any = ((<any>co.onBlur()) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import2.KendoDraggableDirective,[import0.ElementRef],(null as any),{
      kendoDrag: 'kendoDrag',
      kendoPress: 'kendoPress',
      kendoRelease: 'kendoRelease'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'dragHandle',
        1
      ]

    ]
      ,(null as any),7,'span',[[
        'class',
        'k-switch-handle'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'k-switch-label-on'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'k-switch-label-off'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.SwitchComponent = v.component;
    const currVal_0:any = (co.disabled? (undefined as any): 1);
    ck(v,8,0,currVal_0);
    const currVal_1:any = co.onLabel;
    ck(v,14,0,currVal_1);
    const currVal_2:any = co.offLabel;
    ck(v,17,0,currVal_2);
  });
}
function View_SwitchComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-switch',([] as any[]),[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-checked',
        0
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-switch',
        (null as any)
      ]
      ,
      [
        2,
        'k-switch-on',
        (null as any)
      ]
      ,
      [
        2,
        'k-switch-off',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-disabled',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'keydown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).keyDown($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_SwitchComponent_0,RenderType_SwitchComponent)),
      import0.ɵdid(2121728,(null as any),0,import1.SwitchComponent,[[
        2,
        import3.RTL
      ]
    ],(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.SwitchComponent])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).direction;
    const currVal_1:any = import0.ɵnov(v,1).ariaDisabled;
    const currVal_2:any = import0.ɵnov(v,1).ariaChecked;
    const currVal_3:any = import0.ɵnov(v,1).widget;
    const currVal_4:any = import0.ɵnov(v,1).switchClass;
    const currVal_5:any = import0.ɵnov(v,1).switchOn;
    const currVal_6:any = import0.ɵnov(v,1).switchOff;
    const currVal_7:any = import0.ɵnov(v,1).stateDisabled;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  });
}
export const SwitchComponentNgFactory:import0.ComponentFactory<import1.SwitchComponent> = import0.ɵccf('kendo-switch',import1.SwitchComponent,View_SwitchComponent_Host_0,{
  onLabel: 'onLabel',
  offLabel: 'offLabel',
  checked: 'checked',
  disabled: 'disabled'
}
,{valueChange: 'valueChange'},([] as any[]));
