// @progress/kendo-angular-grid/dist/es/cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@angular/forms';
import * as import3 from '@progress/kendo-angular-grid/dist/es/cell.component';
import * as import4 from '@progress/kendo-angular-grid/dist/es/edit.service';
import * as import5 from '@progress/kendo-angular-grid/dist/es/cell-context';
const styles_CellComponent:any[] = ([] as any[]);
export const RenderType_CellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_CellComponent,
  data: {}
}
);
function View_CellComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(null as any),(null as any));
}
function View_CellComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_CellComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_CellComponent_4)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      '$implicit',
      'isNew',
      'column',
      'dataItem',
      'formGroup',
      'rowIndex'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.editTemplateRef;
    const currVal_1:any = ck(v,2,0,co.formGroup,co.isNew,co.column,co.dataItem,co.formGroup,co.rowIndex);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_CellComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'input',[
      [
        'class',
        'k-textbox'
      ]
      ,
      [
        'type',
        'number'
      ]

    ]
    ,[
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]
      ,
      [
        (null as any),
        'change'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,1)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      if (('change' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,2).onChange($event.target.value)) !== false);
        ad = (pd_4 && ad);
      }
      if (('input' === en)) {
        const pd_5:any = ((<any>import0.ɵnov(v,2).onChange($event.target.value)) !== false);
        ad = (pd_5 && ad);
      }
      if (('blur' === en)) {
        const pd_6:any = ((<any>import0.ɵnov(v,2).onTouched()) !== false);
        ad = (pd_6 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import2.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import2.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵdid(8192,(null as any),0,import2.ɵbc,[
      import0.Renderer,
      import0.ElementRef
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import2.NG_VALUE_ACCESSOR,(p0_0:any,p1_0:any) => {
      return [
        p0_0,
        p1_0
      ]
      ;
    },[
      import2.DefaultValueAccessor,
      import2.ɵbc
    ]
    ),
    import0.ɵdid(270336,(null as any),0,import2.FormControlDirective,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import2.NG_VALUE_ACCESSOR
      ]

    ]
      ,{form: [
        0,
        'form'
      ]
    },(null as any)),
    import0.ɵprd(1024,(null as any),import2.NgControl,(null as any),[import2.FormControlDirective]),
    import0.ɵdid(8192,(null as any),0,import2.NgControlStatus,[import2.NgControl],(null as any),(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_7:any = co.formGroup.get(co.column.field);
    ck(v,4,0,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,6).ngClassUntouched;
    const currVal_1:any = import0.ɵnov(v,6).ngClassTouched;
    const currVal_2:any = import0.ɵnov(v,6).ngClassPristine;
    const currVal_3:any = import0.ɵnov(v,6).ngClassDirty;
    const currVal_4:any = import0.ɵnov(v,6).ngClassValid;
    const currVal_5:any = import0.ɵnov(v,6).ngClassInvalid;
    const currVal_6:any = import0.ɵnov(v,6).ngClassPending;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
function View_CellComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'input',[
      [
        'class',
        'k-textbox'
      ]
      ,
      [
        'type',
        'date'
      ]

    ]
    ,[
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,1)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import2.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import2.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import2.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import2.DefaultValueAccessor]),
    import0.ɵdid(270336,(null as any),0,import2.FormControlDirective,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import2.NG_VALUE_ACCESSOR
      ]

    ]
      ,{form: [
        0,
        'form'
      ]
    },(null as any)),
    import0.ɵprd(1024,(null as any),import2.NgControl,(null as any),[import2.FormControlDirective]),
    import0.ɵdid(8192,(null as any),0,import2.NgControlStatus,[import2.NgControl],(null as any),(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_7:any = co.formGroup.get(co.column.field);
    ck(v,3,0,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,5).ngClassUntouched;
    const currVal_1:any = import0.ɵnov(v,5).ngClassTouched;
    const currVal_2:any = import0.ɵnov(v,5).ngClassPristine;
    const currVal_3:any = import0.ɵnov(v,5).ngClassDirty;
    const currVal_4:any = import0.ɵnov(v,5).ngClassValid;
    const currVal_5:any = import0.ɵnov(v,5).ngClassInvalid;
    const currVal_6:any = import0.ɵnov(v,5).ngClassPending;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
function View_CellComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'input',[[
        'type',
        'checkbox'
      ]
    ],[
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'change'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('change' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onChange($event.target.checked)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import2.CheckboxControlValueAccessor,[
      import0.Renderer,
      import0.ElementRef
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import2.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import2.CheckboxControlValueAccessor]),
    import0.ɵdid(270336,(null as any),0,import2.FormControlDirective,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import2.NG_VALUE_ACCESSOR
      ]

    ]
      ,{form: [
        0,
        'form'
      ]
    },(null as any)),
    import0.ɵprd(1024,(null as any),import2.NgControl,(null as any),[import2.FormControlDirective]),
    import0.ɵdid(8192,(null as any),0,import2.NgControlStatus,[import2.NgControl],(null as any),(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_7:any = co.formGroup.get(co.column.field);
    ck(v,3,0,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,5).ngClassUntouched;
    const currVal_1:any = import0.ɵnov(v,5).ngClassTouched;
    const currVal_2:any = import0.ɵnov(v,5).ngClassPristine;
    const currVal_3:any = import0.ɵnov(v,5).ngClassDirty;
    const currVal_4:any = import0.ɵnov(v,5).ngClassValid;
    const currVal_5:any = import0.ɵnov(v,5).ngClassInvalid;
    const currVal_6:any = import0.ɵnov(v,5).ngClassPending;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
function View_CellComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'input',[
      [
        'class',
        'k-textbox'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,1)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import2.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import2.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import2.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import2.DefaultValueAccessor]),
    import0.ɵdid(270336,(null as any),0,import2.FormControlDirective,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import2.NG_VALUE_ACCESSOR
      ]

    ]
      ,{form: [
        0,
        'form'
      ]
    },(null as any)),
    import0.ɵprd(1024,(null as any),import2.NgControl,(null as any),[import2.FormControlDirective]),
    import0.ɵdid(8192,(null as any),0,import2.NgControlStatus,[import2.NgControl],(null as any),(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_7:any = co.formGroup.get(co.column.field);
    ck(v,3,0,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,5).ngClassUntouched;
    const currVal_1:any = import0.ɵnov(v,5).ngClassTouched;
    const currVal_2:any = import0.ɵnov(v,5).ngClassPristine;
    const currVal_3:any = import0.ɵnov(v,5).ngClassDirty;
    const currVal_4:any = import0.ɵnov(v,5).ngClassValid;
    const currVal_5:any = import0.ɵnov(v,5).ngClassInvalid;
    const currVal_6:any = import0.ɵnov(v,5).ngClassPending;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
function View_CellComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),14,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,(null as any),0,import1.NgSwitch,([] as any[]),{ngSwitch: [
        0,
        'ngSwitch'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_6)),
    import0.ɵdid(139264,(null as any),0,import1.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_7)),
    import0.ɵdid(139264,(null as any),0,import1.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_8)),
    import0.ɵdid(139264,(null as any),0,import1.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_9)),
    import0.ɵdid(8192,(null as any),0,import1.NgSwitchDefault,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.editor;
    ck(v,1,0,currVal_0);
    const currVal_1:any = 'numeric';
    ck(v,4,0,currVal_1);
    const currVal_2:any = 'date';
    ck(v,7,0,currVal_2);
    const currVal_3:any = 'boolean';
    ck(v,10,0,currVal_3);
  },(null as any));
}
function View_CellComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.editTemplateRef;
    ck(v,3,0,currVal_0);
    const currVal_1:boolean = !co.column.editTemplate;
    ck(v,6,0,currVal_1);
  },(null as any));
}
export function View_CellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,(null as any),0,import1.NgSwitch,([] as any[]),{ngSwitch: [
        0,
        'ngSwitch'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_1)),
    import0.ɵdid(139264,(null as any),0,import1.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_CellComponent_2)),
    import0.ɵdid(139264,(null as any),0,import1.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import1.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import3.CellComponent = v.component;
    const currVal_0:any = co.isEdited;
    ck(v,2,0,currVal_0);
    const currVal_1:any = false;
    ck(v,5,0,currVal_1);
    const currVal_2:any = true;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_CellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridCell',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_CellComponent_0,RenderType_CellComponent)),
    import0.ɵdid(24576,(null as any),0,import3.CellComponent,[
      import4.EditService,
      [
        2,
        import5.CELL_CONTEXT
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const CellComponentNgFactory:import0.ComponentFactory<import3.CellComponent> = import0.ɵccf('[kendoGridCell]',import3.CellComponent,View_CellComponent_Host_0,{
  column: 'column',
  isNew: 'isNew',
  rowIndex: 'rowIndex',
  dataItem: 'dataItem'
}
,{},['*']);
