// @progress/kendo-angular-grid/dist/es/col-group.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-grid/dist/es/col-group.component';
const styles_ColGroupComponent:any[] = ([] as any[]);
export const RenderType_ColGroupComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColGroupComponent,
  data: {}
}
);
function View_ColGroupComponent_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'col',([] as any[]),[[
        2,
        'k-group-col',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_ColGroupComponent_3(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'col',([] as any[]),[[
        2,
        'k-hierarchy-col',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_ColGroupComponent_4(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'col',([] as any[]),[[
        4,
        'width',
        'px'
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit.width;
    ck(v,0,0,currVal_0);
  });
}
function View_ColGroupComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ColGroupComponent_2)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ColGroupComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ColGroupComponent_4)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groups;
    ck(v,2,0,currVal_0);
    const currVal_1:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,5,0,currVal_1);
    const currVal_2:any = co.columns;
    ck(v,8,0,currVal_2);
  },(null as any));
}
export function View_ColGroupComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ColGroupComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = true;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_ColGroupComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_ColGroupComponent_0,RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import2.ColGroupComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ColGroupComponentNgFactory:import0.ComponentFactory<import2.ColGroupComponent> = import0.ɵccf('[kendoGridColGroup]',import2.ColGroupComponent,View_ColGroupComponent_Host_0,{
  columns: 'columns',
  groups: 'groups',
  detailTemplate: 'detailTemplate'
}
,{},([] as any[]));
