// @progress/kendo-angular-grid/dist/es/column-group.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/column-group.component';
import * as import2 from '@progress/kendo-angular-grid/dist/es/column-base';
const styles_ColumnGroupComponent:any[] = ([] as any[]);
export const RenderType_ColumnGroupComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnGroupComponent,
  data: {}
}
);
export function View_ColumnGroupComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnGroupComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'kendo-grid-column-group',([] as any[]),(null as any),(null as any),(null as any),View_ColumnGroupComponent_0,RenderType_ColumnGroupComponent)),
      import0.ɵdid(24576,(null as any),3,import1.ColumnGroupComponent,[[
        3,
        import2.ColumnBase
      ]
    ],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{children: 1}),
    import0.ɵqud(301989888,2,{headerTemplates: 1}),
    import0.ɵqud(167772160,3,{footerTemplate: 0}),
      import0.ɵprd(1024,[[
        1,
        4
      ]
    ],import2.ColumnBase,(null as any),[import1.ColumnGroupComponent])
  ]
  ,(null as any),(null as any));
}
export const ColumnGroupComponentNgFactory:import0.ComponentFactory<import1.ColumnGroupComponent> = import0.ɵccf('kendo-grid-column-group',import1.ColumnGroupComponent,View_ColumnGroupComponent_Host_0,{
  title: 'title',
  width: 'width',
  locked: 'locked',
  hidden: 'hidden',
  media: 'media',
  style: 'style',
  headerStyle: 'headerStyle',
  footerStyle: 'footerStyle',
  cssClass: 'class',
  headerClass: 'headerClass',
  footerClass: 'footerClass'
}
,{},([] as any[]));
