// @progress/kendo-angular-grid/dist/es/column.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/column.component';
import * as import2 from '@progress/kendo-angular-grid/dist/es/column-base';
const styles_ColumnComponent:any[] = ([] as any[]);
export const RenderType_ColumnComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ColumnComponent,
  data: {}
}
);
export function View_ColumnComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_ColumnComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'kendo-grid-column',([] as any[]),(null as any),(null as any),(null as any),View_ColumnComponent_0,RenderType_ColumnComponent)),
      import0.ɵdid(24576,(null as any),8,import1.ColumnComponent,[[
        3,
        import2.ColumnBase
      ]
    ],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{children: 1}),
    import0.ɵqud(301989888,2,{headerTemplates: 1}),
    import0.ɵqud(167772160,3,{footerTemplate: 0}),
    import0.ɵqud(167772160,4,{template: 0}),
    import0.ɵqud(167772160,5,{groupHeaderTemplate: 0}),
    import0.ɵqud(167772160,6,{groupFooterTemplate: 0}),
    import0.ɵqud(167772160,7,{editTemplate: 0}),
    import0.ɵqud(167772160,8,{filterCellTemplate: 0}),
      import0.ɵprd(1024,[[
        1,
        4
      ]
    ],import2.ColumnBase,(null as any),[import1.ColumnComponent])
  ]
  ,(null as any),(null as any));
}
export const ColumnComponentNgFactory:import0.ComponentFactory<import1.ColumnComponent> = import0.ɵccf('kendo-grid-column',import1.ColumnComponent,View_ColumnComponent_Host_0,{
  title: 'title',
  width: 'width',
  locked: 'locked',
  hidden: 'hidden',
  media: 'media',
  style: 'style',
  headerStyle: 'headerStyle',
  footerStyle: 'footerStyle',
  cssClass: 'class',
  headerClass: 'headerClass',
  footerClass: 'footerClass',
  field: 'field',
  format: 'format',
  sortable: 'sortable',
  editor: 'editor',
  filter: 'filter',
  filterable: 'filterable',
  editable: 'editable'
}
,{},([] as any[]));
