// @progress/kendo-angular-grid/dist/es/command-column.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/command-column.component';
import * as import2 from '@progress/kendo-angular-grid/dist/es/column-base';
const styles_CommandColumnComponent:any[] = ([] as any[]);
export const RenderType_CommandColumnComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_CommandColumnComponent,
  data: {}
}
);
export function View_CommandColumnComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_CommandColumnComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'kendo-grid-command-column',([] as any[]),(null as any),(null as any),(null as any),View_CommandColumnComponent_0,RenderType_CommandColumnComponent)),
      import0.ɵdid(24576,(null as any),4,import1.CommandColumnComponent,[[
        3,
        import2.ColumnBase
      ]
    ],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{children: 1}),
    import0.ɵqud(301989888,2,{headerTemplates: 1}),
    import0.ɵqud(167772160,3,{footerTemplate: 0}),
    import0.ɵqud(167772160,4,{template: 0}),
      import0.ɵprd(1024,[[
        1,
        4
      ]
    ],import2.ColumnBase,(null as any),[import1.CommandColumnComponent])
  ]
  ,(null as any),(null as any));
}
export const CommandColumnComponentNgFactory:import0.ComponentFactory<import1.CommandColumnComponent> = import0.ɵccf('kendo-grid-command-column',import1.CommandColumnComponent,View_CommandColumnComponent_Host_0,{
  title: 'title',
  width: 'width',
  locked: 'locked',
  hidden: 'hidden',
  media: 'media',
  style: 'style',
  headerStyle: 'headerStyle',
  footerStyle: 'footerStyle',
  cssClass: 'class',
  headerClass: 'headerClass',
  footerClass: 'footerClass'
}
,{},([] as any[]));
