// @progress/kendo-angular-grid/dist/es/footer.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-grid/dist/es/template-context.directive';
import * as import3 from '@progress/kendo-angular-grid/dist/es/footer.component';
const styles_FooterComponent:any[] = ([] as any[]);
export const RenderType_FooterComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FooterComponent,
  data: {}
}
);
function View_FooterComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_FooterComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),[[
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_FooterComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_FooterComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'td',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_FooterComponent_5)),
      import0.ɵdid(73728,(null as any),0,import2.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'columnIndex',
      'column',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit.footerClass;
    ck(v,1,0,currVal_0);
    const currVal_1:any = v.context.$implicit.footerStyle;
    ck(v,2,0,currVal_1);
    const currVal_2:any = ck(v,6,0,v.context.$implicit.footerTemplateRef,(co.lockedColumnsCount + v.context.index),v.context.$implicit,v.context.$implicit);
    ck(v,5,0,currVal_2);
  },(null as any));
}
function View_FooterComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',([] as any[]),[[
        2,
        'k-footer-template',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FooterComponent_2)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FooterComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FooterComponent_4)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.groups;
    ck(v,4,0,currVal_1);
    const currVal_2:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,7,0,currVal_2);
    const currVal_3:any = co.columns;
    ck(v,10,0,currVal_3);
  },(ck,v) => {
    const currVal_0:any = true;
    ck(v,1,0,currVal_0);
  });
}
export function View_FooterComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FooterComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = true;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_FooterComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridFooter',
        ''
      ]
      ],[[
        2,
        'k-grid-footer',
        (null as any)
      ]
    ],(null as any),(null as any),View_FooterComponent_0,RenderType_FooterComponent)),
    import0.ɵdid(24576,(null as any),0,import3.FooterComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).footerClass;
    ck(v,0,0,currVal_0);
  });
}
export const FooterComponentNgFactory:import0.ComponentFactory<import3.FooterComponent> = import0.ɵccf('[kendoGridFooter]',import3.FooterComponent,View_FooterComponent_Host_0,{
  columns: 'columns',
  groups: 'groups',
  detailTemplate: 'detailTemplate',
  scrollable: 'scrollable',
  lockedColumnsCount: 'lockedColumnsCount'
}
,{},([] as any[]));
