// @progress/kendo-angular-grid/dist/es/grid.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './grouping/group-panel.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/grouping/group-connection.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/grouping/group-panel.component';
import * as import4 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import5 from './col-group.component.ngfactory';
import * as import6 from '@progress/kendo-angular-grid/dist/es/col-group.component';
import * as import7 from './header.component.ngfactory';
import * as import8 from '@progress/kendo-angular-grid/dist/es/header.component';
import * as import9 from './footer.component.ngfactory';
import * as import10 from '@progress/kendo-angular-grid/dist/es/footer.component';
import * as import11 from '@angular/common';
import * as import12 from '@progress/kendo-angular-grid/dist/es/resizable.directive';
import * as import13 from '@progress/kendo-angular-grid/dist/es/grid.component';
import * as import14 from './list.component.ngfactory';
import * as import15 from '@progress/kendo-angular-grid/dist/es/list.component';
import * as import16 from '@progress/kendo-angular-grid/dist/es/details.service';
import * as import17 from '@progress/kendo-angular-grid/dist/es/change-notification.service';
import * as import18 from '@progress/kendo-angular-grid/dist/es/suspend.service';
import * as import19 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import20 from './table-body.component.ngfactory';
import * as import21 from '@progress/kendo-angular-grid/dist/es/table-body.component';
import * as import22 from '@progress/kendo-angular-grid/dist/es/grouping/groups.service';
import * as import23 from '@progress/kendo-angular-grid/dist/es/edit.service';
import * as import24 from './pager/pager.component.ngfactory';
import * as import25 from '@progress/kendo-angular-grid/dist/es/pager/pager.component';
import * as import26 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
import * as import27 from '@progress/kendo-angular-grid/dist/es/localization/localized-messages.directive';
import * as import28 from '@progress/kendo-angular-grid/dist/es/localization/messages';
import * as import29 from '@progress/kendo-angular-grid/dist/es/browser-support.service';
import * as import30 from '@progress/kendo-angular-grid/dist/es/selection.service';
import * as import31 from '@progress/kendo-angular-grid/dist/es/grouping/group-info.service';
import * as import32 from '@progress/kendo-angular-grid/dist/es/filtering/filter.service';
import * as import33 from '@progress/kendo-angular-grid/dist/es/pdf/pdf.service';
import * as import34 from '@progress/kendo-angular-grid/dist/es/responsive.service';
import * as import35 from '@progress/kendo-angular-grid/dist/es/cell-context';
import * as import36 from '@progress/kendo-angular-l10n/dist/es/message.service';
const styles_GridComponent:any[] = ([] as any[]);
export const RenderType_GridComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_GridComponent,
  data: {}
}
);
function View_GridComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'kendo-grid-group-panel',([] as any[]),[[
        2,
        'k-grouping-header',
        (null as any)
      ]
      ],[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.groupChange.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import1.View_GroupPanelComponent_0,import1.RenderType_GroupPanelComponent)),
    import0.ɵprd(256,(null as any),import2.GroupDragService,import2.GroupDragService,[
      import2.GroupConnectionService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(2121728,(null as any),0,import3.GroupPanelComponent,[
      import2.GroupConnectionService,
      import2.GroupDragService,
      import0.ElementRef,
      import4.LocalizationService
    ]
    ,{
      text: [
        0,
        'text'
      ]
      ,
      groups: [
        1,
        'groups'
      ]

    }
    ,{change: 'change'}),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.groupable.emptyText;
    const currVal_2:any = co.group;
    ck(v,2,0,currVal_1,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).groupHeaderClass;
    ck(v,0,0,currVal_0);
  });
}
function View_GridComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'lockedHeader',
        1
      ]

    ]
      ,(null as any),13,'div',[[
        'class',
        'k-grid-header-locked'
      ]
      ],[[
        4,
        'width',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'table',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import5.View_ColGroupComponent_0,import5.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import6.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'thead',[[
        'kendoGridHeader',
        ''
      ]
      ],[[
        2,
        'k-grid-header',
        (null as any)
      ]
      ],[[
        (null as any),
        'sortChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('sortChange' === en)) {
        const pd_0:any = ((<any>co.sortChange.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import7.View_HeaderComponent_0,import7.RenderType_HeaderComponent)),
    import0.ɵprd(256,(null as any),import2.GroupDragService,import2.GroupDragService,[
      import2.GroupConnectionService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(2121728,(null as any),0,import8.HeaderComponent,[import2.GroupDragService],{
      totalColumnLevels: [
        0,
        'totalColumnLevels'
      ]
      ,
      columns: [
        1,
        'columns'
      ]
      ,
      groups: [
        2,
        'groups'
      ]
      ,
      detailTemplate: [
        3,
        'detailTemplate'
      ]
      ,
      scrollable: [
        4,
        'scrollable'
      ]
      ,
      filterable: [
        5,
        'filterable'
      ]
      ,
      sort: [
        6,
        'sort'
      ]
      ,
      filter: [
        7,
        'filter'
      ]
      ,
      sortable: [
        8,
        'sortable'
      ]
      ,
      groupable: [
        9,
        'groupable'
      ]

    }
    ,{sortChange: 'sortChange'}),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.lockedLeafColumns;
    const currVal_2:any = co.group;
    const currVal_3:any = co.detailTemplate;
    ck(v,5,0,currVal_1,currVal_2,currVal_3);
    const currVal_5:any = co.totalColumnLevels;
    const currVal_6:any = co.lockedColumns;
    const currVal_7:any = co.group;
    const currVal_8:any = co.detailTemplate;
    const currVal_9:any = true;
    const currVal_10:any = co.filterable;
    const currVal_11:any = co.sort;
    const currVal_12:any = co.filter;
    const currVal_13:any = co.sortable;
    const currVal_14:any = co.showGroupPanel;
    ck(v,10,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.lockedWidth;
    ck(v,0,0,currVal_0);
    const currVal_4:any = import0.ɵnov(v,10).headerClass;
    ck(v,8,0,currVal_4);
  });
}
function View_GridComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',[[
        'class',
        'k-grid-footer-locked'
      ]
      ],[[
        4,
        'width',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'table',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import5.View_ColGroupComponent_0,import5.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import6.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tfoot',[[
        'kendoGridFooter',
        ''
      ]
      ],[[
        2,
        'k-grid-footer',
        (null as any)
      ]
    ],(null as any),(null as any),import9.View_FooterComponent_0,import9.RenderType_FooterComponent)),
    import0.ɵdid(24576,(null as any),0,import10.FooterComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]
      ,
      scrollable: [
        3,
        'scrollable'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.lockedLeafColumns;
    const currVal_2:any = co.group;
    const currVal_3:any = co.detailTemplate;
    ck(v,5,0,currVal_1,currVal_2,currVal_3);
    const currVal_5:any = co.lockedLeafColumns;
    const currVal_6:any = co.group;
    const currVal_7:any = co.detailTemplate;
    const currVal_8:any = true;
    ck(v,9,0,currVal_5,currVal_6,currVal_7,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.lockedWidth;
    ck(v,0,0,currVal_0);
    const currVal_4:any = import0.ɵnov(v,9).footerClass;
    ck(v,8,0,currVal_4);
  });
}
function View_GridComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),18,'div',[[
        'class',
        'k-grid-footer'
      ]
      ],[[
        4,
        'paddingRight',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_5)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'footer',
        1
      ]

    ]
      ,(null as any),13,'div',[[
        'class',
        'k-grid-footer-wrap'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import12.ResizableContainerDirective,[
      import0.ElementRef,
      import0.Renderer2,
      [
        2,
        import13.GridComponent
      ]

    ]
    ,{
      lockedWidth: [
        0,
        'lockedWidth'
      ]
      ,
      kendoGridResizableContainer: [
        1,
        'kendoGridResizableContainer'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'table',([] as any[]),[[
        4,
        'width',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import5.View_ColGroupComponent_0,import5.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import6.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tfoot',[[
        'kendoGridFooter',
        ''
      ]
      ],[[
        2,
        'k-grid-footer',
        (null as any)
      ]
    ],(null as any),(null as any),import9.View_FooterComponent_0,import9.RenderType_FooterComponent)),
    import0.ɵdid(24576,(null as any),0,import10.FooterComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]
      ,
      scrollable: [
        3,
        'scrollable'
      ]
      ,
      lockedColumnsCount: [
        4,
        'lockedColumnsCount'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.lockedLeafColumns.length;
    ck(v,3,0,currVal_1);
    const currVal_2:any = ((co.lockedWidth + co.scrollbarWidth) + 3);
    const currVal_3:any = co.lockedLeafColumns.length;
    ck(v,5,0,currVal_2,currVal_3);
    const currVal_5:any = co.nonLockedLeafColumns;
    const currVal_6:any = (co.isLocked? import0.ɵEMPTY_ARRAY: co.group);
    const currVal_7:any = co.detailTemplate;
    ck(v,10,0,currVal_5,currVal_6,currVal_7);
    const currVal_9:any = co.nonLockedLeafColumns;
    const currVal_10:any = (co.isLocked? import0.ɵEMPTY_ARRAY: co.group);
    const currVal_11:any = co.detailTemplate;
    const currVal_12:any = true;
    const currVal_13:any = co.lockedLeafColumns.length;
    ck(v,14,0,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.scrollbarWidth;
    ck(v,0,0,currVal_0);
    const currVal_4:any = co.nonLockedWidth;
    ck(v,7,0,currVal_4);
    const currVal_8:any = import0.ɵnov(v,14).footerClass;
    ck(v,13,0,currVal_8);
  });
}
function View_GridComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),19,'div',[[
        'class',
        'k-grid-header'
      ]
      ],[[
        4,
        'padding',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_3)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'header',
        1
      ]

    ]
      ,(null as any),14,'div',[[
        'class',
        'k-grid-header-wrap'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import12.ResizableContainerDirective,[
      import0.ElementRef,
      import0.Renderer2,
      [
        2,
        import13.GridComponent
      ]

    ]
    ,{
      lockedWidth: [
        0,
        'lockedWidth'
      ]
      ,
      kendoGridResizableContainer: [
        1,
        'kendoGridResizableContainer'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'table',([] as any[]),[[
        4,
        'width',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import5.View_ColGroupComponent_0,import5.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import6.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'thead',[[
        'kendoGridHeader',
        ''
      ]
      ],[[
        2,
        'k-grid-header',
        (null as any)
      ]
      ],[[
        (null as any),
        'sortChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('sortChange' === en)) {
        const pd_0:any = ((<any>co.sortChange.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import7.View_HeaderComponent_0,import7.RenderType_HeaderComponent)),
    import0.ɵprd(256,(null as any),import2.GroupDragService,import2.GroupDragService,[
      import2.GroupConnectionService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(2121728,(null as any),0,import8.HeaderComponent,[import2.GroupDragService],{
      totalColumnLevels: [
        0,
        'totalColumnLevels'
      ]
      ,
      columns: [
        1,
        'columns'
      ]
      ,
      groups: [
        2,
        'groups'
      ]
      ,
      detailTemplate: [
        3,
        'detailTemplate'
      ]
      ,
      scrollable: [
        4,
        'scrollable'
      ]
      ,
      filterable: [
        5,
        'filterable'
      ]
      ,
      sort: [
        6,
        'sort'
      ]
      ,
      filter: [
        7,
        'filter'
      ]
      ,
      sortable: [
        8,
        'sortable'
      ]
      ,
      groupable: [
        9,
        'groupable'
      ]
      ,
      lockedColumnsCount: [
        10,
        'lockedColumnsCount'
      ]

    }
    ,{sortChange: 'sortChange'}),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'kendo-grid-list',([] as any[]),(null as any),[
      [
        (null as any),
        'containerScroll'
      ]
      ,
      [
        (null as any),
        'pageChange'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('containerScroll' === en)) {
        const pd_0:any = ((<any>co.containerScroll($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('pageChange' === en)) {
        const pd_1:any = ((<any>co.notifyPageChange('list',$event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import14.View_ListComponent_0,import14.RenderType_ListComponent)),
    import0.ɵprd(128,(null as any),import15.SCROLLER_FACTORY_TOKEN,import15.DEFAULT_SCROLLER_FACTORY,([] as any[])),
      import0.ɵdid(2482176,[[
        4,
        4
      ]
    ],0,import15.ListComponent,[
      import15.SCROLLER_FACTORY_TOKEN,
      import16.DetailsService,
      import17.ChangeNotificationService,
      import18.SuspendService,
      [
        2,
        import19.RTL
      ]

    ]
    ,{
      data: [
        0,
        'data'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      total: [
        2,
        'total'
      ]
      ,
      height: [
        3,
        'height'
      ]
      ,
      rowHeight: [
        4,
        'rowHeight'
      ]
      ,
      detailRowHeight: [
        5,
        'detailRowHeight'
      ]
      ,
      take: [
        6,
        'take'
      ]
      ,
      skip: [
        7,
        'skip'
      ]
      ,
      columns: [
        8,
        'columns'
      ]
      ,
      detailTemplate: [
        9,
        'detailTemplate'
      ]
      ,
      noRecordsTemplate: [
        10,
        'noRecordsTemplate'
      ]
      ,
      selectable: [
        11,
        'selectable'
      ]
      ,
      groupable: [
        12,
        'groupable'
      ]

    }
    ,{
      pageChange: 'pageChange',
      containerScroll: 'containerScroll'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_4)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.isLocked;
    ck(v,4,0,currVal_1);
    const currVal_2:any = ((co.lockedWidth + co.scrollbarWidth) + 3);
    const currVal_3:any = co.lockedLeafColumns.length;
    ck(v,6,0,currVal_2,currVal_3);
    const currVal_5:any = co.nonLockedLeafColumns;
    const currVal_6:any = (co.isLocked? import0.ɵEMPTY_ARRAY: co.group);
    const currVal_7:any = co.detailTemplate;
    ck(v,11,0,currVal_5,currVal_6,currVal_7);
    const currVal_9:any = co.totalColumnLevels;
    const currVal_10:any = co.nonLockedColumns;
    const currVal_11:any = (co.isLocked? import0.ɵEMPTY_ARRAY: co.group);
    const currVal_12:any = co.detailTemplate;
    const currVal_13:any = true;
    const currVal_14:any = co.filterable;
    const currVal_15:any = co.sort;
    const currVal_16:any = co.filter;
    const currVal_17:any = co.sortable;
    const currVal_18:any = co.showGroupPanel;
    const currVal_19:any = co.lockedLeafColumns.length;
    ck(v,16,1,[
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17,
      currVal_18,
      currVal_19
    ]
    );
    const currVal_20:any = co.view;
    const currVal_21:any = co.group;
    const currVal_22:any = (co.isVirtual? co.view.total: co.pageSize);
    const currVal_23:any = co.height;
    const currVal_24:any = co.rowHeight;
    const currVal_25:any = co.detailRowHeight;
    const currVal_26:any = co.pageSize;
    const currVal_27:any = co.skip;
    const currVal_28:any = co.columnsContainer;
    const currVal_29:any = co.detailTemplate;
    const currVal_30:any = co.noRecordsTemplate;
    const currVal_31:any = co.selectable;
    const currVal_32:any = co.groupable;
    ck(v,24,1,[
      currVal_20,
      currVal_21,
      currVal_22,
      currVal_23,
      currVal_24,
      currVal_25,
      currVal_26,
      currVal_27,
      currVal_28,
      currVal_29,
      currVal_30,
      currVal_31,
      currVal_32
    ]
    );
    const currVal_33:any = co.showFooter;
    ck(v,28,0,currVal_33);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.headerPadding;
    ck(v,1,0,currVal_0);
    const currVal_4:any = co.nonLockedWidth;
    ck(v,8,0,currVal_4);
    const currVal_8:any = import0.ɵnov(v,16).headerClass;
    ck(v,14,0,currVal_8);
  });
}
function View_GridComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tfoot',[[
        'kendoGridFooter',
        ''
      ]
      ],[[
        2,
        'k-grid-footer',
        (null as any)
      ]
    ],(null as any),(null as any),import9.View_FooterComponent_0,import9.RenderType_FooterComponent)),
    import0.ɵdid(24576,(null as any),0,import10.FooterComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]
      ,
      scrollable: [
        3,
        'scrollable'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.leafColumns;
    const currVal_2:any = co.group;
    const currVal_3:any = co.detailTemplate;
    const currVal_4:any = false;
    ck(v,1,0,currVal_1,currVal_2,currVal_3,currVal_4);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).footerClass;
    ck(v,0,0,currVal_0);
  });
}
function View_GridComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'table',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import5.View_ColGroupComponent_0,import5.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import6.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'thead',[[
        'kendoGridHeader',
        ''
      ]
      ],[[
        2,
        'k-grid-header',
        (null as any)
      ]
      ],[[
        (null as any),
        'sortChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('sortChange' === en)) {
        const pd_0:any = ((<any>co.sortChange.emit($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import7.View_HeaderComponent_0,import7.RenderType_HeaderComponent)),
    import0.ɵprd(256,(null as any),import2.GroupDragService,import2.GroupDragService,[
      import2.GroupConnectionService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(2121728,(null as any),0,import8.HeaderComponent,[import2.GroupDragService],{
      totalColumnLevels: [
        0,
        'totalColumnLevels'
      ]
      ,
      columns: [
        1,
        'columns'
      ]
      ,
      groups: [
        2,
        'groups'
      ]
      ,
      detailTemplate: [
        3,
        'detailTemplate'
      ]
      ,
      scrollable: [
        4,
        'scrollable'
      ]
      ,
      filterable: [
        5,
        'filterable'
      ]
      ,
      sort: [
        6,
        'sort'
      ]
      ,
      filter: [
        7,
        'filter'
      ]
      ,
      sortable: [
        8,
        'sortable'
      ]
      ,
      groupable: [
        9,
        'groupable'
      ]

    }
    ,{sortChange: 'sortChange'}),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tbody',[[
        'kendoGridTableBody',
        ''
      ]
    ],(null as any),(null as any),(null as any),import20.View_TableBodyComponent_0,import20.RenderType_TableBodyComponent)),
    import0.ɵdid(286720,(null as any),0,import21.TableBodyComponent,[
      import16.DetailsService,
      import22.GroupsService,
      import17.ChangeNotificationService,
      import23.EditService,
      import4.LocalizationService
    ]
    ,{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]
      ,
      noRecordsTemplate: [
        3,
        'noRecordsTemplate'
      ]
      ,
      data: [
        4,
        'data'
      ]
      ,
      skip: [
        5,
        'skip'
      ]
      ,
      selectable: [
        6,
        'selectable'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_7)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.leafColumns;
    const currVal_1:any = co.group;
    const currVal_2:any = co.detailTemplate;
    ck(v,4,0,currVal_0,currVal_1,currVal_2);
    const currVal_4:any = co.totalColumnLevels;
    const currVal_5:any = co.columns;
    const currVal_6:any = co.group;
    const currVal_7:any = co.detailTemplate;
    const currVal_8:any = false;
    const currVal_9:any = co.filterable;
    const currVal_10:any = co.sort;
    const currVal_11:any = co.filter;
    const currVal_12:any = co.sortable;
    const currVal_13:any = co.showGroupPanel;
    ck(v,9,0,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13);
    const currVal_14:any = co.leafColumns;
    const currVal_15:any = co.group;
    const currVal_16:any = co.detailTemplate;
    const currVal_17:any = co.noRecordsTemplate;
    const currVal_18:any = co.view;
    const currVal_19:any = co.skip;
    const currVal_20:any = co.selectable;
    ck(v,13,0,currVal_14,currVal_15,currVal_16,currVal_17,currVal_18,currVal_19,currVal_20);
    const currVal_21:any = co.showFooter;
    ck(v,17,0,currVal_21);
  },(ck,v) => {
    const currVal_3:any = import0.ɵnov(v,9).headerClass;
    ck(v,7,0,currVal_3);
  });
}
function View_GridComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-pager',([] as any[]),[
      [
        2,
        'k-pager-wrap',
        (null as any)
      ]
      ,
      [
        2,
        'k-grid-pager',
        (null as any)
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'pageChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('pageChange' === en)) {
        const pd_0:any = ((<any>co.notifyPageChange('pager',$event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import24.View_PagerComponent_0,import24.RenderType_PagerComponent)),
    import0.ɵdid(385024,(null as any),0,import25.PagerComponent,[import26.PagerContextService],{
      total: [
        0,
        'total'
      ]
      ,
      skip: [
        1,
        'skip'
      ]
      ,
      pageSize: [
        2,
        'pageSize'
      ]
      ,
      options: [
        3,
        'options'
      ]
      ,
      template: [
        4,
        'template'
      ]

    }
    ,{pageChange: 'pageChange'}),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.view.total;
    const currVal_4:any = co.skip;
    const currVal_5:any = co.pageSize;
    const currVal_6:any = co.pageable;
    const currVal_7:any = co.pagerTemplate;
    ck(v,1,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).pagerWrapClass;
    const currVal_1:any = import0.ɵnov(v,1).gridPagerClass;
    const currVal_2:any = import0.ɵnov(v,1).widgetClass;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_GridComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{lockedHeader: 0}),
    import0.ɵqud(335544320,2,{header: 0}),
    import0.ɵqud(335544320,3,{footer: 0}),
    import0.ɵqud(335544320,4,{list: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(270336,(null as any),0,import27.LocalizedMessagesDirective,[import4.LocalizationService],{
      groupPanelEmpty: [
        0,
        'groupPanelEmpty'
      ]
      ,
      noRecords: [
        1,
        'noRecords'
      ]
      ,
      pagerFirstPage: [
        2,
        'pagerFirstPage'
      ]
      ,
      pagerLastPage: [
        3,
        'pagerLastPage'
      ]
      ,
      pagerPreviousPage: [
        4,
        'pagerPreviousPage'
      ]
      ,
      pagerNextPage: [
        5,
        'pagerNextPage'
      ]
      ,
      pagerPage: [
        6,
        'pagerPage'
      ]
      ,
      pagerItemsPerPage: [
        7,
        'pagerItemsPerPage'
      ]
      ,
      pagerOf: [
        8,
        'pagerOf'
      ]
      ,
      pagerItems: [
        9,
        'pagerItems'
      ]
      ,
      filterEqOperator: [
        10,
        'filterEqOperator'
      ]
      ,
      filterNotEqOperator: [
        11,
        'filterNotEqOperator'
      ]
      ,
      filterIsNullOperator: [
        12,
        'filterIsNullOperator'
      ]
      ,
      filterIsNotNullOperator: [
        13,
        'filterIsNotNullOperator'
      ]
      ,
      filterIsEmptyOperator: [
        14,
        'filterIsEmptyOperator'
      ]
      ,
      filterIsNotEmptyOperator: [
        15,
        'filterIsNotEmptyOperator'
      ]
      ,
      filterStartsWithOperator: [
        16,
        'filterStartsWithOperator'
      ]
      ,
      filterContainsOperator: [
        17,
        'filterContainsOperator'
      ]
      ,
      filterNotContainsOperator: [
        18,
        'filterNotContainsOperator'
      ]
      ,
      filterEndsWithOperator: [
        19,
        'filterEndsWithOperator'
      ]
      ,
      filterGteOperator: [
        20,
        'filterGteOperator'
      ]
      ,
      filterGtOperator: [
        21,
        'filterGtOperator'
      ]
      ,
      filterLteOperator: [
        22,
        'filterLteOperator'
      ]
      ,
      filterLtOperator: [
        23,
        'filterLtOperator'
      ]
      ,
      filterIsTrue: [
        24,
        'filterIsTrue'
      ]
      ,
      filterIsFalse: [
        25,
        'filterIsFalse'
      ]
      ,
      filterBooleanAll: [
        26,
        'filterBooleanAll'
      ]

    }
    ,(null as any)),
    import0.ɵprd(3072,(null as any),import28.Messages,(null as any),[import27.LocalizedMessagesDirective]),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_1)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_2)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_6)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GridComponent_8)),
    import0.ɵdid(8192,(null as any),0,import11.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import13.GridComponent = v.component;
    const currVal_0:any = 'Drag a column header and drop it here to group by that column';
    const currVal_1:any = 'No records available.';
    const currVal_2:any = 'Go to the first page';
    const currVal_3:any = 'Go to the last page';
    const currVal_4:any = 'Go to the previous page';
    const currVal_5:any = 'Go to the next page';
    const currVal_6:any = 'Page';
    const currVal_7:any = 'items per page';
    const currVal_8:any = 'of';
    const currVal_9:any = 'items';
    const currVal_10:any = 'Is equal to';
    const currVal_11:any = 'Is not equal to';
    const currVal_12:any = 'Is null';
    const currVal_13:any = 'Is not null';
    const currVal_14:any = 'Is empty';
    const currVal_15:any = 'Is not empty';
    const currVal_16:any = 'Starts with';
    const currVal_17:any = 'Contains';
    const currVal_18:any = 'Does not contain';
    const currVal_19:any = 'Ends with';
    const currVal_20:any = 'Is greater than or equal to';
    const currVal_21:any = 'Is greater than';
    const currVal_22:any = 'Is less than or equal to';
    const currVal_23:any = 'Is less than';
    const currVal_24:any = 'Is True';
    const currVal_25:any = 'Is False';
    const currVal_26:any = '(All)';
    ck(v,6,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17,
      currVal_18,
      currVal_19,
      currVal_20,
      currVal_21,
      currVal_22,
      currVal_23,
      currVal_24,
      currVal_25,
      currVal_26
    ]
    );
    const currVal_27:any = co.showGroupPanel;
    ck(v,13,0,currVal_27);
    const currVal_28:any = co.isScrollable;
    ck(v,16,0,currVal_28);
    const currVal_29:boolean = !co.isScrollable;
    ck(v,19,0,currVal_29);
    const currVal_30:any = co.showPager;
    ck(v,22,0,currVal_30);
  },(null as any));
}
function View_GridComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),21,'kendo-grid',([] as any[]),[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-grid',
        (null as any)
      ]
      ,
      [
        2,
        'k-grid-lockedcolumns',
        (null as any)
      ]
      ,
      [
        2,
        'k-grid-virtual',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_GridComponent_0,RenderType_GridComponent)),
    import0.ɵprd(256,(null as any),import29.BrowserSupportService,import29.BrowserSupportService,([] as any[])),
    import0.ɵprd(256,(null as any),import30.SelectionService,import30.SelectionService,([] as any[])),
    import0.ɵprd(256,(null as any),import31.GroupInfoService,import31.GroupInfoService,([] as any[])),
    import0.ɵprd(256,(null as any),import22.GroupsService,import22.GroupsService,([] as any[])),
    import0.ɵprd(256,(null as any),import17.ChangeNotificationService,import17.ChangeNotificationService,[import0.NgZone]),
    import0.ɵprd(256,(null as any),import16.DetailsService,import16.DetailsService,([] as any[])),
    import0.ɵprd(256,(null as any),import23.EditService,import23.EditService,([] as any[])),
    import0.ɵprd(256,(null as any),import32.FilterService,import32.FilterService,([] as any[])),
    import0.ɵprd(256,(null as any),import33.PDFService,import33.PDFService,([] as any[])),
    import0.ɵprd(256,(null as any),import34.ResponsiveService,import34.ResponsiveService,([] as any[])),
    import0.ɵdid(4022272,(null as any),4,import13.GridComponent,[
      import29.BrowserSupportService,
      import30.SelectionService,
      import0.ElementRef,
      import31.GroupInfoService,
      import22.GroupsService,
      import17.ChangeNotificationService,
      import16.DetailsService,
      import23.EditService,
      import32.FilterService,
      import33.PDFService,
      import34.ResponsiveService,
      import0.Renderer2,
      [
        2,
        import19.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{columns: 1}),
    import0.ɵqud(167772160,2,{detailTemplate: 0}),
    import0.ɵqud(167772160,3,{noRecordsTemplate: 0}),
    import0.ɵqud(167772160,4,{pagerTemplate: 0}),
    import0.ɵprd(128,(null as any),import35.CELL_CONTEXT,{},([] as any[])),
    import0.ɵprd(128,(null as any),import4.L10N_PREFIX,'kendo.grid',([] as any[])),
    import0.ɵprd(2304,(null as any),import4.LocalizationService,import4.LocalizationService,[
      import4.L10N_PREFIX,
      [
        2,
        import36.MessageService
      ]

    ]
    ),
    import0.ɵprd(2304,(null as any),import2.GroupConnectionService,import2.GroupConnectionService,([] as any[])),
    import0.ɵprd(2304,(null as any),import18.SuspendService,import18.SuspendService,([] as any[])),
    import0.ɵprd(2304,(null as any),import26.PagerContextService,import26.PagerContextService,([] as any[]))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,11).dir;
    const currVal_1:any = import0.ɵnov(v,11).hostClasses;
    const currVal_2:any = import0.ɵnov(v,11).hostClasses;
    const currVal_3:any = import0.ɵnov(v,11).lockedClasses;
    const currVal_4:any = import0.ɵnov(v,11).virtualClasses;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export const GridComponentNgFactory:import0.ComponentFactory<import13.GridComponent> = import0.ɵccf('kendo-grid',import13.GridComponent,View_GridComponent_Host_0,{
  data: 'data',
  pageSize: 'pageSize',
  height: 'height',
  rowHeight: 'rowHeight',
  detailRowHeight: 'detailRowHeight',
  skip: 'skip',
  scrollable: 'scrollable',
  selectable: 'selectable',
  sort: 'sort',
  filter: 'filter',
  group: 'group',
  filterable: 'filterable',
  sortable: 'sortable',
  pageable: 'pageable',
  groupable: 'groupable'
}
,{
  filterChange: 'filterChange',
  pageChange: 'pageChange',
  groupChange: 'groupChange',
  sortChange: 'sortChange',
  selectionChange: 'selectionChange',
  dataStateChange: 'dataStateChange',
  groupExpand: 'groupExpand',
  groupCollapse: 'groupCollapse',
  detailExpand: 'detailExpand',
  detailCollapse: 'detailCollapse',
  edit: 'edit',
  cancel: 'cancel',
  save: 'save',
  remove: 'remove',
  add: 'add',
  pdfExport: 'pdfExport'
}
,['kendo-grid-toolbar']);
