// @progress/kendo-angular-grid/dist/es/header.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/template-context.directive';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-grid/dist/es/draggable.directive';
import * as import4 from './filtering/filter-row.component.ngfactory';
import * as import5 from '@progress/kendo-angular-grid/dist/es/filtering/filter-row.component';
import * as import6 from '@progress/kendo-angular-grid/dist/es/grouping/group-connection.service';
import * as import7 from '@progress/kendo-angular-grid/dist/es/header.component';
const styles_HeaderComponent:any[] = ([] as any[]);
export const RenderType_HeaderComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_HeaderComponent,
  data: {}
}
);
function View_HeaderComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',([] as any[]),[
      [
        2,
        'k-group-cell',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_HeaderComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',([] as any[]),[
      [
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_HeaderComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                        ']))],(null as any),(null as any));
}
function View_HeaderComponent_9(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    const currVal_0:any = ((<any>(<any>(<any>v.parent).parent).parent).context.$implicit.title || (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.field);
    ck(v,0,0,currVal_0);
  });
}
function View_HeaderComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_HeaderComponent_8)),
      import0.ɵdid(73728,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'columnIndex',
      'column',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,3,0,(<any>(<any>v.parent).parent).context.$implicit.headerTemplateRef,(co.lockedColumnsCount + (<any>(<any>v.parent).parent).context.index),(<any>(<any>v.parent).parent).context.$implicit,(<any>(<any>v.parent).parent).context.$implicit);
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !(<any>(<any>v.parent).parent).context.$implicit.headerTemplateRef;
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_HeaderComponent_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                            ']))],(null as any),(null as any));
}
function View_HeaderComponent_12(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    const currVal_0:any = ((<any>(<any>(<any>v.parent).parent).parent).context.$implicit.title || (<any>(<any>(<any>v.parent).parent).parent).context.$implicit.field);
    ck(v,0,0,currVal_0);
  });
}
function View_HeaderComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                        '])),
      (l()(),import0.ɵeld(0,[[
        'link',
        1
      ]
    ],(null as any),11,'a',[
      [
        'class',
        'k-link'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.sortColumn((<any>(<any>v.parent).parent).context.$implicit,$event,import0.ɵnov(v,1),import0.ɵnov(v,10))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_HeaderComponent_11)),
      import0.ɵdid(73728,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'columnIndex',
      'column',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_12)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,[[
        'sortSpan',
        1
      ]
    ],(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,5,0,(<any>(<any>v.parent).parent).context.$implicit.headerTemplateRef,(<any>(<any>v.parent).parent).context.index,(<any>(<any>v.parent).parent).context.$implicit,(<any>(<any>v.parent).parent).context.$implicit);
    ck(v,4,0,currVal_0);
    const currVal_1:boolean = !(<any>(<any>v.parent).parent).context.$implicit.headerTemplateRef;
    ck(v,8,0,currVal_1);
    const currVal_2:any = co.sortIcon((<any>(<any>v.parent).parent).context.$implicit.field);
    ck(v,11,0,currVal_2);
  },(null as any));
}
function View_HeaderComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'th',([] as any[]),[
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        2,
        'k-first',
        (null as any)
      ]
      ,
      [
        1,
        'rowspan',
        0
      ]
      ,
      [
        1,
        'colspan',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
      import0.ɵdid(73728,[[
        1,
        4
      ]
      ],0,import3.DraggableDirective,[import0.ElementRef],{kendoGridDraggable: [
        0,
        'kendoGridDraggable'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_10)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = (<any>v.parent).context.$implicit.headerClass;
    ck(v,1,0,currVal_4);
    const currVal_5:any = (<any>v.parent).context.$implicit.headerStyle;
    ck(v,2,0,currVal_5);
    const currVal_6:any = (<any>v.parent).context.$implicit;
    ck(v,3,0,currVal_6);
    const currVal_7:boolean = !co.isSortable((<any>v.parent).context.$implicit);
    ck(v,6,0,currVal_7);
    const currVal_8:any = co.isSortable((<any>v.parent).context.$implicit);
    ck(v,9,0,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = true;
    const currVal_1:any = co.isFirstOnRow((<any>v.parent).context.$implicit,(<any>v.parent).context.index);
    const currVal_2:any = (<any>v.parent).context.$implicit.rowspan(co.totalColumnLevels);
    const currVal_3:any = (<any>v.parent).context.$implicit.colspan;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_HeaderComponent_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                        ']))],(null as any),(null as any));
}
function View_HeaderComponent_15(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.title;
    ck(v,0,0,currVal_0);
  });
}
function View_HeaderComponent_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'th',([] as any[]),[
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        2,
        'k-first',
        (null as any)
      ]
      ,
      [
        1,
        'rowspan',
        0
      ]
      ,
      [
        1,
        'colspan',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import2.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_HeaderComponent_14)),
      import0.ɵdid(73728,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'columnIndex',
      'column',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_15)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = (<any>v.parent).context.$implicit.headerClass;
    ck(v,1,0,currVal_4);
    const currVal_5:any = (<any>v.parent).context.$implicit.headerStyle;
    ck(v,2,0,currVal_5);
    const currVal_6:any = ck(v,6,0,(<any>v.parent).context.$implicit.headerTemplateRef,(co.lockedColumnsCount + (<any>v.parent).context.index),(<any>v.parent).context.$implicit,(<any>v.parent).context.$implicit);
    ck(v,5,0,currVal_6);
    const currVal_7:boolean = !(<any>v.parent).context.$implicit.headerTemplateRef;
    ck(v,9,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = true;
    const currVal_1:any = co.isFirstOnRow((<any>v.parent).context.$implicit,(<any>v.parent).context.index);
    const currVal_2:any = (<any>v.parent).context.$implicit.rowspan(co.totalColumnLevels);
    const currVal_3:any = (<any>v.parent).context.$implicit.colspan;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_HeaderComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_13)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.isColumnGroupComponent(v.context.$implicit);
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.isColumnGroupComponent(v.context.$implicit);
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_HeaderComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_3)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_5)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groups;
    ck(v,3,0,currVal_0);
    const currVal_1:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.columnsForLevel(v.context.index);
    ck(v,9,0,currVal_2);
  },(null as any));
}
function View_HeaderComponent_16(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'tr',[[
        'kendoGridFilterRow',
        ''
      ]
      ],[[
        2,
        'k-filter-row',
        (null as any)
      ]
    ],(null as any),(null as any),import4.View_FilterRowComponent_0,import4.RenderType_FilterRowComponent)),
    import0.ɵdid(24576,(null as any),0,import5.FilterRowComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      filter: [
        1,
        'filter'
      ]
      ,
      groups: [
        2,
        'groups'
      ]
      ,
      detailTemplate: [
        3,
        'detailTemplate'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.leafColumns;
    const currVal_2:any = co.filter;
    const currVal_3:any = co.groups;
    const currVal_4:any = co.detailTemplate;
    ck(v,1,0,currVal_1,currVal_2,currVal_3,currVal_4);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).filterRowClass;
    ck(v,0,0,currVal_0);
  });
}
function View_HeaderComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_16)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.columnLevels;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.filterable;
    ck(v,5,0,currVal_1);
  },(null as any));
}
export function View_HeaderComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{draggables: 1}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_HeaderComponent_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = true;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_HeaderComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'kendoGridHeader',
        ''
      ]
      ],[[
        2,
        'k-grid-header',
        (null as any)
      ]
    ],(null as any),(null as any),View_HeaderComponent_0,RenderType_HeaderComponent)),
    import0.ɵprd(256,(null as any),import6.GroupDragService,import6.GroupDragService,[
      import6.GroupConnectionService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(2121728,(null as any),0,import7.HeaderComponent,[import6.GroupDragService],(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).headerClass;
    ck(v,0,0,currVal_0);
  });
}
export const HeaderComponentNgFactory:import0.ComponentFactory<import7.HeaderComponent> = import0.ɵccf('[kendoGridHeader]',import7.HeaderComponent,View_HeaderComponent_Host_0,{
  totalColumnLevels: 'totalColumnLevels',
  columns: 'columns',
  groups: 'groups',
  detailTemplate: 'detailTemplate',
  scrollable: 'scrollable',
  filterable: 'filterable',
  sort: 'sort',
  filter: 'filter',
  sortable: 'sortable',
  groupable: 'groupable',
  lockedColumnsCount: 'lockedColumnsCount'
}
,{sortChange: 'sortChange'},([] as any[]));
