// @progress/kendo-angular-grid/dist/es/list.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from './col-group.component.ngfactory';
import * as import3 from '@progress/kendo-angular-grid/dist/es/col-group.component';
import * as import4 from './table-body.component.ngfactory';
import * as import5 from '@progress/kendo-angular-grid/dist/es/table-body.component';
import * as import6 from '@progress/kendo-angular-grid/dist/es/details.service';
import * as import7 from '@progress/kendo-angular-grid/dist/es/grouping/groups.service';
import * as import8 from '@progress/kendo-angular-grid/dist/es/change-notification.service';
import * as import9 from '@progress/kendo-angular-grid/dist/es/edit.service';
import * as import10 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import11 from '@progress/kendo-angular-grid/dist/es/resizable.directive';
import * as import12 from '@progress/kendo-angular-grid/dist/es/grid.component';
import * as import13 from '@progress/kendo-angular-grid/dist/es/list.component';
import * as import14 from '@progress/kendo-angular-grid/dist/es/suspend.service';
import * as import15 from '@progress/kendo-angular-l10n/dist/es/rtl';
const styles_ListComponent:any[] = ([] as any[]);
export const RenderType_ListComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ListComponent,
  data: {}
}
);
function View_ListComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'lockedContainer',
        1
      ]

    ]
      ,(null as any),18,'div',[[
        'class',
        'k-grid-content-locked'
      ]
    ],[
      [
        4,
        'height',
        'px'
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'table',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import2.View_ColGroupComponent_0,import2.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import3.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tbody',[[
        'kendoGridTableBody',
        ''
      ]
    ],(null as any),(null as any),(null as any),import4.View_TableBodyComponent_0,import4.RenderType_TableBodyComponent)),
    import0.ɵdid(286720,(null as any),0,import5.TableBodyComponent,[
      import6.DetailsService,
      import7.GroupsService,
      import8.ChangeNotificationService,
      import9.EditService,
      import10.LocalizationService
    ]
    ,{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]
      ,
      data: [
        3,
        'data'
      ]
      ,
      skip: [
        4,
        'skip'
      ]
      ,
      selectable: [
        5,
        'selectable'
      ]
      ,
      noRecordsText: [
        6,
        'noRecordsText'
      ]
      ,
      showGroupFooters: [
        7,
        'showGroupFooters'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'k-height-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',([] as any[]),[[
        4,
        'height',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.style;
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.lockedLeafColumns;
    const currVal_4:any = co.groups;
    const currVal_5:any = co.detailTemplate;
    ck(v,6,0,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = co.lockedLeafColumns;
    const currVal_7:any = co.groups;
    const currVal_8:any = co.detailTemplate;
    const currVal_9:any = co.data;
    const currVal_10:any = co.skip;
    const currVal_11:any = co.selectable;
    const currVal_12:any = '';
    const currVal_13:any = co.showFooter;
    ck(v,10,0,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.height;
    const currVal_1:any = co.lockedWidth;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_14:any = co.totalHeight;
    ck(v,16,0,currVal_14);
  });
}
export function View_ListComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{container: 0}),
    import0.ɵqud(335544320,2,{lockedContainer: 0}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),19,'div',[[
        'class',
        'k-grid-content k-virtual-content'
      ]
      ],[[
        4,
        'height',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import11.ResizableContainerDirective,[
      import0.ElementRef,
      import0.Renderer2,
      [
        2,
        import12.GridComponent
      ]

    ]
    ,{
      lockedWidth: [
        0,
        'lockedWidth'
      ]
      ,
      kendoGridResizableContainer: [
        1,
        'kendoGridResizableContainer'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'table',([] as any[]),[[
        4,
        'width',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'colgroup',[[
        'kendoGridColGroup',
        ''
      ]
    ],(null as any),(null as any),(null as any),import2.View_ColGroupComponent_0,import2.RenderType_ColGroupComponent)),
    import0.ɵdid(24576,(null as any),0,import3.ColGroupComponent,([] as any[]),{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tbody',[[
        'kendoGridTableBody',
        ''
      ]
    ],(null as any),(null as any),(null as any),import4.View_TableBodyComponent_0,import4.RenderType_TableBodyComponent)),
    import0.ɵdid(286720,(null as any),0,import5.TableBodyComponent,[
      import6.DetailsService,
      import7.GroupsService,
      import8.ChangeNotificationService,
      import9.EditService,
      import10.LocalizationService
    ]
    ,{
      columns: [
        0,
        'columns'
      ]
      ,
      groups: [
        1,
        'groups'
      ]
      ,
      detailTemplate: [
        2,
        'detailTemplate'
      ]
      ,
      noRecordsTemplate: [
        3,
        'noRecordsTemplate'
      ]
      ,
      data: [
        4,
        'data'
      ]
      ,
      skip: [
        5,
        'skip'
      ]
      ,
      selectable: [
        6,
        'selectable'
      ]
      ,
      skipGroupDecoration: [
        7,
        'skipGroupDecoration'
      ]
      ,
      showGroupFooters: [
        8,
        'showGroupFooters'
      ]
      ,
      lockedColumnsCount: [
        9,
        'lockedColumnsCount'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'k-height-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',([] as any[]),[[
        4,
        'height',
        'px'
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import13.ListComponent = v.component;
    const currVal_0:any = co.isLocked;
    ck(v,4,0,currVal_0);
    const currVal_2:any = (co.lockedWidth + 1);
    const currVal_3:any = co.lockedLeafColumns.length;
    ck(v,6,0,currVal_2,currVal_3);
    const currVal_5:any = co.style;
    ck(v,9,0,currVal_5);
    const currVal_6:any = co.nonLockedLeafColumns;
    const currVal_7:any = (co.isLocked? import0.ɵEMPTY_ARRAY: co.groups);
    const currVal_8:any = co.detailTemplate;
    ck(v,12,0,currVal_6,currVal_7,currVal_8);
    const currVal_9:any = co.nonLockedLeafColumns;
    const currVal_10:any = co.groups;
    const currVal_11:any = co.detailTemplate;
    const currVal_12:any = co.noRecordsTemplate;
    const currVal_13:any = co.data;
    const currVal_14:any = co.skip;
    const currVal_15:any = co.selectable;
    const currVal_16:any = co.isLocked;
    const currVal_17:any = co.showFooter;
    const currVal_18:any = co.lockedLeafColumns.length;
    ck(v,16,0,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17,currVal_18);
  },(ck,v) => {
    var co:import13.ListComponent = v.component;
    const currVal_1:any = co.height;
    ck(v,5,0,currVal_1);
    const currVal_4:any = co.nonLockedWidth;
    ck(v,8,0,currVal_4);
    const currVal_19:any = co.totalHeight;
    ck(v,22,0,currVal_19);
  });
}
function View_ListComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-list',([] as any[]),(null as any),(null as any),(null as any),View_ListComponent_0,RenderType_ListComponent)),
    import0.ɵprd(128,(null as any),import13.SCROLLER_FACTORY_TOKEN,import13.DEFAULT_SCROLLER_FACTORY,([] as any[])),
    import0.ɵdid(2482176,(null as any),0,import13.ListComponent,[
      import13.SCROLLER_FACTORY_TOKEN,
      import6.DetailsService,
      import8.ChangeNotificationService,
      import14.SuspendService,
      [
        2,
        import15.RTL
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(null as any));
}
export const ListComponentNgFactory:import0.ComponentFactory<import13.ListComponent> = import0.ɵccf('kendo-grid-list',import13.ListComponent,View_ListComponent_Host_0,{
  data: 'data',
  groups: 'groups',
  total: 'total',
  height: 'height',
  rowHeight: 'rowHeight',
  detailRowHeight: 'detailRowHeight',
  take: 'take',
  skip: 'skip',
  columns: 'columns',
  detailTemplate: 'detailTemplate',
  noRecordsTemplate: 'noRecordsTemplate',
  selectable: 'selectable',
  groupable: 'groupable'
}
,{
  pageChange: 'pageChange',
  containerScroll: 'containerScroll'
}
,([] as any[]));
