// @progress/kendo-angular-grid/dist/es/table-body.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-grid/dist/es/field-accessor.pipe';
import * as import3 from '@progress/kendo-angular-intl/dist/es/intl.service';
import * as import4 from './cell.component.ngfactory';
import * as import5 from '@progress/kendo-angular-grid/dist/es/cell.component';
import * as import6 from '@progress/kendo-angular-grid/dist/es/edit.service';
import * as import7 from '@progress/kendo-angular-grid/dist/es/cell-context';
import * as import8 from '@progress/kendo-angular-grid/dist/es/template-context.directive';
import * as import9 from './grouping/group-header.component.ngfactory';
import * as import10 from '@progress/kendo-angular-grid/dist/es/grouping/group-header.component';
import * as import11 from '@progress/kendo-angular-grid/dist/es/grouping/groups.service';
import * as import12 from '@progress/kendo-angular-grid/dist/es/grouping/group-info.service';
import * as import13 from '@progress/kendo-angular-grid/dist/es/selectable.directive';
import * as import14 from '@progress/kendo-angular-grid/dist/es/selection.service';
import * as import15 from '@progress/kendo-angular-grid/dist/es/table-body.component';
import * as import16 from '@progress/kendo-angular-grid/dist/es/details.service';
import * as import17 from '@progress/kendo-angular-grid/dist/es/change-notification.service';
import * as import18 from '@progress/kendo-angular-l10n/dist/es/localization.service';
const styles_TableBodyComponent:any[] = ([] as any[]);
export const RenderType_TableBodyComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TableBodyComponent,
  data: {}
}
);
function View_TableBodyComponent_3(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'td',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_3)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groups;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_TableBodyComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),[[
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_TableBodyComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    import0.ɵpid(0,import2.FieldAccessorPipe,[import3.IntlService])
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵunv(v,0,0,import0.ɵnov(v,1).transform(co.newDataItem,(<any>v.parent).context.$implicit.field,(<any>v.parent).context.$implicit.format));
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'td',[[
        'kendoGridCell',
        ''
      ]
    ],(null as any),(null as any),(null as any),import4.View_CellComponent_0,import4.RenderType_CellComponent)),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵdid(24576,(null as any),0,import5.CellComponent,[
      import6.EditService,
      [
        2,
        import7.CELL_CONTEXT
      ]

    ]
    ,{
      column: [
        0,
        'column'
      ]
      ,
      isNew: [
        1,
        'isNew'
      ]
      ,
      rowIndex: [
        2,
        'rowIndex'
      ]
      ,
      dataItem: [
        3,
        'dataItem'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,2,(null as any),View_TableBodyComponent_6)),
      import0.ɵdid(73728,(null as any),0,import8.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'dataItem',
      'column',
      'columnIndex',
      'rowIndex',
      'isNew',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_TableBodyComponent_7)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit.cssClass;
    ck(v,1,0,currVal_0);
    const currVal_1:any = v.context.$implicit.style;
    ck(v,2,0,currVal_1);
    const currVal_2:any = v.context.$implicit;
    const currVal_3:any = true;
    const currVal_4:any = (0 - 1);
    const currVal_5:any = co.newDataItem;
    ck(v,3,0,currVal_2,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = ck(v,7,0,v.context.$implicit.templateRef,co.newDataItem,v.context.$implicit,v.context.index,(0 - 1),true,co.item);
    ck(v,6,0,currVal_6);
    const currVal_7:any = co.isBoundColumn(v.context.$implicit);
    ck(v,10,0,currVal_7);
  },(null as any));
}
function View_TableBodyComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',[[
        'class',
        'k-grid-add-row k-grid-edit-row'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_5)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.skipGroupDecoration;
    ck(v,4,0,currVal_0);
    const currVal_1:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,7,0,currVal_1);
    const currVal_2:any = co.columns;
    ck(v,10,0,currVal_2);
  },(null as any));
}
function View_TableBodyComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_TableBodyComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                ',
      '\n            '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.noRecordsText;
    ck(v,1,0,currVal_0);
  });
}
function View_TableBodyComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'tr',[[
        'class',
        'k-grid-norecords'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'td',([] as any[]),[[
        1,
        'colspan',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),3,(null as any),View_TableBodyComponent_9)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      import0.ɵdid(73728,(null as any),0,import8.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_10)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = ((co.noRecordsTemplate == null)? (null as any): co.noRecordsTemplate.templateRef);
    ck(v,5,0,currVal_1);
    const currVal_2:any = ck(v,7,0,((co.noRecordsTemplate == null)? (null as any): co.noRecordsTemplate.templateRef));
    ck(v,6,0,currVal_2);
    const currVal_3:boolean = !((co.noRecordsTemplate == null)? (null as any): co.noRecordsTemplate.templateRef);
    ck(v,10,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.colSpan;
    ck(v,2,0,currVal_0);
  });
}
function View_TableBodyComponent_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'tr',[[
        'kendoGridGroupHeader',
        ''
      ]
      ],[[
        2,
        'k-grouping-row',
        (null as any)
      ]
    ],(null as any),(null as any),import9.View_GroupHeaderComponent_0,import9.RenderType_GroupHeaderComponent)),
    import0.ɵdid(24576,(null as any),0,import10.GroupHeaderComponent,[
      import11.GroupsService,
      import12.GroupInfoService
    ]
    ,{
      item: [
        0,
        'item'
      ]
      ,
      skipGroupDecoration: [
        1,
        'skipGroupDecoration'
      ]
      ,
      columns: [
        2,
        'columns'
      ]
      ,
      groups: [
        3,
        'groups'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (<any>v.parent).context.$implicit;
    const currVal_2:any = co.skipGroupDecoration;
    const currVal_3:any = co.columns;
    const currVal_4:any = co.groups;
    ck(v,1,0,currVal_1,currVal_2,currVal_3,currVal_4);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).groupItemClass;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_15(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'td',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_15)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groups;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_TableBodyComponent_17(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'class',
        'k-icon'
      ]
      ,
      [
        'href',
        '#'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleRow((<any>(<any>(<any>v.parent).parent).parent).context.$implicit.index,(<any>(<any>(<any>v.parent).parent).parent).context.$implicit.data)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'k-icon';
    const currVal_1:any = co.detailButtonStyles((<any>(<any>(<any>v.parent).parent).parent).context.$implicit.index);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_TableBodyComponent_16(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',([] as any[]),[[
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_17)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.detailTemplate.showIf((<any>(<any>v.parent).parent).context.$implicit.data,(<any>(<any>v.parent).parent).context.$implicit.index);
    ck(v,3,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_19(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_TableBodyComponent_20(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    import0.ɵpid(0,import2.FieldAccessorPipe,[import3.IntlService])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵunv(v,0,0,import0.ɵnov(v,1).transform((<any>(<any>(<any>v.parent).parent).parent).context.$implicit.data,(<any>v.parent).context.$implicit.field,(<any>v.parent).context.$implicit.format));
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_18(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'td',[[
        'kendoGridCell',
        ''
      ]
    ],(null as any),(null as any),(null as any),import4.View_CellComponent_0,import4.RenderType_CellComponent)),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵdid(24576,(null as any),0,import5.CellComponent,[
      import6.EditService,
      [
        2,
        import7.CELL_CONTEXT
      ]

    ]
    ,{
      column: [
        0,
        'column'
      ]
      ,
      rowIndex: [
        1,
        'rowIndex'
      ]
      ,
      dataItem: [
        2,
        'dataItem'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,2,(null as any),View_TableBodyComponent_19)),
      import0.ɵdid(73728,(null as any),0,import8.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'dataItem',
      'column',
      'columnIndex',
      'rowIndex',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_TableBodyComponent_20)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit.cssClass;
    ck(v,1,0,currVal_0);
    const currVal_1:any = v.context.$implicit.style;
    ck(v,2,0,currVal_1);
    const currVal_2:any = v.context.$implicit;
    const currVal_3:any = (<any>(<any>v.parent).parent).context.$implicit.index;
    const currVal_4:any = (<any>(<any>v.parent).parent).context.$implicit.data;
    ck(v,3,0,currVal_2,currVal_3,currVal_4);
    const currVal_5:any = ck(v,7,0,v.context.$implicit.templateRef,(<any>(<any>v.parent).parent).context.$implicit.data,v.context.$implicit,(co.lockedColumnsCount + v.context.index),(<any>(<any>v.parent).parent).context.$implicit.index,(<any>(<any>v.parent).parent).context.$implicit.data);
    ck(v,6,0,currVal_5);
    const currVal_6:any = co.isBoundColumn(v.context.$implicit);
    ck(v,10,0,currVal_6);
  },(null as any));
}
function View_TableBodyComponent_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'tr',([] as any[]),[
      [
        2,
        'k-alt',
        (null as any)
      ]
      ,
      [
        2,
        'k-master-row',
        (null as any)
      ]
      ,
      [
        2,
        'k-grid-edit-row',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-selected',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onClick($event.target)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import13.SelectableDirective,[import14.SelectionService],{
      index: [
        0,
        'index'
      ]
      ,
      kendoGridSelectable: [
        1,
        'kendoGridSelectable'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_14)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_16)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_18)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = (<any>v.parent).context.$implicit.index;
    const currVal_5:any = co.selectable;
    ck(v,1,0,currVal_4,currVal_5);
    const currVal_6:boolean = !co.skipGroupDecoration;
    ck(v,4,0,currVal_6);
    const currVal_7:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,7,0,currVal_7);
    const currVal_8:any = co.columns;
    ck(v,10,0,currVal_8);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isOdd((<any>v.parent).context.$implicit);
    const currVal_1:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    const currVal_2:any = co.editService.isEdited((<any>v.parent).context.$implicit.index);
    const currVal_3:any = import0.ɵnov(v,1).className;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_TableBodyComponent_22(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'td',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_23(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_TableBodyComponent_21(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),13,'tr',([] as any[]),[
      [
        2,
        'k-detail-row',
        (null as any)
      ]
      ,
      [
        2,
        'k-alt',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_22)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'td',([] as any[]),[[
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'td',([] as any[]),[
      [
        2,
        'k-detail-cell',
        (null as any)
      ]
      ,
      [
        1,
        'colspan',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_TableBodyComponent_23)),
      import0.ɵdid(73728,(null as any),0,import8.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'dataItem',
      'rowIndex',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.groups;
    ck(v,3,0,currVal_2);
    const currVal_6:any = ck(v,11,0,((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef),(<any>v.parent).context.$implicit.data,(<any>v.parent).context.$implicit.index,(<any>v.parent).context.$implicit.data);
    ck(v,10,0,currVal_6);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = true;
    const currVal_1:any = co.isOdd((<any>v.parent).context.$implicit);
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_3:any = true;
    ck(v,5,0,currVal_3);
    const currVal_4:any = true;
    const currVal_5:any = co.columns.length;
    ck(v,7,0,currVal_4,currVal_5);
  });
}
function View_TableBodyComponent_26(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'td',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_25(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_26)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groups;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_TableBodyComponent_27(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),[[
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_29(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_TableBodyComponent_28(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'td',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_TableBodyComponent_29)),
      import0.ɵdid(73728,(null as any),0,import8.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'group',
      'field',
      'column',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n           ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,4,0,v.context.$implicit.groupFooterTemplateRef,(<any>(<any>v.parent).parent).context.$implicit.data,v.context.$implicit.field,v.context.$implicit,(((<any>(<any>v.parent).parent).context.$implicit.data == null)? (null as any): (<any>(<any>v.parent).parent).context.$implicit.data.aggregates));
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_TableBodyComponent_24(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',([] as any[]),[[
        2,
        'k-group-footer',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_25)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_27)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_28)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:boolean = !co.skipGroupDecoration;
    ck(v,3,0,currVal_1);
    const currVal_2:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,6,0,currVal_2);
    const currVal_3:any = co.columns;
    ck(v,9,0,currVal_3);
  },(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_TableBodyComponent_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_12)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_13)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_21)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_24)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.isGroup(v.context.$implicit) && co.isParentGroupExpanded(v.context.$implicit));
    ck(v,2,0,currVal_0);
    const currVal_1:any = (co.isDataItem(v.context.$implicit) && co.isInExpandedGroup(v.context.$implicit));
    ck(v,5,0,currVal_1);
    const currVal_2:any = (((co.isDataItem(v.context.$implicit) && ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef)) && co.detailTemplate.showIf(v.context.$implicit.data,v.context.$implicit.index)) && co.isExpanded(v.context.$implicit.index));
    ck(v,8,0,currVal_2);
    const currVal_3:any = (co.isFooter(v.context.$implicit) && (co.isInExpandedGroup(v.context.$implicit) || (co.showGroupFooters && co.isParentGroupExpanded(v.context.$implicit))));
    ck(v,11,0,currVal_3);
  },(null as any));
}
export function View_TableBodyComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_8)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_TableBodyComponent_11)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
    ,{
      ngForOf: [
        0,
        'ngForOf'
      ]
      ,
      ngForTrackBy: [
        1,
        'ngForTrackBy'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import15.TableBodyComponent = v.component;
    const currVal_0:any = co.editService.hasNewItem;
    ck(v,2,0,currVal_0);
    const currVal_1:any = ((((co.data == null)? (null as any): co.data.length) === 0) || (co.data == (null as any)));
    ck(v,5,0,currVal_1);
    const currVal_2:any = co.data;
    const currVal_3:any = co.trackByFn;
    ck(v,8,0,currVal_2,currVal_3);
  },(null as any));
}
function View_TableBodyComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridTableBody',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_TableBodyComponent_0,RenderType_TableBodyComponent)),
    import0.ɵdid(286720,(null as any),0,import15.TableBodyComponent,[
      import16.DetailsService,
      import11.GroupsService,
      import17.ChangeNotificationService,
      import6.EditService,
      import18.LocalizationService
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const TableBodyComponentNgFactory:import0.ComponentFactory<import15.TableBodyComponent> = import0.ɵccf('[kendoGridTableBody]',import15.TableBodyComponent,View_TableBodyComponent_Host_0,{
  columns: 'columns',
  groups: 'groups',
  detailTemplate: 'detailTemplate',
  noRecordsTemplate: 'noRecordsTemplate',
  data: 'data',
  skip: 'skip',
  selectable: 'selectable',
  noRecordsText: 'noRecordsText',
  skipGroupDecoration: 'skipGroupDecoration',
  showGroupFooters: 'showGroupFooters',
  lockedColumnsCount: 'lockedColumnsCount'
}
,{},([] as any[]));
