// @progress/kendo-angular-grid/dist/es/toolbar.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/toolbar.component';
const styles_ToolbarComponent:any[] = ([] as any[]);
export const RenderType_ToolbarComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ToolbarComponent,
  data: {}
}
);
export function View_ToolbarComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_ToolbarComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-grid-toolbar',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),View_ToolbarComponent_0,RenderType_ToolbarComponent)),
    import0.ɵdid(24576,(null as any),0,import1.ToolbarComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).classNames;
    ck(v,0,0,currVal_0);
  });
}
export const ToolbarComponentNgFactory:import0.ComponentFactory<import1.ToolbarComponent> = import0.ɵccf('kendo-grid-toolbar',import1.ToolbarComponent,View_ToolbarComponent_Host_0,{},{},['*']);
