// @angular/http/http.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/http';
class HttpModuleInjector extends import0.ɵNgModuleInjector<import1.HttpModule> {
  _HttpModule_0:import1.HttpModule;
  __BrowserXhr_1:import1.BrowserXhr;
  __ResponseOptions_2:import1.BaseResponseOptions;
  __XSRFStrategy_3:any;
  __XHRBackend_4:import1.XHRBackend;
  __RequestOptions_5:import1.BaseRequestOptions;
  __Http_6:any;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _BrowserXhr_1():import1.BrowserXhr {
    if ((this.__BrowserXhr_1 == null)) { (this.__BrowserXhr_1 = new import1.BrowserXhr()); }
    return this.__BrowserXhr_1;
  }
  get _ResponseOptions_2():import1.BaseResponseOptions {
    if ((this.__ResponseOptions_2 == null)) { (this.__ResponseOptions_2 = new import1.BaseResponseOptions()); }
    return this.__ResponseOptions_2;
  }
  get _XSRFStrategy_3():any {
    if ((this.__XSRFStrategy_3 == null)) { (this.__XSRFStrategy_3 = import1.ɵb()); }
    return this.__XSRFStrategy_3;
  }
  get _XHRBackend_4():import1.XHRBackend {
    if ((this.__XHRBackend_4 == null)) { (this.__XHRBackend_4 = new import1.XHRBackend(this._BrowserXhr_1,this._ResponseOptions_2,this._XSRFStrategy_3)); }
    return this.__XHRBackend_4;
  }
  get _RequestOptions_5():import1.BaseRequestOptions {
    if ((this.__RequestOptions_5 == null)) { (this.__RequestOptions_5 = new import1.BaseRequestOptions()); }
    return this.__RequestOptions_5;
  }
  get _Http_6():any {
    if ((this.__Http_6 == null)) { (this.__Http_6 = import1.ɵc(this._XHRBackend_4,this._RequestOptions_5)); }
    return this.__Http_6;
  }
  createInternal():import1.HttpModule {
    this._HttpModule_0 = new import1.HttpModule();
    return this._HttpModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.HttpModule)) { return this._HttpModule_0; }
    if ((token === import1.BrowserXhr)) { return this._BrowserXhr_1; }
    if ((token === import1.ResponseOptions)) { return this._ResponseOptions_2; }
    if ((token === import1.XSRFStrategy)) { return this._XSRFStrategy_3; }
    if ((token === import1.XHRBackend)) { return this._XHRBackend_4; }
    if ((token === import1.RequestOptions)) { return this._RequestOptions_5; }
    if ((token === import1.Http)) { return this._Http_6; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const HttpModuleNgFactory:import0.NgModuleFactory<import1.HttpModule> = new import0.NgModuleFactory<any>(HttpModuleInjector,import1.HttpModule);
class JsonpModuleInjector extends import0.ɵNgModuleInjector<import1.JsonpModule> {
  _JsonpModule_0:import1.JsonpModule;
  __ɵg_1:import1.ɵg;
  __ResponseOptions_2:import1.BaseResponseOptions;
  __JSONPBackend_3:import1.ɵa;
  __RequestOptions_4:import1.BaseRequestOptions;
  __Jsonp_5:any;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _ɵg_1():import1.ɵg {
    if ((this.__ɵg_1 == null)) { (this.__ɵg_1 = new import1.ɵg()); }
    return this.__ɵg_1;
  }
  get _ResponseOptions_2():import1.BaseResponseOptions {
    if ((this.__ResponseOptions_2 == null)) { (this.__ResponseOptions_2 = new import1.BaseResponseOptions()); }
    return this.__ResponseOptions_2;
  }
  get _JSONPBackend_3():import1.ɵa {
    if ((this.__JSONPBackend_3 == null)) { (this.__JSONPBackend_3 = new import1.ɵa(this._ɵg_1,this._ResponseOptions_2)); }
    return this.__JSONPBackend_3;
  }
  get _RequestOptions_4():import1.BaseRequestOptions {
    if ((this.__RequestOptions_4 == null)) { (this.__RequestOptions_4 = new import1.BaseRequestOptions()); }
    return this.__RequestOptions_4;
  }
  get _Jsonp_5():any {
    if ((this.__Jsonp_5 == null)) { (this.__Jsonp_5 = import1.ɵd(this._JSONPBackend_3,this._RequestOptions_4)); }
    return this.__Jsonp_5;
  }
  createInternal():import1.JsonpModule {
    this._JsonpModule_0 = new import1.JsonpModule();
    return this._JsonpModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.JsonpModule)) { return this._JsonpModule_0; }
    if ((token === import1.ɵg)) { return this._ɵg_1; }
    if ((token === import1.ResponseOptions)) { return this._ResponseOptions_2; }
    if ((token === import1.JSONPBackend)) { return this._JSONPBackend_3; }
    if ((token === import1.RequestOptions)) { return this._RequestOptions_4; }
    if ((token === import1.Jsonp)) { return this._Jsonp_5; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const JsonpModuleNgFactory:import0.NgModuleFactory<import1.JsonpModule> = new import0.NgModuleFactory<any>(JsonpModuleInjector,import1.JsonpModule);
