// @progress/kendo-angular-inputs/dist/es/numerictextbox/numerictextbox.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-inputs/dist/es/numerictextbox/localization/localized-messages.directive';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-inputs/dist/es/numerictextbox/localization/messages';
import * as import4 from '@progress/kendo-angular-inputs/dist/es/numerictextbox/numerictextbox.component';
import * as import5 from '@angular/common';
import * as import6 from '@progress/kendo-angular-l10n/dist/es/message.service';
import * as import7 from '@progress/kendo-angular-intl/dist/es/intl.service';
import * as import8 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import9 from '@angular/forms';
const styles_NumericTextBoxComponent:any[] = ([] as any[]);
export const RenderType_NumericTextBoxComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NumericTextBoxComponent,
  data: {}
}
);
function View_NumericTextBoxComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'span',[[
        'class',
        'k-select'
      ]
      ],(null as any),[[
        (null as any),
        'mouseup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mouseup' === en)) {
        const pd_0:any = ((<any>co.releaseArrow()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'k-link k-link-increase'
      ]
    ],[
      [
        1,
        'aria-label',
        0
      ]
      ,
      [
        8,
        'title',
        0
      ]
      ,
      [
        2,
        'k-state-active',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.arrowPress(co.ArrowDirection.Up,$event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_1:any = ((<any>co.releaseArrow()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon k-i-arrow-n'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'k-link k-link-decrease'
      ]
    ],[
      [
        1,
        'aria-label',
        0
      ]
      ,
      [
        8,
        'title',
        0
      ]
      ,
      [
        2,
        'k-state-active',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('mousedown' === en)) {
        const pd_0:any = ((<any>co.arrowPress(co.ArrowDirection.Down,$event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_1:any = ((<any>co.releaseArrow()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon k-i-arrow-s'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.incrementTitle;
    const currVal_1:any = co.incrementTitle;
    const currVal_2:any = (co.arrowDirection === co.ArrowDirection.Up);
    ck(v,2,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = co.decrementTitle;
    const currVal_4:any = co.decrementTitle;
    const currVal_5:any = (co.arrowDirection === co.ArrowDirection.Down);
    ck(v,7,0,currVal_3,currVal_4,currVal_5);
  });
}
export function View_NumericTextBoxComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{numericInput: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(270336,(null as any),0,import1.LocalizedMessagesDirective,[import2.LocalizationService],{
      decrement: [
        0,
        'decrement'
      ]
      ,
      increment: [
        1,
        'increment'
      ]

    }
    ,(null as any)),
    import0.ɵprd(3072,(null as any),import3.Messages,(null as any),[import1.LocalizedMessagesDirective]),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'span',[[
        'class',
        'k-numeric-wrap'
      ]
    ],[
      [
        2,
        'k-state-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'numericInput',
        1
      ]

    ]
    ,(null as any),0,'input',[
      [
        'class',
        'k-input'
      ]
      ,
      [
        'role',
        'spinbutton'
      ]

    ]
    ,[
      [
        1,
        'aria-valuemin',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'title',
        0
      ]
      ,
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]
      ,
      [
        2,
        'k-formatted-value',
        (null as any)
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import4.NumericTextBoxComponent = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.handleInput()) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.handleFocus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.handleBlur()) !== false);
        ad = (pd_2 && ad);
      }
      if (('keydown' === en)) {
        const pd_3:any = ((<any>co.handleKeyDown($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NumericTextBoxComponent_1)),
    import0.ɵdid(8192,(null as any),0,import5.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:import4.NumericTextBoxComponent = v.component;
    const currVal_0:any = 'Decrease value';
    const currVal_1:any = 'Increase value';
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_12:any = co.spinners;
    ck(v,12,0,currVal_12);
  },(ck,v) => {
    var co:import4.NumericTextBoxComponent = v.component;
    const currVal_2:any = co.disabled;
    const currVal_3:any = co.focused;
    ck(v,7,0,currVal_2,currVal_3);
    const currVal_4:any = co.min;
    const currVal_5:any = co.max;
    const currVal_6:any = co.value;
    const currVal_7:any = co.title;
    const currVal_8:any = co.placeholder;
    const currVal_9:any = co.tabIndex;
    const currVal_10:boolean = !co.focused;
    const currVal_11:any = co.disabled;
    ck(v,9,0,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11);
  });
}
function View_NumericTextBoxComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'kendo-numerictextbox',([] as any[]),[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-numerictextbox',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_NumericTextBoxComponent_0,RenderType_NumericTextBoxComponent)),
    import0.ɵprd(128,(null as any),import2.L10N_PREFIX,'kendo.numerictextbox',([] as any[])),
    import0.ɵprd(256,(null as any),import2.LocalizationService,import2.LocalizationService,[
      import2.L10N_PREFIX,
      [
        2,
        import6.MessageService
      ]

    ]
    ),
    import0.ɵdid(352256,(null as any),0,import4.NumericTextBoxComponent,[
      import7.IntlService,
      import0.Renderer,
      import2.LocalizationService,
      [
        2,
        import8.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import9.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import4.NumericTextBoxComponent]),
    import0.ɵprd(2560,(null as any),import9.NG_VALIDATORS,(p0_0:any) => {
      return [p0_0];
    },[import4.NumericTextBoxComponent])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).direction;
    const currVal_1:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_2:any = import0.ɵnov(v,3).widgetClasses;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const NumericTextBoxComponentNgFactory:import0.ComponentFactory<import4.NumericTextBoxComponent> = import0.ɵccf('kendo-numerictextbox',import4.NumericTextBoxComponent,View_NumericTextBoxComponent_Host_0,{
  disabled: 'disabled',
  title: 'title',
  autoCorrect: 'autoCorrect',
  format: 'format',
  max: 'max',
  min: 'min',
  decimals: 'decimals',
  placeholder: 'placeholder',
  step: 'step',
  spinners: 'spinners',
  rangeValidation: 'rangeValidation',
  tabIndex: 'tabIndex',
  value: 'value'
}
,{
  valueChange: 'valueChange',
  onFocus: 'focus',
  onBlur: 'blur'
}
,([] as any[]));
