// @progress/kendo-angular-grid/dist/es/localization/custom-messages.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/localization/custom-messages.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/localization/messages';
const styles_CustomMessagesComponent:any[] = ([] as any[]);
export const RenderType_CustomMessagesComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_CustomMessagesComponent,
  data: {}
}
);
export function View_CustomMessagesComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_CustomMessagesComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-messages',([] as any[]),(null as any),(null as any),(null as any),View_CustomMessagesComponent_0,RenderType_CustomMessagesComponent)),
    import0.ɵdid(286720,(null as any),0,import1.CustomMessagesComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.Messages,(null as any),[import1.CustomMessagesComponent])
  ]
  ,(null as any),(null as any));
}
export const CustomMessagesComponentNgFactory:import0.ComponentFactory<import1.CustomMessagesComponent> = import0.ɵccf('kendo-grid-messages',import1.CustomMessagesComponent,View_CustomMessagesComponent_Host_0,{
  groupPanelEmpty: 'groupPanelEmpty',
  noRecords: 'noRecords',
  pagerFirstPage: 'pagerFirstPage',
  pagerLastPage: 'pagerLastPage',
  pagerPreviousPage: 'pagerPreviousPage',
  pagerNextPage: 'pagerNextPage',
  pagerPage: 'pagerPage',
  pagerItemsPerPage: 'pagerItemsPerPage',
  pagerOf: 'pagerOf',
  pagerItems: 'pagerItems',
  filterEqOperator: 'filterEqOperator',
  filterNotEqOperator: 'filterNotEqOperator',
  filterIsNullOperator: 'filterIsNullOperator',
  filterIsNotNullOperator: 'filterIsNotNullOperator',
  filterIsEmptyOperator: 'filterIsEmptyOperator',
  filterIsNotEmptyOperator: 'filterIsNotEmptyOperator',
  filterStartsWithOperator: 'filterStartsWithOperator',
  filterContainsOperator: 'filterContainsOperator',
  filterNotContainsOperator: 'filterNotContainsOperator',
  filterEndsWithOperator: 'filterEndsWithOperator',
  filterGteOperator: 'filterGteOperator',
  filterGtOperator: 'filterGtOperator',
  filterLteOperator: 'filterLteOperator',
  filterLtOperator: 'filterLtOperator',
  filterIsTrue: 'filterIsTrue',
  filterIsFalse: 'filterIsFalse',
  filterBooleanAll: 'filterBooleanAll'
}
,{},([] as any[]));
