// @progress/kendo-angular-buttons/dist/es/buttongroup/buttongroup.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-buttons/dist/es/button/button.service';
import * as import2 from '@progress/kendo-angular-buttons/dist/es/buttongroup/buttongroup.component';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/rtl';
const styles_ButtonGroupComponent:any[] = ([] as any[]);
export const RenderType_ButtonGroupComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ButtonGroupComponent,
  data: {}
}
);
export function View_ButtonGroupComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(null as any));
}
function View_ButtonGroupComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'kendo-buttongroup',([] as any[]),[
      [
        2,
        'k-button-group',
        (null as any)
      ]
      ,
      [
        2,
        'k-button-group-stretched',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]
      ,
      [
        4,
        'width',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'keydown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,2).keydown($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_ButtonGroupComponent_0,RenderType_ButtonGroupComponent)),
    import0.ɵprd(256,(null as any),import1.KendoButtonService,import1.KendoButtonService,([] as any[])),
    import0.ɵdid(5890048,(null as any),1,import2.ButtonGroupComponent,[
      import1.KendoButtonService,
      [
        2,
        import3.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{buttons: 1})
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).wrapperClass;
    const currVal_1:any = import0.ɵnov(v,2).stretchedClass;
    const currVal_2:any = import0.ɵnov(v,2).getRole;
    const currVal_3:any = import0.ɵnov(v,2).dir;
    const currVal_4:any = import0.ɵnov(v,2).wrapperWidth;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export const ButtonGroupComponentNgFactory:import0.ComponentFactory<import2.ButtonGroupComponent> = import0.ɵccf('kendo-buttongroup',import2.ButtonGroupComponent,View_ButtonGroupComponent_Host_0,{
  disabled: 'disabled',
  selection: 'selection',
  width: 'width'
}
,{},['[kendoButton]']);
