// @progress/kendo-angular-inputs/dist/es/slider/slider-ticks.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-inputs/dist/es/slider/slider-ticks.component';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/rtl';
const styles_KendoSliderTicksComponent:any[] = ([] as any[]);
export const RenderType_KendoSliderTicksComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_KendoSliderTicksComponent,
  data: {}
}
);
function View_KendoSliderTicksComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'tickElement',
        1
      ]

    ]
      ,(null as any),2,'li',[[
        'role',
        'presentation'
      ]
      ],[[
        8,
        'title',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.tickClasses(v.context.$implicit);
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',co.title((v.context.$implicit.index * co.step)),'');
    ck(v,0,0,currVal_0);
  });
}
export function View_KendoSliderTicksComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{container: 0}),
    import0.ɵqud(335544320,2,{tickElements: 1}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'container',
        1
      ]

    ]
      ,(null as any),4,'ul',[[
        'class',
        'k-reset k-slider-items'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_KendoSliderTicksComponent_1)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.KendoSliderTicksComponent = v.component;
    const currVal_0:any = co.ticks;
    ck(v,6,0,currVal_0);
  },(null as any));
}
function View_KendoSliderTicksComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-slider-ticks',([] as any[]),(null as any),(null as any),(null as any),View_KendoSliderTicksComponent_0,RenderType_KendoSliderTicksComponent)),
      import0.ɵdid(286720,(null as any),0,import2.KendoSliderTicksComponent,[[
        2,
        import3.RTL
      ]
    ],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const KendoSliderTicksComponentNgFactory:import0.ComponentFactory<import2.KendoSliderTicksComponent> = import0.ɵccf('kendo-slider-ticks',import2.KendoSliderTicksComponent,View_KendoSliderTicksComponent_Host_0,{
  vertical: 'vertical',
  ticksCount: 'ticksCount',
  title: 'title',
  step: 'step'
}
,{tickClick: 'tickClick'},([] as any[]));
