// @progress/kendo-angular-inputs/dist/es/slider/slider.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './slider-ticks.component.ngfactory';
import * as import2 from '@progress/kendo-angular-inputs/dist/es/slider/slider-ticks.component';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import4 from '@angular/common';
import * as import5 from '@progress/kendo-angular-inputs/dist/es/slider/slider.component';
import * as import6 from '@progress/kendo-angular-inputs/dist/es/common/draggable';
import * as import7 from '../../../../kendo-angular-resize-sensor/dist/es/resize-sensor.component.ngfactory';
import * as import8 from '@progress/kendo-angular-resize-sensor/dist/es/resize-sensor.component';
import * as import9 from '@angular/forms';
const styles_SliderComponent:any[] = ([] as any[]);
export const RenderType_SliderComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SliderComponent,
  data: {}
}
);
function View_SliderComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        5,
        0
      ]
      ,
      [
        'decreaseButton',
        1
      ]

    ]
      ,(null as any),4,'a',[[
        'class',
        'k-button k-button-decrease'
      ]
    ],[
      [
        8,
        'title',
        0
      ]
      ,
      [
        1,
        'aria-label',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),[
      [
        2,
        'k-icon',
        (null as any)
      ]
      ,
      [
        2,
        'k-i-arrow-w',
        (null as any)
      ]
      ,
      [
        2,
        'k-i-arrow-s',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.decrementTitle;
    const currVal_1:any = co.decrementTitle;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = true;
    const currVal_3:boolean = !co.vertical;
    const currVal_4:any = co.vertical;
    ck(v,2,0,currVal_2,currVal_3,currVal_4);
  });
}
function View_SliderComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        6,
        0
      ]
      ,
      [
        'increaseButton',
        1
      ]

    ]
      ,(null as any),4,'a',[[
        'class',
        'k-button k-button-increase'
      ]
    ],[
      [
        8,
        'title',
        0
      ]
      ,
      [
        1,
        'aria-label',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),[
      [
        2,
        'k-icon',
        (null as any)
      ]
      ,
      [
        2,
        'k-i-arrow-e',
        (null as any)
      ]
      ,
      [
        2,
        'k-i-arrow-n',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.incrementTitle;
    const currVal_1:any = co.incrementTitle;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = true;
    const currVal_3:boolean = !co.vertical;
    const currVal_4:any = co.vertical;
    ck(v,2,0,currVal_2,currVal_3,currVal_4);
  });
}
function View_SliderComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-slider-ticks',([] as any[]),(null as any),[[
        (null as any),
        'tickClick'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('tickClick' === en)) {
        const pd_0:any = ((<any>co.ifEnabled(co.onTickClick,$event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import1.View_KendoSliderTicksComponent_0,import1.RenderType_KendoSliderTicksComponent)),
    import0.ɵdid(286720,[
      [
        4,
        4
      ]
      ,
      [
        'ticks',
        4
      ]

    ]
      ,0,import2.KendoSliderTicksComponent,[[
        2,
        import3.RTL
      ]
    ],{
      vertical: [
        0,
        'vertical'
      ]
      ,
      ticksCount: [
        1,
        'ticksCount'
      ]
      ,
      title: [
        2,
        'title'
      ]
      ,
      step: [
        3,
        'step'
      ]

    }
    ,{tickClick: 'tickClick'}),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.vertical;
    const currVal_1:any = co.ticksCount;
    const currVal_2:any = co.title;
    const currVal_3:any = co.smallStep;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
  },(null as any));
}
export function View_SliderComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{track: 0}),
    import0.ɵqud(201326592,2,{draghandle: 0}),
    import0.ɵqud(201326592,3,{sliderSelection: 0}),
    import0.ɵqud(335544320,4,{ticks: 0}),
    import0.ɵqud(335544320,5,{decreaseButton: 0}),
    import0.ɵqud(335544320,6,{increaseButton: 0}),
    (l()(),import0.ɵted((null as any),['\n\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),25,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import4.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-slider-wrap',
      'k-slider-buttons'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SliderComponent_1)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SliderComponent_2)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_SliderComponent_3)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'track',
        1
      ]

    ]
      ,(null as any),8,'div',[[
        'class',
        'k-slider-track'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import5.SliderComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.ifEnabled(co.onTrackClick,$event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'sliderSelection',
        1
      ]

    ]
    ,(null as any),1,'div',([] as any[]),[
      [
        2,
        'k-slider-selection',
        (null as any)
      ]
      ,
      [
        2,
        'k-pressed',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'draghandle',
        1
      ]

    ]
      ,(null as any),2,'a',[[
        'kendoDraggable',
        ''
      ]
    ],[
      [
        2,
        'k-draghandle',
        (null as any)
      ]
      ,
      [
        2,
        'k-pressed',
        (null as any)
      ]
      ,
      [
        8,
        'title',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'kendoDrag'
      ]
      ,
      [
        (null as any),
        'kendoRelease'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import5.SliderComponent = v.component;
      if (('kendoDrag' === en)) {
        const pd_0:any = ((<any>co.ifEnabled(co.onHandleDrag,$event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('kendoRelease' === en)) {
        const pd_1:any = ((<any>co.ifEnabled(co.onHandleRelease,$event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import6.KendoDraggableDirective,[import0.ElementRef],(null as any),{
      kendoDrag: 'kendoDrag',
      kendoRelease: 'kendoRelease'
    }
    ),
    (l()(),import0.ɵted((null as any),['Drag'])),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-resize-sensor',([] as any[]),(null as any),[[
        (null as any),
        'resize'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import5.SliderComponent = v.component;
      if (('resize' === en)) {
        const pd_0:any = ((<any>co.sizeComponent(false)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import7.View_ResizeSensorComponent_0,import7.RenderType_ResizeSensorComponent)),
    import0.ɵdid(4284416,(null as any),0,import8.ResizeSensorComponent,[import0.ElementRef],(null as any),{resize: 'resize'}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n\n  ']))
  ]
  ,(ck,v) => {
    var co:import5.SliderComponent = v.component;
    const currVal_0:any = ck(v,9,0,true,co.showButtons);
    ck(v,8,0,currVal_0);
    const currVal_1:any = co.showButtons;
    ck(v,12,0,currVal_1);
    const currVal_2:any = co.showButtons;
    ck(v,15,0,currVal_2);
    const currVal_3:any = (co.tickPlacement !== 'none');
    ck(v,18,0,currVal_3);
  },(ck,v) => {
    var co:import5.SliderComponent = v.component;
    const currVal_4:any = true;
    const currVal_5:any = co.dragging;
    ck(v,22,0,currVal_4,currVal_5);
    const currVal_6:any = true;
    const currVal_7:any = co.dragging;
    const currVal_8:any = co.dragHandleTitle;
    ck(v,25,0,currVal_6,currVal_7,currVal_8);
  });
}
function View_SliderComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-slider',([] as any[]),[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        2,
        'k-slider-horizontal',
        (null as any)
      ]
      ,
      [
        2,
        'k-slider-vertical',
        (null as any)
      ]
      ,
      [
        2,
        'k-slider',
        (null as any)
      ]
      ,
      [
        2,
        'k-slider-transitions',
        (null as any)
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-default',
        (null as any)
      ]
      ,
      [
        2,
        'k-slider-topleft',
        (null as any)
      ]
      ,
      [
        2,
        'k-slider-bottomright',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-disabled',
        (null as any)
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-valuemin',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).active()) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).active()) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,1).keydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('blur' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,1).blur()) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },View_SliderComponent_0,RenderType_SliderComponent)),
    import0.ɵdid(2449408,(null as any),0,import5.SliderComponent,[
      import0.ElementRef,
      [
        2,
        import3.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import9.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import5.SliderComponent])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).direction;
    const currVal_1:any = import0.ɵnov(v,1).horizontalClass;
    const currVal_2:any = import0.ɵnov(v,1).verticalClass;
    const currVal_3:any = import0.ɵnov(v,1).sliderClass;
    const currVal_4:any = import0.ɵnov(v,1).transitionsClass;
    const currVal_5:any = import0.ɵnov(v,1).widgetClass;
    const currVal_6:any = import0.ɵnov(v,1).stateDefaultClass;
    const currVal_7:any = import0.ɵnov(v,1).topLeftClass;
    const currVal_8:any = import0.ɵnov(v,1).bottomRightClass;
    const currVal_9:any = import0.ɵnov(v,1).disabledClass;
    const currVal_10:any = import0.ɵnov(v,1).tabIndex;
    const currVal_11:any = import0.ɵnov(v,1).ariaDisabled;
    const currVal_12:any = import0.ɵnov(v,1).ariaMin;
    const currVal_13:any = import0.ɵnov(v,1).ariaMax;
    const currVal_14:any = import0.ɵnov(v,1).ariaValue;
    ck(v,0,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14
    ]
    );
  });
}
export const SliderComponentNgFactory:import0.ComponentFactory<import5.SliderComponent> = import0.ɵccf('kendo-slider',import5.SliderComponent,View_SliderComponent_Host_0,{
  incrementTitle: 'incrementTitle',
  decrementTitle: 'decrementTitle',
  showButtons: 'showButtons',
  tickPlacement: 'tickPlacement',
  title: 'title',
  dragHandleTitle: 'dragHandleTitle',
  vertical: 'vertical',
  min: 'min',
  max: 'max',
  smallStep: 'smallStep',
  fixedTickWidth: 'fixedTickWidth',
  disabled: 'disabled',
  value: 'value'
}
,{valueChange: 'valueChange'},([] as any[]));
