// @progress/kendo-angular-dialog/dist/es/dialog.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './dialog-titlebar.component.ngfactory';
import * as import2 from '@progress/kendo-angular-dialog/dist/es/dialog-titlebar.component';
import * as import3 from '@angular/common';
import * as import4 from './dialog-actions.component.ngfactory';
import * as import5 from '@progress/kendo-angular-dialog/dist/es/dialog-actions.component';
import * as import6 from '@progress/kendo-angular-dialog/dist/es/dialog.component';
import * as import7 from '@progress/kendo-angular-l10n/dist/es/rtl';
const styles_DialogComponent:any[] = ([] as any[]);
export const RenderType_DialogComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DialogComponent,
    data: {'animation': [
      {
        name: 'overlayAppear',
        definitions: [
          {
            type: 0,
            name: 'in',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'void => *',
            animation: [
              {
                type: 6,
                styles: {opacity: 0.1}
              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '.3s cubic-bezier(.2, .6, .4, 1)'
              }

            ]

          }

        ]

      }
      ,
      {
        name: 'dialogSlideInAppear',
        definitions: [
          {
            type: 0,
            name: 'in',
            styles: {
              type: 6,
              styles: {transform: 'translate(-50%, -50%)'}
            }

          }
          ,
          {
            type: 1,
            expr: 'void => *',
            animation: [
              {
                type: 6,
                styles: {transform: 'translate(-50%, -60%)'}
              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '.3s cubic-bezier(.2, 1, .2, 1)'
              }

            ]

          }

        ]

      }

    ]
  }
}
);
function View_DialogComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-dialog-titlebar',([] as any[]),[
      [
        2,
        'k-window-titlebar',
        (null as any)
      ]
      ,
      [
        2,
        'k-dialog-titlebar',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_DialogTitleBarComponent_0,import1.RenderType_DialogTitleBarComponent)),
      import0.ɵdid(24576,[[
        1,
        4
      ]
    ],0,import2.DialogTitleBarComponent,([] as any[]),(null as any),(null as any)),
    (l()(),import0.ɵted(0,[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵnov(v,1).className;
    const currVal_1:any = import0.ɵnov(v,1).className;
    const currVal_2:any = import0.ɵnov(v,1).className;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = co.title;
    ck(v,2,0,currVal_3);
  });
}
function View_DialogComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(null as any),(null as any));
}
function View_DialogComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(null as any),(null as any));
}
function View_DialogComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_DialogComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_5)),
      import0.ɵdid(270336,(null as any),0,import3.NgTemplateOutlet,[import0.ViewContainerRef],{ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
    },(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.contentTemplate;
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_DialogComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),2),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(null as any),(null as any));
}
function View_DialogComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-dialog-actions',([] as any[]),[
      [
        2,
        'k-button-group',
        (null as any)
      ]
      ,
      [
        2,
        'k-dialog-buttongroup',
        (null as any)
      ]
      ,
      [
        2,
        'k-dialog-button-layout-stretched',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import4.View_DialogActionsComponent_0,import4.RenderType_DialogActionsComponent)),
      import0.ɵdid(24576,[[
        2,
        4
      ]
      ],0,import5.DialogActionsComponent,([] as any[]),{actions: [
        0,
        'actions'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.actions;
    ck(v,1,0,currVal_3);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).className;
    const currVal_1:any = import0.ɵnov(v,1).className;
    const currVal_2:any = import0.ɵnov(v,1).className;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_DialogComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{titlebarView: 0}),
    import0.ɵqud(335544320,2,{actionsView: 0}),
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'k-overlay'
      ]
      ],[[
        24,
        '@overlayAppear',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),23,'div',[[
        'class',
        'k-widget k-window k-dialog k-centered k-dialog-centered'
      ]
      ],[[
        24,
        '@dialogSlideInAppear',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import3.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_1)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_2)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'k-content k-window-content k-dialog-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_3)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_4)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_6)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogComponent_7)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import6.DialogComponent = v.component;
    const currVal_2:any = co.styles;
    ck(v,6,0,currVal_2);
    const currVal_3:any = co.title;
    ck(v,9,0,currVal_3);
    const currVal_4:boolean = !co.title;
    ck(v,12,0,currVal_4);
    const currVal_5:boolean = !co.contentTemplate;
    ck(v,17,0,currVal_5);
    const currVal_6:any = co.contentTemplate;
    ck(v,20,0,currVal_6);
    const currVal_7:boolean = !co.actions;
    ck(v,24,0,currVal_7);
    const currVal_8:any = co.actions;
    ck(v,27,0,currVal_8);
  },(ck,v) => {
    const currVal_0:any = (null as any);
    ck(v,3,0,currVal_0);
    const currVal_1:any = (null as any);
    ck(v,5,0,currVal_1);
  });
}
function View_DialogComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-dialog',([] as any[]),[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        2,
        'k-dialog-wrapper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_DialogComponent_0,RenderType_DialogComponent)),
    import0.ɵdid(2678784,(null as any),1,import6.DialogComponent,[
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import7.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{titlebarContent: 0})
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).dir;
    const currVal_1:any = import0.ɵnov(v,1).wrapperClass;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const DialogComponentNgFactory:import0.ComponentFactory<import6.DialogComponent> = import0.ɵccf('kendo-dialog',import6.DialogComponent,View_DialogComponent_Host_0,{
  title: 'title',
  actions: 'actions',
  width: 'width',
  height: 'height'
}
,{
  action: 'action',
  close: 'close'
}
,[
  'kendo-dialog-titlebar',
  '*',
  'kendo-dialog-actions'
]
);
