// @progress/kendo-angular-dialog/dist/es/dialog-actions.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-dialog/dist/es/dialog-actions.component';
const styles_DialogActionsComponent:any[] = ([] as any[]);
export const RenderType_DialogActionsComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DialogActionsComponent,
  data: {}
}
);
function View_DialogActionsComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(null as any),(null as any));
}
function View_DialogActionsComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onButtonClick(v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '\n        ',
      '\n      '
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.buttonClass(v.context.$implicit);
    ck(v,1,0,currVal_0);
  },(ck,v) => {
    const currVal_1:any = v.context.$implicit.text;
    ck(v,2,0,currVal_1);
  });
}
function View_DialogActionsComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogActionsComponent_3)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.actions;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_DialogActionsComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_DialogActionsComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogActionsComponent_5)),
      import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
    },(null as any)),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.actions;
    ck(v,1,0,currVal_0);
  },(null as any));
}
export function View_DialogActionsComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogActionsComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogActionsComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DialogActionsComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.DialogActionsComponent = v.component;
    const currVal_0:boolean = !co.actions;
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !co.actionTemplate();
    ck(v,5,0,currVal_1);
    const currVal_2:any = co.actionTemplate();
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_DialogActionsComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-dialog-actions',([] as any[]),[
      [
        2,
        'k-button-group',
        (null as any)
      ]
      ,
      [
        2,
        'k-dialog-buttongroup',
        (null as any)
      ]
      ,
      [
        2,
        'k-dialog-button-layout-stretched',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_DialogActionsComponent_0,RenderType_DialogActionsComponent)),
    import0.ɵdid(24576,(null as any),0,import2.DialogActionsComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).className;
    const currVal_1:any = import0.ɵnov(v,1).className;
    const currVal_2:any = import0.ɵnov(v,1).className;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const DialogActionsComponentNgFactory:import0.ComponentFactory<import2.DialogActionsComponent> = import0.ɵccf('kendo-dialog-actions',import2.DialogActionsComponent,View_DialogActionsComponent_Host_0,{actions: 'actions'},{action: 'action'},['*']);
