// @progress/kendo-angular-dialog/dist/es/dialog-titlebar.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-dialog/dist/es/dialog-titlebar.component';
const styles_DialogTitleBarComponent:any[] = ([] as any[]);
export const RenderType_DialogTitleBarComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DialogTitleBarComponent,
  data: {}
}
);
export function View_DialogTitleBarComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'k-window-title k-dialog-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'k-window-actions k-dialog-actions'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'aria-label',
        'Close'
      ]
      ,
      [
        'class',
        'k-button k-bare k-button-icon k-window-action k-dialog-action k-dialog-close'
      ]
      ,
      [
        'href',
        '#'
      ]
      ,
      [
        'role',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.DialogTitleBarComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onCloseClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon k-i-x'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(null as any),(null as any));
}
function View_DialogTitleBarComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-dialog-titlebar',([] as any[]),[
      [
        2,
        'k-window-titlebar',
        (null as any)
      ]
      ,
      [
        2,
        'k-dialog-titlebar',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_DialogTitleBarComponent_0,RenderType_DialogTitleBarComponent)),
    import0.ɵdid(24576,(null as any),0,import1.DialogTitleBarComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).className;
    const currVal_1:any = import0.ɵnov(v,1).className;
    const currVal_2:any = import0.ɵnov(v,1).className;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const DialogTitleBarComponentNgFactory:import0.ComponentFactory<import1.DialogTitleBarComponent> = import0.ɵccf('kendo-dialog-titlebar',import1.DialogTitleBarComponent,View_DialogTitleBarComponent_Host_0,{},{close: 'close'},['*']);
