// @progress/kendo-angular-inputs/dist/es/numerictextbox/localization/custom-messages.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-inputs/dist/es/numerictextbox/localization/custom-messages.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-inputs/dist/es/numerictextbox/localization/messages';
const styles_NumericTextBoxCustomMessagesComponent:any[] = ([] as any[]);
export const RenderType_NumericTextBoxCustomMessagesComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NumericTextBoxCustomMessagesComponent,
  data: {}
}
);
export function View_NumericTextBoxCustomMessagesComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_NumericTextBoxCustomMessagesComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-numerictextbox-messages',([] as any[]),(null as any),(null as any),(null as any),View_NumericTextBoxCustomMessagesComponent_0,RenderType_NumericTextBoxCustomMessagesComponent)),
    import0.ɵdid(286720,(null as any),0,import1.NumericTextBoxCustomMessagesComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.Messages,(null as any),[import1.NumericTextBoxCustomMessagesComponent])
  ]
  ,(null as any),(null as any));
}
export const NumericTextBoxCustomMessagesComponentNgFactory:import0.ComponentFactory<import1.NumericTextBoxCustomMessagesComponent> = import0.ɵccf('kendo-numerictextbox-messages',import1.NumericTextBoxCustomMessagesComponent,View_NumericTextBoxCustomMessagesComponent_Host_0,{
  decrement: 'decrement',
  increment: 'increment'
}
,{},([] as any[]));
