// @progress/kendo-angular-grid/dist/es/filtering/autocomplete-filter-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './filter-cell-wrapper.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-wrapper.component';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/filter.service';
import * as import4 from '../../../../kendo-angular-dropdowns/dist/es/autocomplete.component.ngfactory';
import * as import5 from '@progress/kendo-angular-dropdowns/dist/es/selection.service';
import * as import6 from '@progress/kendo-angular-dropdowns/dist/es/navigation.service';
import * as import7 from '@progress/kendo-angular-dropdowns/dist/es/autocomplete.component';
import * as import8 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import9 from '@angular/forms';
import * as import10 from '@progress/kendo-angular-grid/dist/es/filtering/filter-input.directive';
import * as import11 from '@progress/kendo-angular-grid/dist/es/filtering/autocomplete-filter-cell.component';
import * as import12 from '@progress/kendo-angular-grid/dist/es/column.component';
import * as import13 from '@progress/kendo-angular-l10n/dist/es/localization.service';
const styles_AutoCompleteFilterCellComponent:any[] = ([] as any[]);
export const RenderType_AutoCompleteFilterCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_AutoCompleteFilterCellComponent,
  data: {}
}
);
export function View_AutoCompleteFilterCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),16,'kendo-grid-filter-wrapper-cell',([] as any[]),[
      [
        2,
        'k-filtercell',
        (null as any)
      ]
      ,
      [
        2,
        'k-filtercell-wrapper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_FilterCellWrapperComponent_0,import1.RenderType_FilterCellWrapperComponent)),
    import0.ɵdid(548864,(null as any),2,import2.FilterCellWrapperComponent,[import3.FilterService],{
      showOperators: [
        0,
        'showOperators'
      ]
      ,
      operators: [
        1,
        'operators'
      ]
      ,
      column: [
        2,
        'column'
      ]
      ,
      filter: [
        3,
        'filter'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1}),
    import0.ɵqud(167772160,2,{input: 0}),
    (l()(),import0.ɵted(0,['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),0,10,'kendo-autocomplete',[[
        'kendoFilterInput',
        ''
      ]
    ],[
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-autocomplete',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-disabled',
        (null as any)
      ]
      ,
      [
        4,
        'max-height',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-haspopup',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        1,
        'aria-activedescendant',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,(null as any),(null as any),import4.View_AutoCompleteComponent_0,import4.RenderType_AutoCompleteComponent)),
    import0.ɵprd(256,(null as any),import5.SelectionService,import5.SelectionService,([] as any[])),
    import0.ɵprd(256,(null as any),import6.NavigationService,import6.NavigationService,([] as any[])),
    import0.ɵdid(352256,(null as any),4,import7.AutoCompleteComponent,[
      [
        2,
        import8.RTL
      ]
      ,
      import5.SelectionService,
      import6.NavigationService,
      import0.ElementRef
    ]
    ,{
      data: [
        0,
        'data'
      ]
      ,
      value: [
        1,
        'value'
      ]
      ,
      valueField: [
        2,
        'valueField'
      ]

    }
    ,(null as any)),
    import0.ɵqud(167772160,3,{template: 0}),
    import0.ɵqud(167772160,4,{headerTemplate: 0}),
    import0.ɵqud(167772160,5,{footerTemplate: 0}),
    import0.ɵqud(167772160,6,{noDataTemplate: 0}),
    import0.ɵprd(512,(null as any),import9.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import7.AutoCompleteComponent]),
      import0.ɵdid(40960,[[
        2,
        4
      ]
      ],0,import10.FilterInputDirective,[import9.NG_VALUE_ACCESSOR],{value: [
        0,
        'value'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import11.AutoCompleteFilterCellComponent = v.component;
    const currVal_2:any = co.showOperators;
    const currVal_3:any = co.operators;
    const currVal_4:any = co.column;
    const currVal_5:any = co.filter;
    ck(v,2,0,currVal_2,currVal_3,currVal_4,currVal_5);
    const currVal_20:any = co.data;
    const currVal_21:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    const currVal_22:any = co.valueField;
    ck(v,9,0,currVal_20,currVal_21,currVal_22);
    const currVal_23:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    ck(v,15,0,currVal_23);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).overrideBaseClasses;
    const currVal_1:any = import0.ɵnov(v,2).hostClasses;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_6:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_7:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_8:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_9:any = import0.ɵnov(v,9).isFocused;
    const currVal_10:any = import0.ɵnov(v,9).isDisabled;
    const currVal_11:any = import0.ɵnov(v,9).widgetHeight;
    const currVal_12:any = import0.ɵnov(v,9).role;
    const currVal_13:any = import0.ɵnov(v,9).widgetTabIndex;
    const currVal_14:any = import0.ɵnov(v,9).ariaDisabled;
    const currVal_15:any = import0.ɵnov(v,9).ariaHasPopup;
    const currVal_16:any = import0.ɵnov(v,9).ariaExpanded;
    const currVal_17:any = import0.ɵnov(v,9).ariaOwns;
    const currVal_18:any = import0.ɵnov(v,9).ariaActivedescendant;
    const currVal_19:any = import0.ɵnov(v,9).dir;
    ck(v,6,1,[
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16,
      currVal_17,
      currVal_18,
      currVal_19
    ]
    );
  });
}
function View_AutoCompleteFilterCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-autocomplete-filter-cell',([] as any[]),[[
        2,
        'k-filtercell',
        (null as any)
      ]
    ],(null as any),(null as any),View_AutoCompleteFilterCellComponent_0,RenderType_AutoCompleteFilterCellComponent)),
    import0.ɵdid(548864,(null as any),1,import11.AutoCompleteFilterCellComponent,[
      import3.FilterService,
      import12.ColumnComponent,
      import13.LocalizationService
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hostClasses;
    ck(v,0,0,currVal_0);
  });
}
export const AutoCompleteFilterCellComponentNgFactory:import0.ComponentFactory<import11.AutoCompleteFilterCellComponent> = import0.ɵccf('kendo-grid-autocomplete-filter-cell',import11.AutoCompleteFilterCellComponent,View_AutoCompleteFilterCellComponent_Host_0,{
  showOperators: 'showOperators',
  column: 'column',
  filter: 'filter',
  data: 'data',
  valueField: 'valueField'
}
,{},([] as any[]));
