// @progress/kendo-angular-grid/dist/es/filtering/boolean-filter-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './filter-cell-wrapper.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-wrapper.component';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/filter.service';
import * as import4 from '../../../../kendo-angular-dropdowns/dist/es/dropdownlist.component.ngfactory';
import * as import5 from '@progress/kendo-angular-dropdowns/dist/es/selection.service';
import * as import6 from '@progress/kendo-angular-dropdowns/dist/es/navigation.service';
import * as import7 from '@progress/kendo-angular-dropdowns/dist/es/dropdownlist.component';
import * as import8 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import9 from '@angular/forms';
import * as import10 from '@progress/kendo-angular-grid/dist/es/filtering/filter-input.directive';
import * as import11 from '@progress/kendo-angular-grid/dist/es/filtering/boolean-filter-cell.component';
import * as import12 from '@progress/kendo-angular-l10n/dist/es/localization.service';
const styles_BooleanFilterCellComponent:any[] = ([] as any[]);
export const RenderType_BooleanFilterCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_BooleanFilterCellComponent,
  data: {}
}
);
export function View_BooleanFilterCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),18,'kendo-grid-filter-wrapper-cell',([] as any[]),[
      [
        2,
        'k-filtercell',
        (null as any)
      ]
      ,
      [
        2,
        'k-filtercell-wrapper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_FilterCellWrapperComponent_0,import1.RenderType_FilterCellWrapperComponent)),
    import0.ɵdid(548864,(null as any),2,import2.FilterCellWrapperComponent,[import3.FilterService],{
      showOperators: [
        0,
        'showOperators'
      ]
      ,
      column: [
        1,
        'column'
      ]
      ,
      filter: [
        2,
        'filter'
      ]
      ,
      defaultOperator: [
        3,
        'defaultOperator'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1}),
    import0.ɵqud(167772160,2,{input: 0}),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,12,'kendo-dropdownlist',[
      [
        'kendoFilterInput',
        ''
      ]
      ,
      [
        'textField',
        'text'
      ]
      ,
      [
        'valueField',
        'value'
      ]

    ]
    ,[
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-dropdown',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-haspopup',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        1,
        'aria-activedescendant',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keypress'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('blur' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,9).blur()) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,9).focus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,9).keydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keypress' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,9).keypress($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('click' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,9).click()) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },import4.View_DropDownListComponent_0,import4.RenderType_DropDownListComponent)),
    import0.ɵprd(256,(null as any),import5.SelectionService,import5.SelectionService,([] as any[])),
    import0.ɵprd(256,(null as any),import6.NavigationService,import6.NavigationService,([] as any[])),
    import0.ɵdid(352256,(null as any),5,import7.DropDownListComponent,[
      [
        2,
        import8.RTL
      ]
      ,
      import5.SelectionService,
      import6.NavigationService,
      import0.ElementRef,
      import0.Renderer,
      import0.NgZone
    ]
    ,{
      data: [
        0,
        'data'
      ]
      ,
      value: [
        1,
        'value'
      ]
      ,
      textField: [
        2,
        'textField'
      ]
      ,
      valueField: [
        3,
        'valueField'
      ]
      ,
      popupSettings: [
        4,
        'popupSettings'
      ]
      ,
      defaultItem: [
        5,
        'defaultItem'
      ]
      ,
      valuePrimitive: [
        6,
        'valuePrimitive'
      ]

    }
    ,(null as any)),
    import0.ɵqud(167772160,3,{itemTemplate: 0}),
    import0.ɵqud(167772160,4,{valueTemplate: 0}),
    import0.ɵqud(167772160,5,{headerTemplate: 0}),
    import0.ɵqud(167772160,6,{footerTemplate: 0}),
    import0.ɵqud(167772160,7,{noDataTemplate: 0}),
    import0.ɵpod(['width']),
    import0.ɵprd(512,(null as any),import9.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import7.DropDownListComponent]),
      import0.ɵdid(40960,[[
        2,
        4
      ]
      ],0,import10.FilterInputDirective,[import9.NG_VALUE_ACCESSOR],{value: [
        0,
        'value'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import11.BooleanFilterCellComponent = v.component;
    const currVal_2:any = false;
    const currVal_3:any = co.column;
    const currVal_4:any = co.filter;
    const currVal_5:any = co.operator;
    ck(v,2,0,currVal_2,currVal_3,currVal_4,currVal_5);
    const currVal_17:any = co.items;
    const currVal_18:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    const currVal_19:any = 'text';
    const currVal_20:any = 'value';
    const currVal_21:any = ck(v,15,0,'auto');
    const currVal_22:any = co.defaultItem;
    const currVal_23:any = true;
    ck(v,9,0,currVal_17,currVal_18,currVal_19,currVal_20,currVal_21,currVal_22,currVal_23);
    const currVal_24:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    ck(v,17,0,currVal_24);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).overrideBaseClasses;
    const currVal_1:any = import0.ɵnov(v,2).hostClasses;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_6:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_7:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_8:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_9:any = import0.ɵnov(v,9).role;
    const currVal_10:any = import0.ɵnov(v,9).widgetTabIndex;
    const currVal_11:any = import0.ɵnov(v,9).ariaDisabled;
    const currVal_12:any = import0.ɵnov(v,9).ariaHasPopup;
    const currVal_13:any = import0.ɵnov(v,9).ariaExpanded;
    const currVal_14:any = import0.ɵnov(v,9).ariaOwns;
    const currVal_15:any = import0.ɵnov(v,9).ariaActivedescendant;
    const currVal_16:any = import0.ɵnov(v,9).dir;
    ck(v,6,1,[
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13,
      currVal_14,
      currVal_15,
      currVal_16
    ]
    );
  });
}
function View_BooleanFilterCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-boolean-filter-cell',([] as any[]),[
      [
        2,
        'k-filtercell',
        (null as any)
      ]
      ,
      [
        2,
        'k-filtercell-boolean',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_BooleanFilterCellComponent_0,RenderType_BooleanFilterCellComponent)),
    import0.ɵdid(548864,(null as any),1,import11.BooleanFilterCellComponent,[
      import3.FilterService,
      import12.LocalizationService
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hostClasses;
    const currVal_1:any = import0.ɵnov(v,1).hostClasses;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const BooleanFilterCellComponentNgFactory:import0.ComponentFactory<import11.BooleanFilterCellComponent> = import0.ɵccf('kendo-grid-boolean-filter-cell',import11.BooleanFilterCellComponent,View_BooleanFilterCellComponent_Host_0,{
  column: 'column',
  filter: 'filter'
}
,{},([] as any[]));
