// @progress/kendo-angular-grid/dist/es/filtering/filter-row.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './filter-cell.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell.component';
import * as import3 from '@angular/common';
import * as import4 from '@progress/kendo-angular-grid/dist/es/filtering/filter-row.component';
const styles_FilterRowComponent:any[] = ([] as any[]);
export const RenderType_FilterRowComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FilterRowComponent,
  data: {}
}
);
function View_FilterRowComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_FilterRowComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',([] as any[]),[[
        2,
        'k-hierarchy-cell',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_FilterRowComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'th',[[
        'kendoGridFilterCell',
        ''
      ]
    ],(null as any),(null as any),(null as any),import1.View_FilterCellComponent_0,import1.RenderType_FilterCellComponent)),
    import0.ɵdid(24576,(null as any),0,import2.FilterCellComponent,([] as any[]),{
      column: [
        0,
        'column'
      ]
      ,
      filter: [
        1,
        'filter'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = co.filter;
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
export function View_FilterRowComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterRowComponent_1)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterRowComponent_2)),
    import0.ɵdid(8192,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterRowComponent_3)),
    import0.ɵdid(401408,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' \n    ']))
  ]
  ,(ck,v) => {
    var co:import4.FilterRowComponent = v.component;
    const currVal_0:any = co.groups;
    ck(v,2,0,currVal_0);
    const currVal_1:any = ((co.detailTemplate == null)? (null as any): co.detailTemplate.templateRef);
    ck(v,5,0,currVal_1);
    const currVal_2:any = co.columns;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_FilterRowComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridFilterRow',
        ''
      ]
      ],[[
        2,
        'k-filter-row',
        (null as any)
      ]
    ],(null as any),(null as any),View_FilterRowComponent_0,RenderType_FilterRowComponent)),
    import0.ɵdid(24576,(null as any),0,import4.FilterRowComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).filterRowClass;
    ck(v,0,0,currVal_0);
  });
}
export const FilterRowComponentNgFactory:import0.ComponentFactory<import4.FilterRowComponent> = import0.ɵccf('[kendoGridFilterRow]',import4.FilterRowComponent,View_FilterRowComponent_Host_0,{
  columns: 'columns',
  filter: 'filter',
  groups: 'groups',
  detailTemplate: 'detailTemplate'
}
,{},([] as any[]));
