// @progress/kendo-angular-grid/dist/es/filtering/filter-cell-wrapper.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-wrapper.component';
import * as import2 from './filter-cell-operators.component.ngfactory';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-operators.component';
import * as import4 from '@progress/kendo-angular-grid/dist/es/filtering/filter.service';
const styles_FilterCellWrapperComponent:any[] = ([] as any[]);
export const RenderType_FilterCellWrapperComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FilterCellWrapperComponent,
  data: {}
}
);
export function View_FilterCellWrapperComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-filter-cell-operators',([] as any[]),[[
        2,
        'k-filtercell-operator',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'clear'
      ]
      ,
      [
        (null as any),
        'valueChange'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.FilterCellWrapperComponent = v.component;
      if (('clear' === en)) {
        const pd_0:any = ((<any>co.onClear()) !== false);
        ad = (pd_0 && ad);
      }
      if (('valueChange' === en)) {
        const pd_1:any = ((<any>(co.currentOperator = $event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import2.View_FilterCellOperatorsComponent_0,import2.RenderType_FilterCellOperatorsComponent)),
    import0.ɵdid(24576,(null as any),0,import3.FilterCellOperatorsComponent,([] as any[]),{
      operators: [
        0,
        'operators'
      ]
      ,
      showButton: [
        1,
        'showButton'
      ]
      ,
      showOperators: [
        2,
        'showOperators'
      ]
      ,
      value: [
        3,
        'value'
      ]

    }
    ,{
      valueChange: 'valueChange',
      clear: 'clear'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.FilterCellWrapperComponent = v.component;
    const currVal_1:any = co.operators;
    const currVal_2:any = co.showButton;
    const currVal_3:any = co.showOperators;
    const currVal_4:any = co.currentOperator;
    ck(v,4,0,currVal_1,currVal_2,currVal_3,currVal_4);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,4).hostClasses;
    ck(v,3,0,currVal_0);
  });
}
function View_FilterCellWrapperComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'kendo-grid-filter-wrapper-cell',([] as any[]),[
      [
        2,
        'k-filtercell',
        (null as any)
      ]
      ,
      [
        2,
        'k-filtercell-wrapper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_FilterCellWrapperComponent_0,RenderType_FilterCellWrapperComponent)),
    import0.ɵdid(548864,(null as any),2,import1.FilterCellWrapperComponent,[import4.FilterService],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1}),
    import0.ɵqud(167772160,2,{input: 0})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).overrideBaseClasses;
    const currVal_1:any = import0.ɵnov(v,1).hostClasses;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const FilterCellWrapperComponentNgFactory:import0.ComponentFactory<import1.FilterCellWrapperComponent> = import0.ɵccf('kendo-grid-filter-wrapper-cell',import1.FilterCellWrapperComponent,View_FilterCellWrapperComponent_Host_0,{
  showOperators: 'showOperators',
  operators: 'operators',
  column: 'column',
  filter: 'filter',
  defaultOperator: 'defaultOperator'
}
,{},['*']);
