// @progress/kendo-angular-grid/dist/es/filtering/filter-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/filter-host.directive';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell.component';
const styles_FilterCellComponent:any[] = ([] as any[]);
export const RenderType_FilterCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FilterCellComponent,
  data: {}
}
);
function View_FilterCellComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(8388608,(null as any),(null as any),1,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(368640,(null as any),0,import1.FilterHostDirective,[
      import0.ViewContainerRef,
      import0.ComponentFactoryResolver
    ]
    ,{
      column: [
        0,
        'column'
      ]
      ,
      filter: [
        1,
        'filter'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column;
    const currVal_1:any = co.filter;
    ck(v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_FilterCellComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                    ']))],(null as any),(null as any));
}
function View_FilterCellComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_FilterCellComponent_5)),
    import0.ɵdid(270336,(null as any),0,import2.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      '$implicit',
      'column'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.filterCellTemplateRef;
    const currVal_1:any = ck(v,2,0,co.filter,co.column);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_FilterCellComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterCellComponent_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.column.filterCellTemplateRef;
    ck(v,3,0,currVal_0);
  },(null as any));
}
function View_FilterCellComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(8192,(null as any),0,import2.NgSwitch,([] as any[]),{ngSwitch: [
        0,
        'ngSwitch'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterCellComponent_2)),
    import0.ɵdid(139264,(null as any),0,import2.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import2.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterCellComponent_3)),
    import0.ɵdid(139264,(null as any),0,import2.NgSwitchCase,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import2.NgSwitch
    ]
      ,{ngSwitchCase: [
        0,
        'ngSwitchCase'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hasTemplate;
    ck(v,2,0,currVal_0);
    const currVal_1:any = false;
    ck(v,5,0,currVal_1);
    const currVal_2:any = true;
    ck(v,8,0,currVal_2);
  },(null as any));
}
export function View_FilterCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterCellComponent_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import3.FilterCellComponent = v.component;
    const currVal_0:any = co.isFilterable;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_FilterCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridFilterCell',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_FilterCellComponent_0,RenderType_FilterCellComponent)),
    import0.ɵdid(24576,(null as any),0,import3.FilterCellComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const FilterCellComponentNgFactory:import0.ComponentFactory<import3.FilterCellComponent> = import0.ɵccf('[kendoGridFilterCell]',import3.FilterCellComponent,View_FilterCellComponent_Host_0,{
  column: 'column',
  filter: 'filter'
}
,{},([] as any[]));
