// @progress/kendo-angular-grid/dist/es/filtering/filter-cell-operators.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../../../../kendo-angular-dropdowns/dist/es/dropdownlist.component.ngfactory';
import * as import2 from '@progress/kendo-angular-dropdowns/dist/es/selection.service';
import * as import3 from '@progress/kendo-angular-dropdowns/dist/es/navigation.service';
import * as import4 from '@progress/kendo-angular-dropdowns/dist/es/dropdownlist.component';
import * as import5 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import6 from '@angular/forms';
import * as import7 from '@angular/common';
import * as import8 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-operators.component';
const styles_FilterCellOperatorsComponent:any[] = ([] as any[]);
export const RenderType_FilterCellOperatorsComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FilterCellOperatorsComponent,
  data: {}
}
);
function View_FilterCellOperatorsComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'kendo-dropdownlist',[[
        'class',
        'k-dropdown-operator'
      ]
    ],[
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-dropdown',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-haspopup',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        1,
        'aria-activedescendant',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'valueChange'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keypress'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('blur' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).blur()) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,3).focus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,3).keydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keypress' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,3).keypress($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('click' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,3).click()) !== false);
        ad = (pd_4 && ad);
      }
      if (('valueChange' === en)) {
        const pd_5:any = ((<any>co.onChange($event)) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },import1.View_DropDownListComponent_0,import1.RenderType_DropDownListComponent)),
    import0.ɵprd(256,(null as any),import2.SelectionService,import2.SelectionService,([] as any[])),
    import0.ɵprd(256,(null as any),import3.NavigationService,import3.NavigationService,([] as any[])),
    import0.ɵdid(352256,(null as any),5,import4.DropDownListComponent,[
      [
        2,
        import5.RTL
      ]
      ,
      import2.SelectionService,
      import3.NavigationService,
      import0.ElementRef,
      import0.Renderer,
      import0.NgZone
    ]
    ,{
      iconClass: [
        0,
        'iconClass'
      ]
      ,
      data: [
        1,
        'data'
      ]
      ,
      value: [
        2,
        'value'
      ]
      ,
      textField: [
        3,
        'textField'
      ]
      ,
      valueField: [
        4,
        'valueField'
      ]
      ,
      popupSettings: [
        5,
        'popupSettings'
      ]
      ,
      valuePrimitive: [
        6,
        'valuePrimitive'
      ]

    }
    ,{valueChange: 'valueChange'}),
    import0.ɵqud(167772160,1,{itemTemplate: 0}),
    import0.ɵqud(167772160,2,{valueTemplate: 0}),
    import0.ɵqud(167772160,3,{headerTemplate: 0}),
    import0.ɵqud(167772160,4,{footerTemplate: 0}),
    import0.ɵqud(167772160,5,{noDataTemplate: 0}),
    import0.ɵpod(['width']),
    import0.ɵprd(2560,(null as any),import6.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import4.DropDownListComponent]),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_11:any = 'k-i-filter';
    const currVal_12:any = co.operators;
    const currVal_13:any = co.value;
    const currVal_14:any = 'text';
    const currVal_15:any = 'value';
    const currVal_16:any = ck(v,9,0,'auto');
    const currVal_17:any = true;
    ck(v,3,0,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_1:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_2:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_3:any = import0.ɵnov(v,3).role;
    const currVal_4:any = import0.ɵnov(v,3).widgetTabIndex;
    const currVal_5:any = import0.ɵnov(v,3).ariaDisabled;
    const currVal_6:any = import0.ɵnov(v,3).ariaHasPopup;
    const currVal_7:any = import0.ɵnov(v,3).ariaExpanded;
    const currVal_8:any = import0.ɵnov(v,3).ariaOwns;
    const currVal_9:any = import0.ɵnov(v,3).ariaActivedescendant;
    const currVal_10:any = import0.ɵnov(v,3).dir;
    ck(v,0,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10
    ]
    );
  });
}
function View_FilterCellOperatorsComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'button',[
      [
        'class',
        'k-button k-button-icon'
      ]
      ,
      [
        'title',
        'Clear'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clearClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon k-i-filter-clear'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(null as any));
}
export function View_FilterCellOperatorsComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterCellOperatorsComponent_1)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_FilterCellOperatorsComponent_2)),
    import0.ɵdid(8192,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import8.FilterCellOperatorsComponent = v.component;
    const currVal_0:any = co.showOperators;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.showButton;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_FilterCellOperatorsComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-grid-filter-cell-operators',([] as any[]),[[
        2,
        'k-filtercell-operator',
        (null as any)
      ]
    ],(null as any),(null as any),View_FilterCellOperatorsComponent_0,RenderType_FilterCellOperatorsComponent)),
    import0.ɵdid(24576,(null as any),0,import8.FilterCellOperatorsComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hostClasses;
    ck(v,0,0,currVal_0);
  });
}
export const FilterCellOperatorsComponentNgFactory:import0.ComponentFactory<import8.FilterCellOperatorsComponent> = import0.ɵccf('kendo-grid-filter-cell-operators',import8.FilterCellOperatorsComponent,View_FilterCellOperatorsComponent_Host_0,{
  operators: 'operators',
  showButton: 'showButton',
  showOperators: 'showOperators',
  value: 'value'
}
,{
  valueChange: 'valueChange',
  clear: 'clear'
}
,([] as any[]));
