// @progress/kendo-angular-grid/dist/es/filtering/numeric-filter-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './filter-cell-wrapper.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-wrapper.component';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/filter.service';
import * as import4 from '../../../../kendo-angular-inputs/dist/es/numerictextbox/numerictextbox.component.ngfactory';
import * as import5 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import6 from '@progress/kendo-angular-l10n/dist/es/message.service';
import * as import7 from '@progress/kendo-angular-inputs/dist/es/numerictextbox/numerictextbox.component';
import * as import8 from '@progress/kendo-angular-intl/dist/es/intl.service';
import * as import9 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import10 from '@angular/forms';
import * as import11 from '@progress/kendo-angular-grid/dist/es/filtering/filter-input.directive';
import * as import12 from '@progress/kendo-angular-grid/dist/es/filtering/numeric-filter-cell.component';
const styles_NumericFilterCellComponent:any[] = ([] as any[]);
export const RenderType_NumericFilterCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NumericFilterCellComponent,
  data: {}
}
);
export function View_NumericFilterCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),13,'kendo-grid-filter-wrapper-cell',([] as any[]),[
      [
        2,
        'k-filtercell',
        (null as any)
      ]
      ,
      [
        2,
        'k-filtercell-wrapper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_FilterCellWrapperComponent_0,import1.RenderType_FilterCellWrapperComponent)),
    import0.ɵdid(548864,(null as any),2,import2.FilterCellWrapperComponent,[import3.FilterService],{
      showOperators: [
        0,
        'showOperators'
      ]
      ,
      operators: [
        1,
        'operators'
      ]
      ,
      column: [
        2,
        'column'
      ]
      ,
      filter: [
        3,
        'filter'
      ]
      ,
      defaultOperator: [
        4,
        'defaultOperator'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1}),
    import0.ɵqud(167772160,2,{input: 0}),
    (l()(),import0.ɵted(0,['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'kendo-numerictextbox',[[
        'kendoFilterInput',
        ''
      ]
    ],[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-numerictextbox',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import4.View_NumericTextBoxComponent_0,import4.RenderType_NumericTextBoxComponent)),
    import0.ɵprd(128,(null as any),import5.L10N_PREFIX,'kendo.numerictextbox',([] as any[])),
    import0.ɵprd(256,(null as any),import5.LocalizationService,import5.LocalizationService,[
      import5.L10N_PREFIX,
      [
        2,
        import6.MessageService
      ]

    ]
    ),
    import0.ɵdid(352256,(null as any),0,import7.NumericTextBoxComponent,[
      import8.IntlService,
      import0.Renderer,
      import5.LocalizationService,
      [
        2,
        import9.RTL
      ]

    ]
    ,{
      autoCorrect: [
        0,
        'autoCorrect'
      ]
      ,
      format: [
        1,
        'format'
      ]
      ,
      max: [
        2,
        'max'
      ]
      ,
      min: [
        3,
        'min'
      ]
      ,
      decimals: [
        4,
        'decimals'
      ]
      ,
      step: [
        5,
        'step'
      ]
      ,
      spinners: [
        6,
        'spinners'
      ]
      ,
      value: [
        7,
        'value'
      ]

    }
    ,(null as any)),
    import0.ɵprd(512,(null as any),import10.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import7.NumericTextBoxComponent]),
      import0.ɵdid(40960,[[
        2,
        4
      ]
      ],0,import11.FilterInputDirective,[import10.NG_VALUE_ACCESSOR],{value: [
        0,
        'value'
      ]
    },(null as any)),
    import0.ɵprd(2560,(null as any),import10.NG_VALIDATORS,(p0_0:any) => {
      return [p0_0];
    },[import7.NumericTextBoxComponent]),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import12.NumericFilterCellComponent = v.component;
    const currVal_2:any = co.showOperators;
    const currVal_3:any = co.operators;
    const currVal_4:any = co.column;
    const currVal_5:any = co.filter;
    const currVal_6:any = co.operator;
    ck(v,2,0,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
    const currVal_10:any = true;
    const currVal_11:any = co.format;
    const currVal_12:any = co.max;
    const currVal_13:any = co.min;
    const currVal_14:any = co.decimals;
    const currVal_15:any = co.step;
    const currVal_16:any = co.spinners;
    const currVal_17:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    ck(v,9,0,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17);
    const currVal_18:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    ck(v,11,0,currVal_18);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).overrideBaseClasses;
    const currVal_1:any = import0.ɵnov(v,2).hostClasses;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_7:any = import0.ɵnov(v,9).direction;
    const currVal_8:any = import0.ɵnov(v,9).widgetClasses;
    const currVal_9:any = import0.ɵnov(v,9).widgetClasses;
    ck(v,6,0,currVal_7,currVal_8,currVal_9);
  });
}
function View_NumericFilterCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-numeric-filter-cell',([] as any[]),[[
        2,
        'k-filtercell',
        (null as any)
      ]
    ],(null as any),(null as any),View_NumericFilterCellComponent_0,RenderType_NumericFilterCellComponent)),
    import0.ɵdid(548864,(null as any),1,import12.NumericFilterCellComponent,[
      import3.FilterService,
      import5.LocalizationService
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hostClasses;
    ck(v,0,0,currVal_0);
  });
}
export const NumericFilterCellComponentNgFactory:import0.ComponentFactory<import12.NumericFilterCellComponent> = import0.ɵccf('kendo-grid-numeric-filter-cell',import12.NumericFilterCellComponent,View_NumericFilterCellComponent_Host_0,{
  showOperators: 'showOperators',
  column: 'column',
  filter: 'filter',
  operator: 'operator',
  step: 'step',
  min: 'min',
  max: 'max',
  spinners: 'spinners',
  decimals: 'decimals',
  format: 'format'
}
,{},([] as any[]));
