// @progress/kendo-angular-grid/dist/es/filtering/string-filter-cell.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './filter-cell-wrapper.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/filtering/filter-cell-wrapper.component';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/filter.service';
import * as import4 from '@angular/forms';
import * as import5 from '@progress/kendo-angular-grid/dist/es/filtering/filter-input.directive';
import * as import6 from '@progress/kendo-angular-grid/dist/es/filtering/string-filter-cell.component';
import * as import7 from '@progress/kendo-angular-l10n/dist/es/localization.service';
const styles_StringFilterCellComponent:any[] = ([] as any[]);
export const RenderType_StringFilterCellComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_StringFilterCellComponent,
  data: {}
}
);
export function View_StringFilterCellComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),12,'kendo-grid-filter-wrapper-cell',([] as any[]),[
      [
        2,
        'k-filtercell',
        (null as any)
      ]
      ,
      [
        2,
        'k-filtercell-wrapper',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_FilterCellWrapperComponent_0,import1.RenderType_FilterCellWrapperComponent)),
    import0.ɵdid(548864,(null as any),2,import2.FilterCellWrapperComponent,[import3.FilterService],{
      showOperators: [
        0,
        'showOperators'
      ]
      ,
      operators: [
        1,
        'operators'
      ]
      ,
      column: [
        2,
        'column'
      ]
      ,
      filter: [
        3,
        'filter'
      ]
      ,
      defaultOperator: [
        4,
        'defaultOperator'
      ]

    }
    ,(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1}),
    import0.ɵqud(167772160,2,{input: 0}),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,6,'input',[
      [
        'class',
        'k-textbox'
      ]
      ,
      [
        'kendoFilterInput',
        ''
      ]

    ]
    ,[
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,7)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,7).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,7)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,7)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import4.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import4.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import4.DefaultValueAccessor]),
    import0.ɵdid(335872,(null as any),0,import4.NgModel,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import4.NG_VALUE_ACCESSOR
      ]

    ]
      ,{model: [
        0,
        'model'
      ]
    },(null as any)),
    import0.ɵprd(1024,(null as any),import4.NgControl,(null as any),[import4.NgModel]),
    import0.ɵdid(8192,(null as any),0,import4.NgControlStatus,[import4.NgControl],(null as any),(null as any)),
      import0.ɵdid(40960,[[
        2,
        4
      ]
    ],0,import5.FilterInputDirective,[import4.NG_VALUE_ACCESSOR],(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import6.StringFilterCellComponent = v.component;
    const currVal_2:any = co.showOperators;
    const currVal_3:any = co.operators;
    const currVal_4:any = co.column;
    const currVal_5:any = co.filter;
    const currVal_6:any = co.operator;
    ck(v,2,0,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
    const currVal_14:any = ((co.currentFilter == null)? (null as any): co.currentFilter.value);
    ck(v,9,0,currVal_14);
    ck(v,12,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).overrideBaseClasses;
    const currVal_1:any = import0.ɵnov(v,2).hostClasses;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_7:any = import0.ɵnov(v,11).ngClassUntouched;
    const currVal_8:any = import0.ɵnov(v,11).ngClassTouched;
    const currVal_9:any = import0.ɵnov(v,11).ngClassPristine;
    const currVal_10:any = import0.ɵnov(v,11).ngClassDirty;
    const currVal_11:any = import0.ɵnov(v,11).ngClassValid;
    const currVal_12:any = import0.ɵnov(v,11).ngClassInvalid;
    const currVal_13:any = import0.ɵnov(v,11).ngClassPending;
    ck(v,6,0,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13);
  });
}
function View_StringFilterCellComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-string-filter-cell',([] as any[]),[[
        2,
        'k-filtercell',
        (null as any)
      ]
    ],(null as any),(null as any),View_StringFilterCellComponent_0,RenderType_StringFilterCellComponent)),
    import0.ɵdid(548864,(null as any),1,import6.StringFilterCellComponent,[
      import3.FilterService,
      import7.LocalizationService
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(301989888,1,{operatorList: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hostClasses;
    ck(v,0,0,currVal_0);
  });
}
export const StringFilterCellComponentNgFactory:import0.ComponentFactory<import6.StringFilterCellComponent> = import0.ɵccf('kendo-grid-string-filter-cell',import6.StringFilterCellComponent,View_StringFilterCellComponent_Host_0,{
  showOperators: 'showOperators',
  column: 'column',
  filter: 'filter',
  operator: 'operator'
}
,{},([] as any[]));
