// @progress/kendo-angular-grid/dist/es/filtering/operators/ends-with-filter-operator.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/operators/ends-with-filter-operator.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/operators/filter-operator.base';
const styles_EndsWithFilterOperatorComponent:any[] = ([] as any[]);
export const RenderType_EndsWithFilterOperatorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_EndsWithFilterOperatorComponent,
  data: {}
}
);
export function View_EndsWithFilterOperatorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_EndsWithFilterOperatorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-filter-endswith-operator',([] as any[]),(null as any),(null as any),(null as any),View_EndsWithFilterOperatorComponent_0,RenderType_EndsWithFilterOperatorComponent)),
    import0.ɵdid(24576,(null as any),0,import1.EndsWithFilterOperatorComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.FilterOperatorBase,(null as any),[import1.EndsWithFilterOperatorComponent])
  ]
  ,(null as any),(null as any));
}
export const EndsWithFilterOperatorComponentNgFactory:import0.ComponentFactory<import1.EndsWithFilterOperatorComponent> = import0.ɵccf('kendo-filter-endswith-operator',import1.EndsWithFilterOperatorComponent,View_EndsWithFilterOperatorComponent_Host_0,{text: 'text'},{},([] as any[]));
