// @progress/kendo-angular-grid/dist/es/filtering/operators/gte-filter-operator.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/operators/gte-filter-operator.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/operators/filter-operator.base';
const styles_GreaterOrEqualToFilterOperatorComponent:any[] = ([] as any[]);
export const RenderType_GreaterOrEqualToFilterOperatorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_GreaterOrEqualToFilterOperatorComponent,
  data: {}
}
);
export function View_GreaterOrEqualToFilterOperatorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_GreaterOrEqualToFilterOperatorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-filter-gte-operator',([] as any[]),(null as any),(null as any),(null as any),View_GreaterOrEqualToFilterOperatorComponent_0,RenderType_GreaterOrEqualToFilterOperatorComponent)),
    import0.ɵdid(24576,(null as any),0,import1.GreaterOrEqualToFilterOperatorComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.FilterOperatorBase,(null as any),[import1.GreaterOrEqualToFilterOperatorComponent])
  ]
  ,(null as any),(null as any));
}
export const GreaterOrEqualToFilterOperatorComponentNgFactory:import0.ComponentFactory<import1.GreaterOrEqualToFilterOperatorComponent> = import0.ɵccf('kendo-filter-gte-operator',import1.GreaterOrEqualToFilterOperatorComponent,View_GreaterOrEqualToFilterOperatorComponent_Host_0,{text: 'text'},{},([] as any[]));
