// @progress/kendo-angular-grid/dist/es/filtering/operators/is-not-empty-filter-operator.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/operators/is-not-empty-filter-operator.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/operators/filter-operator.base';
const styles_IsNotEmptyFilterOperatorComponent:any[] = ([] as any[]);
export const RenderType_IsNotEmptyFilterOperatorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_IsNotEmptyFilterOperatorComponent,
  data: {}
}
);
export function View_IsNotEmptyFilterOperatorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_IsNotEmptyFilterOperatorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-filter-isempty-operator',([] as any[]),(null as any),(null as any),(null as any),View_IsNotEmptyFilterOperatorComponent_0,RenderType_IsNotEmptyFilterOperatorComponent)),
    import0.ɵdid(24576,(null as any),0,import1.IsNotEmptyFilterOperatorComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.FilterOperatorBase,(null as any),[import1.IsNotEmptyFilterOperatorComponent])
  ]
  ,(null as any),(null as any));
}
export const IsNotEmptyFilterOperatorComponentNgFactory:import0.ComponentFactory<import1.IsNotEmptyFilterOperatorComponent> = import0.ɵccf('kendo-filter-isempty-operator',import1.IsNotEmptyFilterOperatorComponent,View_IsNotEmptyFilterOperatorComponent_Host_0,{text: 'text'},{},([] as any[]));
