// @progress/kendo-angular-grid/dist/es/filtering/operators/lte-filter-operator.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/operators/lte-filter-operator.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/operators/filter-operator.base';
const styles_LessOrEqualToFilterOperatorComponent:any[] = ([] as any[]);
export const RenderType_LessOrEqualToFilterOperatorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_LessOrEqualToFilterOperatorComponent,
  data: {}
}
);
export function View_LessOrEqualToFilterOperatorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_LessOrEqualToFilterOperatorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-filter-lte-operator',([] as any[]),(null as any),(null as any),(null as any),View_LessOrEqualToFilterOperatorComponent_0,RenderType_LessOrEqualToFilterOperatorComponent)),
    import0.ɵdid(24576,(null as any),0,import1.LessOrEqualToFilterOperatorComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.FilterOperatorBase,(null as any),[import1.LessOrEqualToFilterOperatorComponent])
  ]
  ,(null as any),(null as any));
}
export const LessOrEqualToFilterOperatorComponentNgFactory:import0.ComponentFactory<import1.LessOrEqualToFilterOperatorComponent> = import0.ɵccf('kendo-filter-lte-operator',import1.LessOrEqualToFilterOperatorComponent,View_LessOrEqualToFilterOperatorComponent_Host_0,{text: 'text'},{},([] as any[]));
