// @progress/kendo-angular-grid/dist/es/filtering/operators/not-contains-filter-operator.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/filtering/operators/not-contains-filter-operator.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/filtering/operators/filter-operator.base';
const styles_DoesNotContainFilterOperatorComponent:any[] = ([] as any[]);
export const RenderType_DoesNotContainFilterOperatorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DoesNotContainFilterOperatorComponent,
  data: {}
}
);
export function View_DoesNotContainFilterOperatorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_DoesNotContainFilterOperatorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-filter-not-contains-operator',([] as any[]),(null as any),(null as any),(null as any),View_DoesNotContainFilterOperatorComponent_0,RenderType_DoesNotContainFilterOperatorComponent)),
    import0.ɵdid(24576,(null as any),0,import1.DoesNotContainFilterOperatorComponent,[import2.LocalizationService],(null as any),(null as any)),
    import0.ɵprd(3072,(null as any),import3.FilterOperatorBase,(null as any),[import1.DoesNotContainFilterOperatorComponent])
  ]
  ,(null as any),(null as any));
}
export const DoesNotContainFilterOperatorComponentNgFactory:import0.ComponentFactory<import1.DoesNotContainFilterOperatorComponent> = import0.ɵccf('kendo-filter-not-contains-operator',import1.DoesNotContainFilterOperatorComponent,View_DoesNotContainFilterOperatorComponent_Host_0,{text: 'text'},{},([] as any[]));
