// @progress/kendo-angular-grid/dist/es/grouping/group-header.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-grid/dist/es/field-accessor.pipe';
import * as import3 from '@progress/kendo-angular-intl/dist/es/intl.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/template-context.directive';
import * as import5 from '@progress/kendo-angular-grid/dist/es/grouping/group-header.component';
import * as import6 from '@progress/kendo-angular-grid/dist/es/grouping/groups.service';
import * as import7 from '@progress/kendo-angular-grid/dist/es/grouping/group-info.service';
const styles_GroupHeaderComponent:any[] = ([] as any[]);
export const RenderType_GroupHeaderComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_GroupHeaderComponent,
  data: {}
}
);
function View_GroupHeaderComponent_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'td',([] as any[]),[[
        2,
        'k-group-cell',
        (null as any)
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
function View_GroupHeaderComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GroupHeaderComponent_2)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.prefixGroupCell(co.item);
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_GroupHeaderComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),[
      '\n                    ',
      ': ',
      '\n                    '
    ]
    )),
    import0.ɵpid(0,import2.FieldAccessorPipe,[import3.IntlService])
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groupTitle(co.item);
    const currVal_1:any = import0.ɵunv(v,0,1,import0.ɵnov(v,1).transform(co.item.data,'value',co.formatForGroup(co.item)));
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_GroupHeaderComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                    ']))],(null as any),(null as any));
}
function View_GroupHeaderComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'a',[
      [
        'href',
        '#'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleGroup(co.item)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GroupHeaderComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_GroupHeaderComponent_5)),
      import0.ɵdid(73728,(null as any),0,import4.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'group',
      'aggregates',
      'value',
      'field',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.groupButtonStyles(co.item.index);
    ck(v,2,0,currVal_0);
    const currVal_1:boolean = !co.groupHeaderTemplate(co.item);
    ck(v,6,0,currVal_1);
    const currVal_2:any = ck(v,10,0,co.groupHeaderTemplate(co.item),co.item.data,((co.item.data == null)? (null as any): co.item.data.aggregates),((co.item.data == null)? (null as any): co.item.data.value),((co.item.data == null)? (null as any): co.item.data.field),co.item.data);
    ck(v,9,0,currVal_2);
  },(null as any));
}
export function View_GroupHeaderComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GroupHeaderComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'td',([] as any[]),[[
        1,
        'colspan',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'p',[[
        'class',
        'k-reset'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GroupHeaderComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import5.GroupHeaderComponent = v.component;
    const currVal_0:boolean = !co.skipGroupDecoration;
    ck(v,2,0,currVal_0);
    const currVal_2:boolean = !co.skipGroupDecoration;
    ck(v,9,0,currVal_2);
  },(ck,v) => {
    var co:import5.GroupHeaderComponent = v.component;
    const currVal_1:any = co.groupSpan(co.item);
    ck(v,4,0,currVal_1);
  });
}
function View_GroupHeaderComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGridGroupHeader',
        ''
      ]
      ],[[
        2,
        'k-grouping-row',
        (null as any)
      ]
    ],(null as any),(null as any),View_GroupHeaderComponent_0,RenderType_GroupHeaderComponent)),
    import0.ɵdid(24576,(null as any),0,import5.GroupHeaderComponent,[
      import6.GroupsService,
      import7.GroupInfoService
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).groupItemClass;
    ck(v,0,0,currVal_0);
  });
}
export const GroupHeaderComponentNgFactory:import0.ComponentFactory<import5.GroupHeaderComponent> = import0.ɵccf('[kendoGridGroupHeader]',import5.GroupHeaderComponent,View_GroupHeaderComponent_Host_0,{
  item: 'item',
  skipGroupDecoration: 'skipGroupDecoration',
  columns: 'columns',
  groups: 'groups'
}
,{},([] as any[]));
