// @progress/kendo-angular-grid/dist/es/grouping/group-indicator.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/grouping/group-indicator.component';
import * as import2 from '@progress/kendo-angular-grid/dist/es/grouping/group-info.service';
const styles_GroupIndicatorComponent:any[] = ([] as any[]);
export const RenderType_GroupIndicatorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_GroupIndicatorComponent,
  data: {}
}
);
export function View_GroupIndicatorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[
      [
        'class',
        'k-link'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.GroupIndicatorComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggleDirection()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon'
      ]
    ],[
      [
        2,
        'k-i-sarrow-n',
        (null as any)
      ]
      ,
      [
        2,
        'k-i-sarrow-s',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n            ',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[[
        'class',
        'k-button k-button-icon k-bare'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.GroupIndicatorComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.removeDescriptor()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon k-i-group-delete'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.GroupIndicatorComponent = v.component;
    const currVal_0:any = (co.dir === 'asc');
    const currVal_1:any = (co.dir === 'desc');
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.title;
    ck(v,4,0,currVal_2);
  });
}
function View_GroupIndicatorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'kendoGroupIndicator',
        ''
      ]
      ],[[
        2,
        'k-group-indicator',
        (null as any)
      ]
    ],(null as any),(null as any),View_GroupIndicatorComponent_0,RenderType_GroupIndicatorComponent)),
    import0.ɵdid(24576,(null as any),0,import1.GroupIndicatorComponent,[import2.GroupInfoService],(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).groupIndicatorClass;
    ck(v,0,0,currVal_0);
  });
}
export const GroupIndicatorComponentNgFactory:import0.ComponentFactory<import1.GroupIndicatorComponent> = import0.ɵccf('[kendoGroupIndicator]',import1.GroupIndicatorComponent,View_GroupIndicatorComponent_Host_0,{group: 'group'},{
  directionChange: 'directionChange',
  remove: 'remove'
}
,([] as any[]));
