// @progress/kendo-angular-grid/dist/es/grouping/group-panel.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './group-indicator.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/draggable.directive';
import * as import3 from '@progress/kendo-angular-grid/dist/es/grouping/group-indicator.component';
import * as import4 from '@progress/kendo-angular-grid/dist/es/grouping/group-info.service';
import * as import5 from '@angular/common';
import * as import6 from '@progress/kendo-angular-grid/dist/es/grouping/group-panel.component';
import * as import7 from '@progress/kendo-angular-grid/dist/es/grouping/group-connection.service';
import * as import8 from '@progress/kendo-angular-l10n/dist/es/localization.service';
const styles_GroupPanelComponent:any[] = ([] as any[]);
export const RenderType_GroupPanelComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_GroupPanelComponent,
  data: {}
}
);
function View_GroupPanelComponent_1(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '\n        ',
      '\n    '
    ]
  ))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.text;
    ck(v,0,0,currVal_0);
  });
}
function View_GroupPanelComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,[[
        2,
        0
      ]
      ],(null as any),3,'div',[[
        'kendoGroupIndicator',
        ''
      ]
      ],[[
        2,
        'k-group-indicator',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'directionChange'
      ]
      ,
      [
        (null as any),
        'remove'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('directionChange' === en)) {
        const pd_0:any = ((<any>co.directionChange($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('remove' === en)) {
        const pd_1:any = ((<any>co.remove($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import1.View_GroupIndicatorComponent_0,import1.RenderType_GroupIndicatorComponent)),
      import0.ɵdid(73728,[[
        1,
        4
      ]
      ],0,import2.DraggableDirective,[import0.ElementRef],{kendoGridDraggable: [
        0,
        'kendoGridDraggable'
      ]
    },(null as any)),
      import0.ɵdid(24576,(null as any),0,import3.GroupIndicatorComponent,[import4.GroupInfoService],{group: [
        0,
        'group'
      ]
    },{
      directionChange: 'directionChange',
      remove: 'remove'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,1,0,currVal_1);
    const currVal_2:any = v.context.$implicit;
    ck(v,2,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).groupIndicatorClass;
    ck(v,0,0,currVal_0);
  });
}
export function View_GroupPanelComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    import0.ɵqud(335544320,1,{draggables: 1}),
    import0.ɵqud(335544320,2,{dropTargets: 1}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GroupPanelComponent_1)),
    import0.ɵdid(8192,(null as any),0,import5.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_GroupPanelComponent_2)),
    import0.ɵdid(401408,(null as any),0,import5.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import6.GroupPanelComponent = v.component;
    const currVal_0:any = (co.groups.length === 0);
    ck(v,4,0,currVal_0);
    const currVal_1:any = co.groups;
    ck(v,7,0,currVal_1);
  },(null as any));
}
function View_GroupPanelComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-grid-group-panel',([] as any[]),[[
        2,
        'k-grouping-header',
        (null as any)
      ]
    ],(null as any),(null as any),View_GroupPanelComponent_0,RenderType_GroupPanelComponent)),
    import0.ɵprd(256,(null as any),import7.GroupDragService,import7.GroupDragService,[
      import7.GroupConnectionService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(2121728,(null as any),0,import6.GroupPanelComponent,[
      import7.GroupConnectionService,
      import7.GroupDragService,
      import0.ElementRef,
      import8.LocalizationService
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).groupHeaderClass;
    ck(v,0,0,currVal_0);
  });
}
export const GroupPanelComponentNgFactory:import0.ComponentFactory<import6.GroupPanelComponent> = import0.ɵccf('kendo-grid-group-panel',import6.GroupPanelComponent,View_GroupPanelComponent_Host_0,{
  text: 'text',
  groups: 'groups'
}
,{change: 'change'},([] as any[]));
