// @progress/kendo-angular-buttons/dist/es/dropdownbutton/dropdownbutton.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../../../../kendo-angular-popup/dist/es/popup.component.ngfactory';
import * as import2 from '@progress/kendo-angular-popup/dist/es/services/dom.service';
import * as import3 from '@progress/kendo-angular-popup/dist/es/services/align.service';
import * as import4 from '@progress/kendo-angular-popup/dist/es/services/position.service';
import * as import5 from '@progress/kendo-angular-popup/dist/es/services/resize.service';
import * as import6 from '@progress/kendo-angular-popup/dist/es/services/scrollable.service';
import * as import7 from '@progress/kendo-angular-popup/dist/es/popup.component';
import * as import8 from '../listbutton/list.component.ngfactory';
import * as import9 from '@progress/kendo-angular-buttons/dist/es/listbutton/list.component';
import * as import10 from '@progress/kendo-angular-buttons/dist/es/dropdownbutton/dropdownbutton.component';
import * as import11 from '@progress/kendo-angular-buttons/dist/es/button/button.directive';
import * as import12 from '@progress/kendo-angular-buttons/dist/es/button/button.service';
import * as import13 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import14 from '@angular/common';
import * as import15 from '@progress/kendo-angular-buttons/dist/es/focusable/focus.service';
import * as import16 from '@progress/kendo-angular-buttons/dist/es/navigation/navigation-config';
import * as import17 from '@progress/kendo-angular-buttons/dist/es/navigation/navigation.service';
const styles_DropDownButtonComponent:any[] = ([] as any[]);
export const RenderType_DropDownButtonComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DropDownButtonComponent,
  data: {}
}
);
function View_DropDownButtonComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'kendo-popup',([] as any[]),[
      [
        4,
        'left',
        'px'
      ]
      ,
      [
        4,
        'top',
        'px'
      ]

    ]
      ,[[
        (null as any),
        'anchorViewportLeave'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('anchorViewportLeave' === en)) {
        const pd_0:any = ((<any>(co.openState = false)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import1.View_PopupComponent_0,import1.RenderType_PopupComponent)),
    import0.ɵprd(256,(null as any),import2.DOMService,import2.DOMService,([] as any[])),
    import0.ɵprd(256,(null as any),import3.AlignService,import3.AlignService,[import2.DOMService]),
    import0.ɵprd(256,(null as any),import4.PositionService,import4.PositionService,[import2.DOMService]),
    import0.ɵprd(256,(null as any),import5.ResizeService,import5.ResizeService,[
      import2.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵprd(256,(null as any),import6.ScrollableService,import6.ScrollableService,[
      import2.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(4317184,(null as any),0,import7.PopupComponent,[
      import3.AlignService,
      import0.ElementRef,
      import0.ChangeDetectorRef,
      import4.PositionService,
      import5.ResizeService,
      import6.ScrollableService,
      import0.Renderer
    ]
    ,{
      animate: [
        0,
        'animate'
      ]
      ,
      anchor: [
        1,
        'anchor'
      ]
      ,
      anchorAlign: [
        2,
        'anchorAlign'
      ]
      ,
      popupAlign: [
        3,
        'popupAlign'
      ]
      ,
      popupClass: [
        4,
        'popupClass'
      ]

    }
    ,{anchorViewportLeave: 'anchorViewportLeave'}),
    (l()(),import0.ɵted(0,['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),0,2,'kendo-button-list',([] as any[]),[[
        8,
        'id',
        0
      ]
      ],[[
        (null as any),
        'onItemClick'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onItemClick' === en)) {
        const pd_0:any = ((<any>co.onItemClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import8.View_ListComponent_0,import8.RenderType_ListComponent)),
    import0.ɵdid(24576,(null as any),0,import9.ListComponent,([] as any[]),{
      data: [
        0,
        'data'
      ]
      ,
      textField: [
        1,
        'textField'
      ]
      ,
      itemTemplate: [
        2,
        'itemTemplate'
      ]

    }
    ,{onItemClick: 'onItemClick'}),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted(0,['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = ((co.popupSettings == null)? (null as any): co.popupSettings.animate);
    const currVal_3:any = import0.ɵnov((<any>v.parent),1);
    const currVal_4:any = co.anchorAlign;
    const currVal_5:any = co.popupAlign;
    const currVal_6:any = co.popupClasses;
    ck(v,6,0,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
    const currVal_8:any = co.data;
    const currVal_9:any = co.textField;
    const currVal_10:any = co.itemTemplate;
    ck(v,9,0,currVal_8,currVal_9,currVal_10);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵnov(v,6).offsetLeft;
    const currVal_1:any = import0.ɵnov(v,6).offsetTop;
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_7:any = co.listId;
    ck(v,8,0,currVal_7);
  });
}
export function View_DropDownButtonComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,[[
        'button',
        1
      ]
    ],(null as any),4,'button',[
      [
        'kendoButton',
        ''
      ]
      ,
      [
        'tabIndex',
        '-1'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        2,
        'k-state-active',
        (null as any)
      ]
      ,
      [
        2,
        'k-button',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'k-primary',
        (null as any)
      ]
      ,
      [
        2,
        'k-bare',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-active',
        (null as any)
      ]
      ,
      [
        2,
        'k-button-icontext',
        (null as any)
      ]
      ,
      [
        2,
        'k-button-icon',
        (null as any)
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import10.DropDownButtonComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,2).onClick()) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>co.openPopup()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(73728,(null as any),0,import11.ButtonDirective,[
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import12.KendoButtonService
      ]
      ,
      [
        2,
        import13.RTL
      ]

    ]
    ,{
      tabIndex: [
        0,
        'tabIndex'
      ]
      ,
      icon: [
        1,
        'icon'
      ]
      ,
      iconClass: [
        2,
        'iconClass'
      ]
      ,
      imageUrl: [
        3,
        'imageUrl'
      ]
      ,
      disabled: [
        4,
        'disabled'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DropDownButtonComponent_1)),
    import0.ɵdid(8192,(null as any),0,import14.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import10.DropDownButtonComponent = v.component;
    const currVal_9:any = '-1';
    const currVal_10:any = co.icon;
    const currVal_11:any = co.iconClass;
    const currVal_12:any = co.imageUrl;
    const currVal_13:any = co.disabled;
    ck(v,2,0,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13);
    const currVal_14:any = co.openState;
    ck(v,8,0,currVal_14);
  },(ck,v) => {
    var co:import10.DropDownButtonComponent = v.component;
    const currVal_0:any = co.active;
    const currVal_1:any = import0.ɵnov(v,2).classButton;
    const currVal_2:any = import0.ɵnov(v,2).classDisabled;
    const currVal_3:any = import0.ɵnov(v,2).classPrimary;
    const currVal_4:any = import0.ɵnov(v,2).classBare;
    const currVal_5:any = import0.ɵnov(v,2).classActive;
    const currVal_6:any = import0.ɵnov(v,2).classIconText;
    const currVal_7:any = import0.ɵnov(v,2).classIcon;
    const currVal_8:any = import0.ɵnov(v,2).dir;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
function View_DropDownButtonComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'kendo-dropdownbutton',([] as any[]),[
      [
        8,
        'tabIndex',
        0
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-haspopup',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-dropdown-button',
        (null as any)
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keypress'
      ]
      ,
      [
        (null as any),
        'keyup'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('focus' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,4).focus()) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,4).blur()) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,4).keydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keypress' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,4).keypress($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('keyup' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,4).keyup($event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('mousedown' === en)) {
        const pd_5:any = ((<any>import0.ɵnov(v,4).mousedown($event)) !== false);
        ad = (pd_5 && ad);
      }
      return ad;
    },View_DropDownButtonComponent_0,RenderType_DropDownButtonComponent)),
    import0.ɵprd(256,(null as any),import15.FocusService,import15.FocusService,([] as any[])),
    import0.ɵprd(128,(null as any),import16.NAVIGATION_CONFIG,{useLeftRightArrows: true},([] as any[])),
    import0.ɵprd(256,(null as any),import17.NavigationService,import17.NavigationService,[import16.NAVIGATION_CONFIG]),
    import0.ɵdid(90112,(null as any),1,import10.DropDownButtonComponent,[
      import15.FocusService,
      import17.NavigationService,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import13.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{itemTemplate: 0})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,4).componentTabIndex;
    const currVal_1:any = import0.ɵnov(v,4).focused;
    const currVal_2:any = import0.ɵnov(v,4).role;
    const currVal_3:any = import0.ɵnov(v,4).ariaDisabled;
    const currVal_4:any = import0.ɵnov(v,4).ariaExpanded;
    const currVal_5:any = import0.ɵnov(v,4).ariaHasPopup;
    const currVal_6:any = import0.ɵnov(v,4).ariaOwns;
    const currVal_7:any = import0.ɵnov(v,4).widgetClasses;
    const currVal_8:any = import0.ɵnov(v,4).widgetClasses;
    const currVal_9:any = import0.ɵnov(v,4).dir;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
export const DropDownButtonComponentNgFactory:import0.ComponentFactory<import10.DropDownButtonComponent> = import0.ɵccf('kendo-dropdownbutton',import10.DropDownButtonComponent,View_DropDownButtonComponent_Host_0,{
  icon: 'icon',
  iconClass: 'iconClass',
  imageUrl: 'imageUrl',
  popupSettings: 'popupSettings',
  textField: 'textField',
  disabled: 'disabled',
  data: 'data',
  tabIndex: 'tabIndex'
}
,{
  itemClick: 'itemClick',
  open: 'open',
  close: 'close'
}
,['*']);
