// @progress/kendo-angular-grid/dist/es/pdf/pdf-margin.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/pdf/pdf-margin.component';
const styles_PDFMarginComponent:any[] = ([] as any[]);
export const RenderType_PDFMarginComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PDFMarginComponent,
  data: {}
}
);
export function View_PDFMarginComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_PDFMarginComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-grid-pdf-margin',([] as any[]),(null as any),(null as any),(null as any),View_PDFMarginComponent_0,RenderType_PDFMarginComponent)),
    import0.ɵdid(24576,(null as any),0,import1.PDFMarginComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PDFMarginComponentNgFactory:import0.ComponentFactory<import1.PDFMarginComponent> = import0.ɵccf('kendo-grid-pdf-margin',import1.PDFMarginComponent,View_PDFMarginComponent_Host_0,{
  bottom: 'bottom',
  left: 'left',
  right: 'right',
  top: 'top'
}
,{},([] as any[]));
