// @progress/kendo-angular-grid/dist/es/pdf/pdf.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/pdf/pdf.component';
import * as import2 from '@progress/kendo-angular-grid/dist/es/pdf/pdf.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/suspend.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/change-notification.service';
const styles_PDFComponent:any[] = ([] as any[]);
export const RenderType_PDFComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PDFComponent,
  data: {}
}
);
export function View_PDFComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_PDFComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'kendo-grid-pdf',([] as any[]),(null as any),(null as any),(null as any),View_PDFComponent_0,RenderType_PDFComponent)),
    import0.ɵdid(90112,(null as any),2,import1.PDFComponent,[
      import2.PDFService,
      import3.SuspendService,
      import0.NgZone,
      import4.ChangeNotificationService
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{pageTemplateDirective: 0}),
    import0.ɵqud(167772160,2,{marginComponent: 0})
  ]
  ,(null as any),(null as any));
}
export const PDFComponentNgFactory:import0.ComponentFactory<import1.PDFComponent> = import0.ɵccf('kendo-grid-pdf',import1.PDFComponent,View_PDFComponent_Host_0,{
  allPages: 'allPages',
  author: 'author',
  avoidLinks: 'avoidLinks',
  creator: 'creator',
  date: 'date',
  fileName: 'fileName',
  forceProxy: 'forceProxy',
  keywords: 'keywords',
  landscape: 'landscape',
  margin: 'margin',
  paperSize: 'paperSize',
  repeatHeaders: 'repeatHeaders',
  scale: 'scale',
  proxyURL: 'proxyURL',
  proxyTarget: 'proxyTarget',
  subject: 'subject',
  title: 'title'
}
,{},([] as any[]));
