// @progress/kendo-angular-dropdowns/dist/es/autocomplete.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-dropdowns/dist/es/templates/template-context.directive';
import * as import2 from '@angular/common';
import * as import3 from '../../../kendo-angular-popup/dist/es/popup.component.ngfactory';
import * as import4 from '@progress/kendo-angular-popup/dist/es/services/dom.service';
import * as import5 from '@progress/kendo-angular-popup/dist/es/services/align.service';
import * as import6 from '@progress/kendo-angular-popup/dist/es/services/position.service';
import * as import7 from '@progress/kendo-angular-popup/dist/es/services/resize.service';
import * as import8 from '@progress/kendo-angular-popup/dist/es/services/scrollable.service';
import * as import9 from '@progress/kendo-angular-popup/dist/es/popup.component';
import * as import10 from './list.component.ngfactory';
import * as import11 from '@progress/kendo-angular-dropdowns/dist/es/list.component';
import * as import12 from '@progress/kendo-angular-dropdowns/dist/es/selection.service';
import * as import13 from '@progress/kendo-angular-dropdowns/dist/es/autocomplete.component';
import * as import14 from './searchbar.component.ngfactory';
import * as import15 from '@progress/kendo-angular-dropdowns/dist/es/searchbar.component';
import * as import16 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import17 from '@progress/kendo-angular-dropdowns/dist/es/navigation.service';
import * as import18 from '@angular/forms';
const styles_AutoCompleteComponent:any[] = ([] as any[]);
export const RenderType_AutoCompleteComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_AutoCompleteComponent,
  data: {}
}
);
function View_AutoCompleteComponent_1(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[
      [
        'class',
        'k-icon k-clear-value k-i-close'
      ]
      ,
      [
        'role',
        'button'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]
      ,
      [
        'title',
        'clear'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clearValue($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
function View_AutoCompleteComponent_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'k-icon k-i-loading'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_AutoCompleteComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_AutoCompleteComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_AutoCompleteComponent_5)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.headerTemplate.templateRef);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_AutoCompleteComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_AutoCompleteComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['NO DATA FOUND.'])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
function View_AutoCompleteComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',[[
        'class',
        'k-nodata'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),3,(null as any),View_AutoCompleteComponent_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoCompleteComponent_8)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.noDataTemplate;
    ck(v,3,0,currVal_0);
    const currVal_1:any = ck(v,5,0,((co.noDataTemplate == null)? (null as any): co.noDataTemplate.templateRef));
    ck(v,4,0,currVal_1);
    const currVal_2:boolean = !co.noDataTemplate;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_AutoCompleteComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_AutoCompleteComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_AutoCompleteComponent_10)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.footerTemplate.templateRef);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_AutoCompleteComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),24,'kendo-popup',([] as any[]),[
      [
        4,
        'width',
        (null as any)
      ]
      ,
      [
        4,
        'minWidth',
        (null as any)
      ]
      ,
      [
        4,
        'left',
        'px'
      ]
      ,
      [
        4,
        'top',
        'px'
      ]

    ]
    ,[
      [
        (null as any),
        'anchorViewportLeave'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('anchorViewportLeave' === en)) {
        const pd_0:any = ((<any>(co.popupOpen = false)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>$event.preventDefault()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import3.View_PopupComponent_0,import3.RenderType_PopupComponent)),
    import0.ɵprd(256,(null as any),import4.DOMService,import4.DOMService,([] as any[])),
    import0.ɵprd(256,(null as any),import5.AlignService,import5.AlignService,[import4.DOMService]),
    import0.ɵprd(256,(null as any),import6.PositionService,import6.PositionService,[import4.DOMService]),
    import0.ɵprd(256,(null as any),import7.ResizeService,import7.ResizeService,[
      import4.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵprd(256,(null as any),import8.ScrollableService,import8.ScrollableService,[
      import4.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(4317184,(null as any),0,import9.PopupComponent,[
      import5.AlignService,
      import0.ElementRef,
      import0.ChangeDetectorRef,
      import6.PositionService,
      import7.ResizeService,
      import8.ScrollableService,
      import0.Renderer
    ]
    ,{
      animate: [
        0,
        'animate'
      ]
      ,
      anchor: [
        1,
        'anchor'
      ]
      ,
      popupClass: [
        2,
        'popupClass'
      ]

    }
    ,{anchorViewportLeave: 'anchorViewportLeave'}),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_AutoCompleteComponent_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,2,'kendo-list',([] as any[]),(null as any),(null as any),(null as any),import10.View_ListComponent_0,import10.RenderType_ListComponent)),
    import0.ɵdid(2187264,(null as any),0,import11.ListComponent,[import12.SelectionService],{
      data: [
        0,
        'data'
      ]
      ,
      textField: [
        1,
        'textField'
      ]
      ,
      valueField: [
        2,
        'valueField'
      ]
      ,
      height: [
        3,
        'height'
      ]
      ,
      template: [
        4,
        'template'
      ]
      ,
      show: [
        5,
        'show'
      ]
      ,
      id: [
        6,
        'id'
      ]
      ,
      optionPrefix: [
        7,
        'optionPrefix'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_AutoCompleteComponent_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_AutoCompleteComponent_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = co.popupSettings.animate;
    const currVal_5:any = co.wrapper;
    const currVal_6:any = co.listContainerClasses;
    ck(v,6,0,currVal_4,currVal_5,currVal_6);
    const currVal_7:any = co.headerTemplate;
    ck(v,10,0,currVal_7);
    const currVal_8:any = co.data;
    const currVal_9:any = co.valueField;
    const currVal_10:any = co.valueField;
    const currVal_11:any = co.height;
    const currVal_12:any = co.template;
    const currVal_13:any = co.popupOpen;
    const currVal_14:any = co.listBoxId;
    const currVal_15:any = co.optionPrefix;
    ck(v,14,0,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15);
    const currVal_16:any = (co.data.length === 0);
    ck(v,19,0,currVal_16);
    const currVal_17:any = co.footerTemplate;
    ck(v,23,0,currVal_17);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.popupWidth;
    const currVal_1:any = co.popupMinWidth;
    const currVal_2:any = import0.ɵnov(v,6).offsetLeft;
    const currVal_3:any = import0.ɵnov(v,6).offsetTop;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export function View_AutoCompleteComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{searchbar: 0}),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-searchbar',([] as any[]),[[
        2,
        'k-searchbar',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'onNavigate'
      ]
      ,
      [
        (null as any),
        'valueChange'
      ]
      ,
      [
        (null as any),
        'onBlur'
      ]
      ,
      [
        (null as any),
        'onFocus'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import13.AutoCompleteComponent = v.component;
      if (('onNavigate' === en)) {
        const pd_0:any = ((<any>co.handleNavigate($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('valueChange' === en)) {
        const pd_1:any = ((<any>co.searchBarChange($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('onBlur' === en)) {
        const pd_2:any = ((<any>co.handleBlur()) !== false);
        ad = (pd_2 && ad);
      }
      if (('onFocus' === en)) {
        const pd_3:any = ((<any>co.handleFocus()) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },import14.View_SearchBarComponent_0,import14.RenderType_SearchBarComponent)),
    import0.ɵdid(286720,[
      [
        1,
        4
      ]
      ,
      [
        'searchbar',
        4
      ]

    ]
    ,0,import15.SearchBarComponent,[
      [
        2,
        import16.RTL
      ]
      ,
      import0.Renderer
    ]
    ,{
      disabled: [
        0,
        'disabled'
      ]
      ,
      placeholder: [
        1,
        'placeholder'
      ]
      ,
      userInput: [
        2,
        'userInput'
      ]
      ,
      suggestedText: [
        3,
        'suggestedText'
      ]

    }
    ,{
      valueChange: 'valueChange',
      onBlur: 'onBlur',
      onFocus: 'onFocus',
      onNavigate: 'onNavigate'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoCompleteComponent_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoCompleteComponent_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_AutoCompleteComponent_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import13.AutoCompleteComponent = v.component;
    const currVal_1:any = co.disabled;
    const currVal_2:any = co.placeholder;
    const currVal_3:any = co.value;
    const currVal_4:any = co.suggestion;
    ck(v,3,0,currVal_1,currVal_2,currVal_3,currVal_4);
    const currVal_5:boolean = (!co.loading && (co.clearButton && ((co.value == null)? (null as any): co.value.length)));
    ck(v,6,0,currVal_5);
    const currVal_6:any = co.loading;
    ck(v,9,0,currVal_6);
    const currVal_7:any = co.popupOpen;
    ck(v,12,0,currVal_7);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).searchBarClass;
    ck(v,2,0,currVal_0);
  });
}
function View_AutoCompleteComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'kendo-autocomplete',([] as any[]),[
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-autocomplete',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-disabled',
        (null as any)
      ]
      ,
      [
        4,
        'max-height',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-haspopup',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        1,
        'aria-activedescendant',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,(null as any),(null as any),View_AutoCompleteComponent_0,RenderType_AutoCompleteComponent)),
    import0.ɵprd(256,(null as any),import12.SelectionService,import12.SelectionService,([] as any[])),
    import0.ɵprd(256,(null as any),import17.NavigationService,import17.NavigationService,([] as any[])),
    import0.ɵdid(352256,(null as any),4,import13.AutoCompleteComponent,[
      [
        2,
        import16.RTL
      ]
      ,
      import12.SelectionService,
      import17.NavigationService,
      import0.ElementRef
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{template: 0}),
    import0.ɵqud(167772160,2,{headerTemplate: 0}),
    import0.ɵqud(167772160,3,{footerTemplate: 0}),
    import0.ɵqud(167772160,4,{noDataTemplate: 0}),
    import0.ɵprd(2560,(null as any),import18.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import13.AutoCompleteComponent])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_1:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_2:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_3:any = import0.ɵnov(v,3).isFocused;
    const currVal_4:any = import0.ɵnov(v,3).isDisabled;
    const currVal_5:any = import0.ɵnov(v,3).widgetHeight;
    const currVal_6:any = import0.ɵnov(v,3).role;
    const currVal_7:any = import0.ɵnov(v,3).widgetTabIndex;
    const currVal_8:any = import0.ɵnov(v,3).ariaDisabled;
    const currVal_9:any = import0.ɵnov(v,3).ariaHasPopup;
    const currVal_10:any = import0.ɵnov(v,3).ariaExpanded;
    const currVal_11:any = import0.ɵnov(v,3).ariaOwns;
    const currVal_12:any = import0.ɵnov(v,3).ariaActivedescendant;
    const currVal_13:any = import0.ɵnov(v,3).dir;
    ck(v,0,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10,
      currVal_11,
      currVal_12,
      currVal_13
    ]
    );
  });
}
export const AutoCompleteComponentNgFactory:import0.ComponentFactory<import13.AutoCompleteComponent> = import0.ɵccf('kendo-autocomplete',import13.AutoCompleteComponent,View_AutoCompleteComponent_Host_0,{
  data: 'data',
  value: 'value',
  valueField: 'valueField',
  placeholder: 'placeholder',
  popupSettings: 'popupSettings',
  loading: 'loading',
  clearButton: 'clearButton',
  suggest: 'suggest',
  disabled: 'disabled',
  tabIndex: 'tabIndex',
  filterable: 'filterable'
}
,{
  valueChange: 'valueChange',
  filterChange: 'filterChange',
  open: 'open',
  close: 'close'
}
,([] as any[]));
