// @progress/kendo-angular-dropdowns/dist/es/dropdownlist.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-dropdowns/dist/es/templates/template-context.directive';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
import * as import4 from '@progress/kendo-angular-dropdowns/dist/es/selectable.directive';
import * as import5 from '@progress/kendo-angular-dropdowns/dist/es/selection.service';
import * as import6 from '../../../kendo-angular-popup/dist/es/popup.component.ngfactory';
import * as import7 from '@progress/kendo-angular-popup/dist/es/services/dom.service';
import * as import8 from '@progress/kendo-angular-popup/dist/es/services/align.service';
import * as import9 from '@progress/kendo-angular-popup/dist/es/services/position.service';
import * as import10 from '@progress/kendo-angular-popup/dist/es/services/resize.service';
import * as import11 from '@progress/kendo-angular-popup/dist/es/services/scrollable.service';
import * as import12 from '@progress/kendo-angular-popup/dist/es/popup.component';
import * as import13 from './list.component.ngfactory';
import * as import14 from '@progress/kendo-angular-dropdowns/dist/es/list.component';
import * as import15 from '@progress/kendo-angular-dropdowns/dist/es/dropdownlist.component';
import * as import16 from '@progress/kendo-angular-dropdowns/dist/es/navigation.service';
import * as import17 from '@progress/kendo-angular-l10n/dist/es/rtl';
const styles_DropDownListComponent:any[] = ([] as any[]);
export const RenderType_DropDownListComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DropDownListComponent,
  data: {}
}
);
function View_DropDownListComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n               ']))],(null as any),(null as any));
}
function View_DropDownListComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DropDownListComponent_2)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.valueTemplate.templateRef,co.value);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_DropDownListComponent_3(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getText();
    ck(v,0,0,currVal_0);
  });
}
function View_DropDownListComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),14,'span',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>$event.stopImmediatePropagation()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['k-list-filter']),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'filterInput',
        1
      ]

    ]
      ,(null as any),5,'input',[[
        'class',
        'k-textbox'
      ]
    ],[
      [
        8,
        'dir',
        0
      ]
      ,
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'ngModelChange'
      ]
      ,
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'compositionstart'
      ]
      ,
      [
        (null as any),
        'compositionend'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,6)._handleInput($event.target.value)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,6).onTouched()) !== false);
        ad = (pd_1 && ad);
      }
      if (('compositionstart' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,6)._compositionStart()) !== false);
        ad = (pd_2 && ad);
      }
      if (('compositionend' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,6)._compositionEnd($event.target.value)) !== false);
        ad = (pd_3 && ad);
      }
      if (('ngModelChange' === en)) {
        const pd_4:any = ((<any>(co.filterText = $event)) !== false);
        ad = (pd_4 && ad);
      }
      if (('input' === en)) {
        const pd_5:any = ((<any>co.handleFilter($event)) !== false);
        ad = (pd_5 && ad);
      }
      if (('focus' === en)) {
        const pd_6:any = ((<any>co.onFilterFocus()) !== false);
        ad = (pd_6 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(8192,(null as any),0,import3.DefaultValueAccessor,[
      import0.Renderer,
      import0.ElementRef,
      [
        2,
        import3.COMPOSITION_BUFFER_MODE
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(512,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import3.DefaultValueAccessor]),
    import0.ɵdid(335872,(null as any),0,import3.NgModel,[
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        8,
        (null as any)
      ]
      ,
      [
        2,
        import3.NG_VALUE_ACCESSOR
      ]

    ]
      ,{model: [
        0,
        'model'
      ]
    },{update: 'ngModelChange'}),
    import0.ɵprd(1024,(null as any),import3.NgControl,(null as any),[import3.NgModel]),
    import0.ɵdid(8192,(null as any),0,import3.NgControlStatus,[import3.NgControl],(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',[[
        'unselectable',
        'on'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-icon',
      'k-i-search'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,3,0,true);
    ck(v,2,0,currVal_0);
    const currVal_9:any = co.filterText;
    ck(v,8,0,currVal_9);
    const currVal_10:any = ck(v,14,0,true,true);
    ck(v,13,0,currVal_10);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.dir;
    const currVal_2:any = import0.ɵnov(v,10).ngClassUntouched;
    const currVal_3:any = import0.ɵnov(v,10).ngClassTouched;
    const currVal_4:any = import0.ɵnov(v,10).ngClassPristine;
    const currVal_5:any = import0.ɵnov(v,10).ngClassDirty;
    const currVal_6:any = import0.ɵnov(v,10).ngClassValid;
    const currVal_7:any = import0.ɵnov(v,10).ngClassInvalid;
    const currVal_8:any = import0.ɵnov(v,10).ngClassPending;
    ck(v,5,0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
function View_DropDownListComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'kendoDropDownsSelectable',
        ''
      ]
    ],[
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-selected',
        (null as any)
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
      import0.ɵdid(8192,(null as any),0,import4.SelectableDirective,[import5.SelectionService],{index: [
        0,
        'index'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '\n                    ',
      '\n                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.setDefaultItemClasses();
    ck(v,2,0,currVal_3);
    const currVal_4:any = (0 - 1);
    ck(v,3,0,currVal_4);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵnov(v,3).focusedClassName;
    const currVal_1:any = import0.ɵnov(v,3).selectedClassName;
    const currVal_2:any = import0.ɵnov(v,3).ariaSelected;
    ck(v,1,0,currVal_0,currVal_1,currVal_2);
    const currVal_5:any = co.getDefaultItemText();
    ck(v,4,0,currVal_5);
  });
}
function View_DropDownListComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                    ']))],(null as any),(null as any));
}
function View_DropDownListComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'kendoDropDownsSelectable',
        ''
      ]
    ],[
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-selected',
        (null as any)
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
      import0.ɵdid(8192,(null as any),0,import4.SelectableDirective,[import5.SelectionService],{index: [
        0,
        'index'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DropDownListComponent_8)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_3:any = co.setDefaultItemClasses();
    ck(v,2,0,currVal_3);
    const currVal_4:any = (0 - 1);
    ck(v,3,0,currVal_4);
    const currVal_5:any = ck(v,7,0,co.itemTemplate.templateRef,co.defaultItem);
    ck(v,6,0,currVal_5);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).focusedClassName;
    const currVal_1:any = import0.ɵnov(v,3).selectedClassName;
    const currVal_2:any = import0.ɵnov(v,3).ariaSelected;
    ck(v,1,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_DropDownListComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_DropDownListComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DropDownListComponent_10)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.headerTemplate.templateRef);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_DropDownListComponent_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_DropDownListComponent_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['NO DATA FOUND.'])),
    (l()(),import0.ɵted((null as any),['\n                ']))
  ]
  ,(null as any),(null as any));
}
function View_DropDownListComponent_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',[[
        'class',
        'k-nodata'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),3,(null as any),View_DropDownListComponent_12)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DropDownListComponent_13)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.noDataTemplate;
    ck(v,3,0,currVal_0);
    const currVal_1:any = ck(v,5,0,(co.noDataTemplate? co.noDataTemplate.templateRef: (undefined as any)));
    ck(v,4,0,currVal_1);
    const currVal_2:boolean = !co.noDataTemplate;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_DropDownListComponent_15(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_DropDownListComponent_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DropDownListComponent_15)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod(['templateRef']),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.footerTemplate.templateRef);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_DropDownListComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),35,'kendo-popup',([] as any[]),[
      [
        4,
        'width',
        (null as any)
      ]
      ,
      [
        4,
        'minWidth',
        (null as any)
      ]
      ,
      [
        4,
        'left',
        'px'
      ]
      ,
      [
        4,
        'top',
        'px'
      ]

    ]
    ,[
      [
        (null as any),
        'anchorViewportLeave'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'open'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('anchorViewportLeave' === en)) {
        const pd_0:any = ((<any>(co.popupOpen = false)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.onMouseDown($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('open' === en)) {
        const pd_2:any = ((<any>co.popupOpened()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },import6.View_PopupComponent_0,import6.RenderType_PopupComponent)),
    import0.ɵprd(256,(null as any),import7.DOMService,import7.DOMService,([] as any[])),
    import0.ɵprd(256,(null as any),import8.AlignService,import8.AlignService,[import7.DOMService]),
    import0.ɵprd(256,(null as any),import9.PositionService,import9.PositionService,[import7.DOMService]),
    import0.ɵprd(256,(null as any),import10.ResizeService,import10.ResizeService,[
      import7.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵprd(256,(null as any),import11.ScrollableService,import11.ScrollableService,[
      import7.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(4317184,(null as any),0,import12.PopupComponent,[
      import8.AlignService,
      import0.ElementRef,
      import0.ChangeDetectorRef,
      import9.PositionService,
      import10.ResizeService,
      import11.ScrollableService,
      import0.Renderer
    ]
    ,{
      animate: [
        0,
        'animate'
      ]
      ,
      anchor: [
        1,
        'anchor'
      ]
      ,
      popupClass: [
        2,
        'popupClass'
      ]

    }
    ,{
      anchorViewportLeave: 'anchorViewportLeave',
      open: 'open'
    }
    ),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DropDownListComponent_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DropDownListComponent_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DropDownListComponent_7)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DropDownListComponent_9)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,2,'kendo-list',([] as any[]),(null as any),(null as any),(null as any),import13.View_ListComponent_0,import13.RenderType_ListComponent)),
    import0.ɵdid(2187264,(null as any),0,import14.ListComponent,[import5.SelectionService],{
      data: [
        0,
        'data'
      ]
      ,
      textField: [
        1,
        'textField'
      ]
      ,
      valueField: [
        2,
        'valueField'
      ]
      ,
      height: [
        3,
        'height'
      ]
      ,
      template: [
        4,
        'template'
      ]
      ,
      show: [
        5,
        'show'
      ]
      ,
      id: [
        6,
        'id'
      ]
      ,
      optionPrefix: [
        7,
        'optionPrefix'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DropDownListComponent_11)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DropDownListComponent_14)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_4:any = co.popupSettings.animate;
    const currVal_5:any = import0.ɵnov((<any>v.parent),2);
    const currVal_6:any = co.listContainerClasses;
    ck(v,6,0,currVal_4,currVal_5,currVal_6);
    const currVal_7:any = co.filterable;
    ck(v,10,0,currVal_7);
    const currVal_8:any = (co.defaultItem && !co.itemTemplate);
    ck(v,14,0,currVal_8);
    const currVal_9:any = (co.defaultItem && co.itemTemplate);
    ck(v,17,0,currVal_9);
    const currVal_10:any = co.headerTemplate;
    ck(v,21,0,currVal_10);
    const currVal_11:any = co.data;
    const currVal_12:any = co.textField;
    const currVal_13:any = co.valueField;
    const currVal_14:any = co.height;
    const currVal_15:any = co.itemTemplate;
    const currVal_16:any = co.popupOpen;
    const currVal_17:any = co.listBoxId;
    const currVal_18:any = co.optionPrefix;
    ck(v,25,0,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17,currVal_18);
    const currVal_19:any = (co.data.length === 0);
    ck(v,30,0,currVal_19);
    const currVal_20:any = co.footerTemplate;
    ck(v,34,0,currVal_20);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.popupWidth;
    const currVal_1:any = co.popupMinWidth;
    const currVal_2:any = import0.ɵnov(v,6).offsetLeft;
    const currVal_3:any = import0.ɵnov(v,6).offsetTop;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export function View_DropDownListComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{filterInput: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,[[
        'anchor',
        1
      ]
      ],(null as any),22,'span',[[
        'unselectable',
        'on'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-dropdown-wrap',
      'k-state-default',
      'k-state-disabled',
      'k-state-focused'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n           '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'span',[[
        'unselectable',
        'on'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['k-input']),
    (l()(),import0.ɵted((null as any),['\n               '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DropDownListComponent_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n               '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DropDownListComponent_3)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n           '])),
    (l()(),import0.ɵted((null as any),['\n           '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'span',[[
        'unselectable',
        'on'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['k-select']),
    (l()(),import0.ɵted((null as any),['\n               '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n           '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DropDownListComponent_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import15.DropDownListComponent = v.component;
    const currVal_0:any = ck(v,4,0,true,!co.disabled,co.disabled,co.isFocused);
    ck(v,3,0,currVal_0);
    const currVal_1:any = ck(v,8,0,true);
    ck(v,7,0,currVal_1);
    const currVal_2:any = co.valueTemplate;
    ck(v,11,0,currVal_2);
    const currVal_3:boolean = !co.valueTemplate;
    ck(v,14,0,currVal_3);
    const currVal_4:any = ck(v,19,0,true);
    ck(v,18,0,currVal_4);
    const currVal_5:any = co.buttonClasses;
    ck(v,22,0,currVal_5);
    const currVal_6:any = co.popupOpen;
    ck(v,27,0,currVal_6);
  },(null as any));
}
function View_DropDownListComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),9,'kendo-dropdownlist',([] as any[]),[
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-dropdown',
        (null as any)
      ]
      ,
      [
        2,
        'k-header',
        (null as any)
      ]
      ,
      [
        1,
        'role',
        0
      ]
      ,
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-haspopup',
        0
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        1,
        'aria-activedescendant',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'keypress'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('blur' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,3).blur()) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,3).focus()) !== false);
        ad = (pd_1 && ad);
      }
      if (('keydown' === en)) {
        const pd_2:any = ((<any>import0.ɵnov(v,3).keydown($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keypress' === en)) {
        const pd_3:any = ((<any>import0.ɵnov(v,3).keypress($event)) !== false);
        ad = (pd_3 && ad);
      }
      if (('click' === en)) {
        const pd_4:any = ((<any>import0.ɵnov(v,3).click()) !== false);
        ad = (pd_4 && ad);
      }
      return ad;
    },View_DropDownListComponent_0,RenderType_DropDownListComponent)),
    import0.ɵprd(256,(null as any),import5.SelectionService,import5.SelectionService,([] as any[])),
    import0.ɵprd(256,(null as any),import16.NavigationService,import16.NavigationService,([] as any[])),
    import0.ɵdid(352256,(null as any),5,import15.DropDownListComponent,[
      [
        2,
        import17.RTL
      ]
      ,
      import5.SelectionService,
      import16.NavigationService,
      import0.ElementRef,
      import0.Renderer,
      import0.NgZone
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(167772160,1,{itemTemplate: 0}),
    import0.ɵqud(167772160,2,{valueTemplate: 0}),
    import0.ɵqud(167772160,3,{headerTemplate: 0}),
    import0.ɵqud(167772160,4,{footerTemplate: 0}),
    import0.ɵqud(167772160,5,{noDataTemplate: 0}),
    import0.ɵprd(2560,(null as any),import3.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import15.DropDownListComponent])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_1:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_2:any = import0.ɵnov(v,3).widgetClasses;
    const currVal_3:any = import0.ɵnov(v,3).role;
    const currVal_4:any = import0.ɵnov(v,3).widgetTabIndex;
    const currVal_5:any = import0.ɵnov(v,3).ariaDisabled;
    const currVal_6:any = import0.ɵnov(v,3).ariaHasPopup;
    const currVal_7:any = import0.ɵnov(v,3).ariaExpanded;
    const currVal_8:any = import0.ɵnov(v,3).ariaOwns;
    const currVal_9:any = import0.ɵnov(v,3).ariaActivedescendant;
    const currVal_10:any = import0.ɵnov(v,3).dir;
    ck(v,0,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10
    ]
    );
  });
}
export const DropDownListComponentNgFactory:import0.ComponentFactory<import15.DropDownListComponent> = import0.ɵccf('kendo-dropdownlist',import15.DropDownListComponent,View_DropDownListComponent_Host_0,{
  iconClass: 'iconClass',
  loading: 'loading',
  data: 'data',
  value: 'value',
  textField: 'textField',
  valueField: 'valueField',
  popupSettings: 'popupSettings',
  defaultItem: 'defaultItem',
  disabled: 'disabled',
  filterable: 'filterable',
  ignoreCase: 'ignoreCase',
  delay: 'delay',
  valuePrimitive: 'valuePrimitive',
  tabIndex: 'tabIndex'
}
,{
  valueChange: 'valueChange',
  filterChange: 'filterChange',
  selectionChange: 'selectionChange',
  open: 'open',
  close: 'close'
}
,([] as any[]));
