// @progress/kendo-angular-dropdowns/dist/es/list.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-dropdowns/dist/es/templates/template-context.directive';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-dropdowns/dist/es/list-item.directive';
import * as import4 from '@progress/kendo-angular-dropdowns/dist/es/selectable.directive';
import * as import5 from '@progress/kendo-angular-dropdowns/dist/es/selection.service';
import * as import6 from '@progress/kendo-angular-dropdowns/dist/es/list.component';
const styles_ListComponent:any[] = ([] as any[]);
export const RenderType_ListComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ListComponent,
  data: {}
}
);
function View_ListComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_ListComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_ListComponent_3)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.template.templateRef,(<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_ListComponent_4(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
  ))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getText((<any>v.parent).context.$implicit);
    ck(v,0,0,currVal_0);
  });
}
function View_ListComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'li',[
      [
        'kendoDropDownsSelectable',
        ''
      ]
      ,
      [
        'role',
        'option'
      ]

    ]
    ,[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'tabIndex',
        0
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-selected',
        (null as any)
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,4).onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['k-item']),
      import0.ɵdid(8192,[[
        1,
        4
      ]
    ],0,import3.ListItemDirective,[import0.ElementRef],(null as any),(null as any)),
    import0.ɵdid(8192,(null as any),0,import4.SelectableDirective,[import5.SelectionService],{
      index: [
        0,
        'index'
      ]
      ,
      multipleSelection: [
        1,
        'multipleSelection'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_5:any = ck(v,2,0,true);
    ck(v,1,0,currVal_5);
    const currVal_6:any = v.context.index;
    const currVal_7:any = co.multipleSelection;
    ck(v,4,0,currVal_6,currVal_7);
    const currVal_8:any = co.template;
    ck(v,7,0,currVal_8);
    const currVal_9:boolean = !co.template;
    ck(v,10,0,currVal_9);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((co.optionPrefix + '-') + co.getValue(v.context.$implicit));
    const currVal_1:any = (0 - 1);
    const currVal_2:any = import0.ɵnov(v,4).focusedClassName;
    const currVal_3:any = import0.ɵnov(v,4).selectedClassName;
    const currVal_4:any = import0.ɵnov(v,4).ariaSelected;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export function View_ListComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(335544320,1,{items: 1}),
    import0.ɵqud(201326592,2,{content: 0}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'content',
        1
      ]

    ]
      ,(null as any),10,'div',[[
        'unselectable',
        'on'
      ]
      ],[[
        4,
        'maxHeight',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'ul',[[
        'role',
        'listbox'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-hidden',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-list',
      'k-reset'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import6.ListComponent = v.component;
    const currVal_1:any = co.setContainerClasses();
    ck(v,4,0,currVal_1);
    const currVal_4:any = ck(v,8,0,true,true);
    ck(v,7,0,currVal_4);
    const currVal_5:any = co.data;
    ck(v,11,0,currVal_5);
  },(ck,v) => {
    var co:import6.ListComponent = v.component;
    const currVal_0:any = co.getHeight();
    ck(v,3,0,currVal_0);
    const currVal_2:any = co.id;
    const currVal_3:boolean = !co.show;
    ck(v,6,0,currVal_2,currVal_3);
  });
}
function View_ListComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-list',([] as any[]),(null as any),(null as any),(null as any),View_ListComponent_0,RenderType_ListComponent)),
    import0.ɵdid(2187264,(null as any),0,import6.ListComponent,[import5.SelectionService],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ListComponentNgFactory:import0.ComponentFactory<import6.ListComponent> = import0.ɵccf('kendo-list',import6.ListComponent,View_ListComponent_Host_0,{
  data: 'data',
  selected: 'selected',
  focused: 'focused',
  textField: 'textField',
  valueField: 'valueField',
  height: 'height',
  template: 'template',
  show: 'show',
  id: 'id',
  optionPrefix: 'optionPrefix',
  multipleSelection: 'multipleSelection'
}
,{onClick: 'onClick'},([] as any[]));
