// @progress/kendo-angular-dropdowns/dist/es/searchbar.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-dropdowns/dist/es/searchbar.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/rtl';
const styles_SearchBarComponent:any[] = ([] as any[]);
export const RenderType_SearchBarComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SearchBarComponent,
  data: {}
}
);
export function View_SearchBarComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{input: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'input',
        1
      ]

    ]
    ,(null as any),0,'input',([] as any[]),[
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'placeholder',
        0
      ]
      ,
      [
        8,
        'className',
        0
      ]
      ,
      [
        1,
        'aria-activedescendant',
        0
      ]
      ,
      [
        1,
        'dir',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.SearchBarComponent = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.handleInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>co.handleFocus($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>co.handleBlur($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keydown' === en)) {
        const pd_3:any = ((<any>co.handleKeydown($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n   ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.SearchBarComponent = v.component;
    const currVal_0:any = co.disabled;
    const currVal_1:any = co.placeholder;
    const currVal_2:any = 'k-input';
    const currVal_3:any = co.activeId;
    const currVal_4:any = co.direction;
    ck(v,2,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_SearchBarComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-searchbar',([] as any[]),[[
        2,
        'k-searchbar',
        (null as any)
      ]
    ],(null as any),(null as any),View_SearchBarComponent_0,RenderType_SearchBarComponent)),
    import0.ɵdid(286720,(null as any),0,import1.SearchBarComponent,[
      [
        2,
        import2.RTL
      ]
      ,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).searchBarClass;
    ck(v,0,0,currVal_0);
  });
}
export const SearchBarComponentNgFactory:import0.ComponentFactory<import1.SearchBarComponent> = import0.ɵccf('kendo-searchbar',import1.SearchBarComponent,View_SearchBarComponent_Host_0,{
  disabled: 'disabled',
  placeholder: 'placeholder',
  activeId: 'activeId',
  userInput: 'userInput',
  suggestedText: 'suggestedText'
}
,{
  valueChange: 'valueChange',
  onBlur: 'onBlur',
  onFocus: 'onFocus',
  onClick: 'onClick',
  onNavigate: 'onNavigate'
}
,([] as any[]));
