// @progress/kendo-angular-popup/dist/es/popup.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-popup/dist/es/popup.component';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-popup/dist/es/services/dom.service';
import * as import4 from '@progress/kendo-angular-popup/dist/es/services/align.service';
import * as import5 from '@progress/kendo-angular-popup/dist/es/services/position.service';
import * as import6 from '@progress/kendo-angular-popup/dist/es/services/resize.service';
import * as import7 from '@progress/kendo-angular-popup/dist/es/services/scrollable.service';
const styles_PopupComponent:any[] = ([] as any[]);
export const RenderType_PopupComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PopupComponent,
      data: {'animation': [{
        name: 'toggle',
        definitions: [
          {
            type: 1,
            expr: 'void => down, up => down',
            animation: [
              {
                type: 6,
                styles: {transform: 'translateY(-100%)'}
              }
              ,
              {
                type: 4,
                styles: {
                  type: 6,
                  styles: {transform: 'translateY(0)'}
                }
                ,
                timings: '0.1s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'down => void',
            animation: [
              {
                type: 6,
                styles: {transform: 'translateY(0)'}
              }
              ,
              {
                type: 4,
                styles: {
                  type: 6,
                  styles: {transform: 'translateY(-100%)'}
                }
                ,
                timings: '0.1s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'void => up, down => up',
            animation: [
              {
                type: 6,
                styles: {transform: 'translateY(100%)'}
              }
              ,
              {
                type: 4,
                styles: {
                  type: 6,
                  styles: {transform: 'translateY(0)'}
                }
                ,
                timings: '0.1s ease-in'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: 'up => void',
            animation: [
              {
                type: 6,
                styles: {transform: 'translateY(0)'}
              }
              ,
              {
                type: 4,
                styles: {
                  type: 6,
                  styles: {transform: 'translateY(100%)'}
                }
                ,
                timings: '0.1s ease-in'
              }

            ]

          }

        ]

      }
  ]}
}
);
export function View_PopupComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'k-popup'
      ]
      ],[[
        24,
        '@toggle',
        0
      ]
      ],[[
        (null as any),
        '@toggle.done'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PopupComponent = v.component;
      if (('@toggle.done' === en)) {
        const pd_0:any = ((<any>co.triggerOpen()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n     ']))
  ]
  ,(ck,v) => {
    var co:import1.PopupComponent = v.component;
    const currVal_1:any = 'k-popup';
    const currVal_2:any = co.popupClass;
    ck(v,2,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:import1.PopupComponent = v.component;
    const currVal_0:any = co.direction;
    ck(v,1,0,currVal_0);
  });
}
function View_PopupComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'kendo-popup',([] as any[]),[
      [
        4,
        'left',
        'px'
      ]
      ,
      [
        4,
        'top',
        'px'
      ]

    ]
    ,(null as any),(null as any),View_PopupComponent_0,RenderType_PopupComponent)),
    import0.ɵprd(256,(null as any),import3.DOMService,import3.DOMService,([] as any[])),
    import0.ɵprd(256,(null as any),import4.AlignService,import4.AlignService,[import3.DOMService]),
    import0.ɵprd(256,(null as any),import5.PositionService,import5.PositionService,[import3.DOMService]),
    import0.ɵprd(256,(null as any),import6.ResizeService,import6.ResizeService,[
      import3.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵprd(256,(null as any),import7.ScrollableService,import7.ScrollableService,[
      import3.DOMService,
      import0.NgZone
    ]
    ),
    import0.ɵdid(4317184,(null as any),0,import1.PopupComponent,[
      import4.AlignService,
      import0.ElementRef,
      import0.ChangeDetectorRef,
      import5.PositionService,
      import6.ResizeService,
      import7.ScrollableService,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,6,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,6).offsetLeft;
    const currVal_1:any = import0.ɵnov(v,6).offsetTop;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const PopupComponentNgFactory:import0.ComponentFactory<import1.PopupComponent> = import0.ɵccf('kendo-popup',import1.PopupComponent,View_PopupComponent_Host_0,{
  animate: 'animate',
  anchor: 'anchor',
  anchorAlign: 'anchorAlign',
  collision: 'collision',
  popupAlign: 'popupAlign',
  popupClass: 'popupClass',
  offset: 'offset'
}
,{
  anchorViewportLeave: 'anchorViewportLeave',
  close: 'close',
  open: 'open'
}
,['*']);
