// @progress/kendo-angular-buttons/dist/es/listbutton/list.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-buttons/dist/es/listbutton/template-context.directive';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-buttons/dist/es/focusable/focusable.directive';
import * as import4 from '@progress/kendo-angular-buttons/dist/es/focusable/focus.service';
import * as import5 from '@progress/kendo-angular-buttons/dist/es/listbutton/list.component';
const styles_ListComponent:any[] = ([] as any[]);
export const RenderType_ListComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ListComponent,
  data: {}
}
);
function View_ListComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                ']))],(null as any),(null as any));
}
function View_ListComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_ListComponent_3)),
      import0.ɵdid(8192,(null as any),0,import1.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      '$implicit'
    ]
    ),
    (l()(),import0.ɵand(0,(null as any),(null as any),0))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((co.itemTemplate == null)? (null as any): co.itemTemplate.templateRef),(<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_ListComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getIconClasses((<any>(<any>v.parent).parent).context.$implicit);
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_ListComponent_6(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'img',[
      [
        'alt',
        ''
      ]
      ,
      [
        'class',
        'k-image'
      ]

    ]
      ,[[
        8,
        'src',
        4
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.imageUrl;
    ck(v,0,0,currVal_0);
  });
}
function View_ListComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_5)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_6)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '\n                    ',
      '\n                '
    ]
    ))
  ]
  ,(ck,v) => {
    const currVal_0:any = ((<any>v.parent).context.$implicit.icon || (<any>v.parent).context.$implicit.iconClass);
    ck(v,2,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.imageUrl;
    ck(v,5,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.getText((<any>v.parent).context.$implicit);
    ck(v,6,0,currVal_2);
  });
}
function View_ListComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'li',[
      [
        'kendoButtonFocusable',
        ''
      ]
      ,
      [
        'role',
        'menuItem'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]
      ,
      [
        'unselectable',
        'on'
      ]

    ]
    ,[
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick(v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-item',
      'k-state-disabled'
    ]
    ),
    import0.ɵdid(73728,(null as any),0,import3.FocusableDirective,[
      import4.FocusService,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{index: [
        0,
        'index'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_2)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_4)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = ck(v,2,0,true,v.context.$implicit.disabled);
    ck(v,1,0,currVal_2);
    const currVal_3:any = v.context.index;
    ck(v,3,0,currVal_3);
    const currVal_4:any = ((co.itemTemplate == null)? (null as any): co.itemTemplate.templateRef);
    ck(v,6,0,currVal_4);
    const currVal_5:boolean = !((co.itemTemplate == null)? (null as any): co.itemTemplate.templateRef);
    ck(v,9,0,currVal_5);
  },(ck,v) => {
    const currVal_0:any = (v.context.$implicit.disabled? true: false);
    const currVal_1:any = import0.ɵnov(v,3).focusedClassName;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export function View_ListComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',[
      [
        'class',
        'k-list k-reset'
      ]
      ,
      [
        'unselectable',
        'on'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_ListComponent_1)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:import5.ListComponent = v.component;
    const currVal_0:any = co.data;
    ck(v,4,0,currVal_0);
  },(null as any));
}
function View_ListComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-button-list',([] as any[]),(null as any),(null as any),(null as any),View_ListComponent_0,RenderType_ListComponent)),
    import0.ɵdid(24576,(null as any),0,import5.ListComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ListComponentNgFactory:import0.ComponentFactory<import5.ListComponent> = import0.ɵccf('kendo-button-list',import5.ListComponent,View_ListComponent_Host_0,{
  data: 'data',
  textField: 'textField',
  itemTemplate: 'itemTemplate'
}
,{
  onItemClick: 'onItemClick',
  onItemBlur: 'onItemBlur'
}
,([] as any[]));
