// @progress/kendo-angular-grid/dist/es/pager/pager-info.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/pager/pager-info.component';
import * as import2 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import3 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
const styles_PagerInfoComponent:any[] = ([] as any[]);
export const RenderType_PagerInfoComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PagerInfoComponent,
  data: {}
}
);
export function View_PagerInfoComponent_0(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(2,[(l()(),import0.ɵted((null as any),[
      '',
      ' - ',
      ' ',
      ' ',
      ' ',
      ''
    ]
  ))],(null as any),(ck,v) => {
    var co:import1.PagerInfoComponent = v.component;
    const currVal_0:any = co.currentPageText;
    const currVal_1:any = co.maxItems;
    const currVal_2:any = co.textFor('pagerOf');
    const currVal_3:any = co.total;
    const currVal_4:any = co.textFor('pagerItems');
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_PagerInfoComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-info',([] as any[]),[
      [
        2,
        'k-pager-info',
        (null as any)
      ]
      ,
      [
        2,
        'k-label',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_PagerInfoComponent_0,RenderType_PagerInfoComponent)),
    import0.ɵdid(122880,(null as any),0,import1.PagerInfoComponent,[
      import2.LocalizationService,
      import0.ChangeDetectorRef,
      import3.PagerContextService
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).classes;
    const currVal_1:any = import0.ɵnov(v,1).classes;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const PagerInfoComponentNgFactory:import0.ComponentFactory<import1.PagerInfoComponent> = import0.ɵccf('kendo-pager-info',import1.PagerInfoComponent,View_PagerInfoComponent_Host_0,{},{},([] as any[]));
