// @progress/kendo-angular-grid/dist/es/pager/pager.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './pager-prev-buttons.component.ngfactory';
import * as import2 from '@progress/kendo-angular-grid/dist/es/pager/pager-prev-buttons.component';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
import * as import5 from './pager-numeric-buttons.component.ngfactory';
import * as import6 from '@progress/kendo-angular-grid/dist/es/pager/pager-numeric-buttons.component';
import * as import7 from './pager-input.component.ngfactory';
import * as import8 from '@progress/kendo-angular-grid/dist/es/pager/pager-input.component';
import * as import9 from './pager-next-buttons.component.ngfactory';
import * as import10 from '@progress/kendo-angular-grid/dist/es/pager/pager-next-buttons.component';
import * as import11 from './pager-info.component.ngfactory';
import * as import12 from '@progress/kendo-angular-grid/dist/es/pager/pager-info.component';
import * as import13 from './pager-page-sizes.component.ngfactory';
import * as import14 from '@progress/kendo-angular-grid/dist/es/pager/pager-page-sizes.component';
import * as import15 from '@angular/common';
import * as import16 from '@progress/kendo-angular-grid/dist/es/template-context.directive';
import * as import17 from '@progress/kendo-angular-grid/dist/es/pager/pager.component';
const styles_PagerComponent:any[] = ([] as any[]);
export const RenderType_PagerComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PagerComponent,
  data: {}
}
);
function View_PagerComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n        ']))],(null as any),(null as any));
}
function View_PagerComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-prev-buttons',([] as any[]),(null as any),(null as any),(null as any),import1.View_PagerPrevButtonsComponent_0,import1.RenderType_PagerPrevButtonsComponent)),
    import0.ɵdid(122880,(null as any),0,import2.PagerPrevButtonsComponent,[
      import3.LocalizationService,
      import4.PagerContextService,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
function View_PagerComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'kendo-pager-numeric-buttons',([] as any[]),(null as any),(null as any),(null as any),import5.View_PagerNumericButtonsComponent_0,import5.RenderType_PagerNumericButtonsComponent)),
    import0.ɵdid(122880,(null as any),0,import6.PagerNumericButtonsComponent,[
      import3.LocalizationService,
      import0.ChangeDetectorRef,
      import4.PagerContextService
    ]
      ,{buttonCount: [
        0,
        'buttonCount'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.settings.buttonCount;
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_PagerComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-input',([] as any[]),(null as any),(null as any),(null as any),import7.View_PagerInputComponent_0,import7.RenderType_PagerInputComponent)),
    import0.ɵdid(122880,(null as any),0,import8.PagerInputComponent,[
      import3.LocalizationService,
      import4.PagerContextService,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
function View_PagerComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-next-buttons',([] as any[]),(null as any),(null as any),(null as any),import9.View_PagerNextButtonsComponent_0,import9.RenderType_PagerNextButtonsComponent)),
    import0.ɵdid(122880,(null as any),0,import10.PagerNextButtonsComponent,[
      import3.LocalizationService,
      import4.PagerContextService,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
function View_PagerComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-info',([] as any[]),[
      [
        2,
        'k-pager-info',
        (null as any)
      ]
      ,
      [
        2,
        'k-label',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import11.View_PagerInfoComponent_0,import11.RenderType_PagerInfoComponent)),
    import0.ɵdid(122880,(null as any),0,import12.PagerInfoComponent,[
      import3.LocalizationService,
      import0.ChangeDetectorRef,
      import4.PagerContextService
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).classes;
    const currVal_1:any = import0.ɵnov(v,1).classes;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_PagerComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-page-sizes',([] as any[]),[
      [
        2,
        'k-pager-sizes',
        (null as any)
      ]
      ,
      [
        2,
        'k-label',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import13.View_PagerPageSizesComponent_0,import13.RenderType_PagerPageSizesComponent)),
    import0.ɵdid(122880,(null as any),0,import14.PagerPageSizesComponent,[
      import3.LocalizationService,
      import0.ChangeDetectorRef,
      import4.PagerContextService
    ]
      ,{pageSizes: [
        0,
        'pageSizes'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.settings.pageSizes;
    ck(v,1,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).classes;
    const currVal_1:any = import0.ɵnov(v,1).classes;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_PagerComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),19,(null as any),(null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_3)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_4)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_5)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_6)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_7)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_8)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.settings.previousNext;
    ck(v,3,0,currVal_0);
    const currVal_1:any = (co.settings.type === 'numeric');
    ck(v,6,0,currVal_1);
    const currVal_2:any = (co.settings.type === 'input');
    ck(v,9,0,currVal_2);
    const currVal_3:any = co.settings.previousNext;
    ck(v,12,0,currVal_3);
    const currVal_4:any = co.settings.info;
    ck(v,15,0,currVal_4);
    const currVal_5:any = co.settings.pageSizes;
    ck(v,18,0,currVal_5);
  },(null as any));
}
export function View_PagerComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),3,(null as any),View_PagerComponent_1)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      import0.ɵdid(73728,(null as any),0,import16.TemplateContextDirective,[import0.ViewContainerRef],{templateContext: [
        0,
        'templateContext'
      ]
    },(null as any)),
    import0.ɵpod([
      'templateRef',
      'totalPages',
      'total',
      'skip',
      'pageSize',
      'currentPage'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerComponent_2)),
    import0.ɵdid(8192,(null as any),0,import15.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import17.PagerComponent = v.component;
    const currVal_0:any = ((co.template == null)? (null as any): co.template.templateRef);
    ck(v,2,0,currVal_0);
    const currVal_1:any = ck(v,4,0,((co.template == null)? (null as any): co.template.templateRef),co.totalPages,co.total,co.skip,co.pageSize,co.currentPage);
    ck(v,3,0,currVal_1);
    const currVal_2:boolean = !((co.template == null)? (null as any): co.template.templateRef);
    ck(v,7,0,currVal_2);
  },(null as any));
}
function View_PagerComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager',([] as any[]),[
      [
        2,
        'k-pager-wrap',
        (null as any)
      ]
      ,
      [
        2,
        'k-grid-pager',
        (null as any)
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_PagerComponent_0,RenderType_PagerComponent)),
    import0.ɵdid(385024,(null as any),0,import17.PagerComponent,[import4.PagerContextService],(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).pagerWrapClass;
    const currVal_1:any = import0.ɵnov(v,1).gridPagerClass;
    const currVal_2:any = import0.ɵnov(v,1).widgetClass;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const PagerComponentNgFactory:import0.ComponentFactory<import17.PagerComponent> = import0.ɵccf('kendo-pager',import17.PagerComponent,View_PagerComponent_Host_0,{
  total: 'total',
  skip: 'skip',
  pageSize: 'pageSize',
  options: 'options',
  template: 'template'
}
,{pageChange: 'pageChange'},([] as any[]));
