// @progress/kendo-angular-grid/dist/es/pager/pager-input.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-grid/dist/es/pager/pager-input.component';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
const styles_PagerInputComponent:any[] = ([] as any[]);
export const RenderType_PagerInputComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PagerInputComponent,
  data: {}
}
);
export function View_PagerInputComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n     '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-pager-input',
      'k-label'
    ]
    ),
    (l()(),import0.ɵted((null as any),[
      '\n        ',
      '\n        '
    ]
    )),
      (l()(),import0.ɵeld(0,[[
        'inputPager',
        1
      ]
    ],(null as any),0,'input',([] as any[]),[
      [
        2,
        'k-textbox',
        (null as any)
      ]
      ,
      [
        8,
        'value',
        0
      ]

    ]
      ,[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.PagerInputComponent = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.onInputChange(import0.ɵnov(v,5).value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n        ',
      ' ',
      '\n     '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,3,0,true,true);
    ck(v,2,0,currVal_0);
  },(ck,v) => {
    var co:import2.PagerInputComponent = v.component;
    const currVal_1:any = co.textFor('pagerPage');
    ck(v,4,0,currVal_1);
    const currVal_2:any = true;
    const currVal_3:any = co.currentPage;
    ck(v,5,0,currVal_2,currVal_3);
    const currVal_4:any = co.textFor('pagerOf');
    const currVal_5:any = co.totalPages;
    ck(v,6,0,currVal_4,currVal_5);
  });
}
function View_PagerInputComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-input',([] as any[]),(null as any),(null as any),(null as any),View_PagerInputComponent_0,RenderType_PagerInputComponent)),
    import0.ɵdid(122880,(null as any),0,import2.PagerInputComponent,[
      import3.LocalizationService,
      import4.PagerContextService,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const PagerInputComponentNgFactory:import0.ComponentFactory<import2.PagerInputComponent> = import0.ɵccf('kendo-pager-input',import2.PagerInputComponent,View_PagerInputComponent_Host_0,{},{},([] as any[]));
