// @progress/kendo-angular-grid/dist/es/pager/pager-numeric-buttons.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@progress/kendo-angular-grid/dist/es/pager/pager-numeric-buttons.component';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
const styles_PagerNumericButtonsComponent:any[] = ([] as any[]);
export const RenderType_PagerNumericButtonsComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PagerNumericButtonsComponent,
  data: {}
}
);
function View_PagerNumericButtonsComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'class',
        'k-link'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePage((co.start - 2))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['...']))
  ]
  ,(null as any),(null as any));
}
function View_PagerNumericButtonsComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'a',[[
        'href',
        '#'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePage((v.context.$implicit - 1))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-link',
      'k-state-selected'
    ]
    ),
    (l()(),import0.ɵted((null as any),[
      '\n                    ',
      '\n                '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,4,0,true,(co.currentPage == v.context.$implicit));
    ck(v,3,0,currVal_0);
  },(ck,v) => {
    const currVal_1:any = v.context.$implicit;
    ck(v,5,0,currVal_1);
  });
}
function View_PagerNumericButtonsComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'class',
        'k-link'
      ]
      ,
      [
        'href',
        '#'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.changePage(co.end)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['...']))
  ]
  ,(null as any),(null as any));
}
export function View_PagerNumericButtonsComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n       '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),12,'ul',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-pager-numbers',
      'k-reset'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerNumericButtonsComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerNumericButtonsComponent_2)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerNumericButtonsComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import2.PagerNumericButtonsComponent = v.component;
    const currVal_0:any = ck(v,3,0,true,true);
    ck(v,2,0,currVal_0);
    const currVal_1:any = (co.start > 1);
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.buttons;
    ck(v,9,0,currVal_2);
    const currVal_3:any = (co.end < co.totalPages);
    ck(v,12,0,currVal_3);
  },(null as any));
}
function View_PagerNumericButtonsComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-numeric-buttons',([] as any[]),(null as any),(null as any),(null as any),View_PagerNumericButtonsComponent_0,RenderType_PagerNumericButtonsComponent)),
    import0.ɵdid(122880,(null as any),0,import2.PagerNumericButtonsComponent,[
      import3.LocalizationService,
      import0.ChangeDetectorRef,
      import4.PagerContextService
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const PagerNumericButtonsComponentNgFactory:import0.ComponentFactory<import2.PagerNumericButtonsComponent> = import0.ɵccf('kendo-pager-numeric-buttons',import2.PagerNumericButtonsComponent,View_PagerNumericButtonsComponent_Host_0,{buttonCount: 'buttonCount'},{},([] as any[]));
