// @progress/kendo-angular-grid/dist/es/pager/pager-prev-buttons.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/pager/pager-prev-buttons.component';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
const styles_PagerPrevButtonsComponent:any[] = ([] as any[]);
export const RenderType_PagerPrevButtonsComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PagerPrevButtonsComponent,
  data: {}
}
);
export function View_PagerPrevButtonsComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'a',[[
        'href',
        '#'
      ]
      ],[[
        8,
        'title',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PagerPrevButtonsComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>((co.currentPage !== 1)? co.changePage(0): false)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-link',
      'k-pager-nav',
      'k-state-disabled',
      'k-pager-first'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',([] as any[]),[[
        1,
        'aria-label',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-icon',
      'k-i-seek-w'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'a',[[
        'href',
        '#'
      ]
      ],[[
        8,
        'title',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PagerPrevButtonsComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>((co.currentPage !== 1)? co.changePage((co.currentPage - 2)): false)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-link',
      'k-pager-nav',
      'k-state-disabled',
      ''
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'span',([] as any[]),[[
        1,
        'aria-label',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import2.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'k-icon',
      'k-i-arrow-w'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import1.PagerPrevButtonsComponent = v.component;
    const currVal_1:any = ck(v,3,0,true,true,co.disabled,true);
    ck(v,2,0,currVal_1);
    const currVal_3:any = ck(v,7,0,true,true);
    ck(v,6,0,currVal_3);
    const currVal_5:any = ck(v,13,0,true,true,co.disabled,true);
    ck(v,12,0,currVal_5);
    const currVal_7:any = ck(v,17,0,true,true);
    ck(v,16,0,currVal_7);
  },(ck,v) => {
    var co:import1.PagerPrevButtonsComponent = v.component;
    const currVal_0:any = co.textFor('pagerFirstPage');
    ck(v,1,0,currVal_0);
    const currVal_2:any = co.textFor('pagerFirstPage');
    ck(v,5,0,currVal_2);
    const currVal_4:any = co.textFor('pagerPreviousPage');
    ck(v,11,0,currVal_4);
    const currVal_6:any = co.textFor('pagerPreviousPage');
    ck(v,15,0,currVal_6);
  });
}
function View_PagerPrevButtonsComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-prev-buttons',([] as any[]),(null as any),(null as any),(null as any),View_PagerPrevButtonsComponent_0,RenderType_PagerPrevButtonsComponent)),
    import0.ɵdid(122880,(null as any),0,import1.PagerPrevButtonsComponent,[
      import3.LocalizationService,
      import4.PagerContextService,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const PagerPrevButtonsComponentNgFactory:import0.ComponentFactory<import1.PagerPrevButtonsComponent> = import0.ɵccf('kendo-pager-prev-buttons',import1.PagerPrevButtonsComponent,View_PagerPrevButtonsComponent_Host_0,{},{},([] as any[]));
