// @progress/kendo-angular-grid/dist/es/pager/pager-page-sizes.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-grid/dist/es/pager/pager-page-sizes.component';
import * as import2 from '@angular/common';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/localization.service';
import * as import4 from '@progress/kendo-angular-grid/dist/es/pager/pager-context.service';
const styles_PagerPageSizesComponent:any[] = ([] as any[]);
export const RenderType_PagerPageSizesComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PagerPageSizesComponent,
  data: {}
}
);
function View_PagerPageSizesComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',([] as any[]),[[
        8,
        'value',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.pageSize;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.pageSize;
    ck(v,1,0,currVal_1);
  });
}
function View_PagerPageSizesComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'option',([] as any[]),[
      [
        8,
        'value',
        0
      ]
      ,
      [
        8,
        'selected',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                ',
      '\n            '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = v.context.$implicit;
    const currVal_1:any = ((v.context.$implicit === co.pageSize)? true: (undefined as any));
    ck(v,0,0,currVal_0,currVal_1);
    const currVal_2:any = v.context.$implicit;
    ck(v,1,0,currVal_2);
  });
}
export function View_PagerPageSizesComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,[[
        'select',
        1
      ]
      ],(null as any),7,'select',([] as any[]),(null as any),[[
        (null as any),
        'change'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PagerPageSizesComponent = v.component;
      if (('change' === en)) {
        const pd_0:any = ((<any>co.pageSizeChange(import0.ɵnov(v,1).value)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerPageSizesComponent_1)),
    import0.ɵdid(8192,(null as any),0,import2.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_PagerPageSizesComponent_2)),
    import0.ɵdid(401408,(null as any),0,import2.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),[
      '\n        ',
      '\n    '
    ]
    ))
  ]
  ,(ck,v) => {
    var co:import1.PagerPageSizesComponent = v.component;
    const currVal_0:any = co.showInitialPageSize;
    ck(v,4,0,currVal_0);
    const currVal_1:any = co.pageSizes;
    ck(v,7,0,currVal_1);
  },(ck,v) => {
    var co:import1.PagerPageSizesComponent = v.component;
    const currVal_2:any = co.textFor('pagerItemsPerPage');
    ck(v,9,0,currVal_2);
  });
}
function View_PagerPageSizesComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-pager-page-sizes',([] as any[]),[
      [
        2,
        'k-pager-sizes',
        (null as any)
      ]
      ,
      [
        2,
        'k-label',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_PagerPageSizesComponent_0,RenderType_PagerPageSizesComponent)),
    import0.ɵdid(122880,(null as any),0,import1.PagerPageSizesComponent,[
      import3.LocalizationService,
      import0.ChangeDetectorRef,
      import4.PagerContextService
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).classes;
    const currVal_1:any = import0.ɵnov(v,1).classes;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const PagerPageSizesComponentNgFactory:import0.ComponentFactory<import1.PagerPageSizesComponent> = import0.ɵccf('kendo-pager-page-sizes',import1.PagerPageSizesComponent,View_PagerPageSizesComponent_Host_0,{pageSizes: 'pageSizes'},{},([] as any[]));
