// @progress/kendo-angular-resize-sensor/dist/es/resize-sensor.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-resize-sensor/dist/es/resize-sensor.component';
const styles_ResizeSensorComponent:any[] = ['[_nghost-%COMP%] { position: absolute; display: block; left: 0; top: 0; right: 0; bottom: 0; z-index: -1;overflow: hidden; visibility: hidden; }'];
export const RenderType_ResizeSensorComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_ResizeSensorComponent,
  data: {}
}
);
export function View_ResizeSensorComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{expand: 0}),
    import0.ɵqud(201326592,2,{expandChild: 0}),
    import0.ɵqud(201326592,3,{shrink: 0}),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'expand',
        1
      ]

    ]
      ,(null as any),2,'div',[[
        'style',
        'position: absolute; display: block; left: 0; top: 0; right: 0; bottom: 0; z-index: -1;overflow: hidden; visibility: hidden;'
      ]
      ],(null as any),[[
        (null as any),
        'scroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ResizeSensorComponent = v.component;
      if (('scroll' === en)) {
        const pd_0:any = ((<any>co.scroll()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['  '])),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'expandChild',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'style',
        'position: absolute; left: 0; top: 0; transition: 0s;'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'shrink',
        1
      ]

    ]
      ,(null as any),2,'div',[[
        'style',
        'position: absolute; display: block; left: 0; top: 0; right: 0; bottom: 0; z-index: -1;overflow: hidden; visibility: hidden;'
      ]
      ],(null as any),[[
        (null as any),
        'scroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.ResizeSensorComponent = v.component;
      if (('scroll' === en)) {
        const pd_0:any = ((<any>co.scroll()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['  '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'style',
        'position: absolute; left: 0; top: 0; transition: 0s;width: 200%; height: 200%;'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_ResizeSensorComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'kendo-resize-sensor',([] as any[]),(null as any),(null as any),(null as any),View_ResizeSensorComponent_0,RenderType_ResizeSensorComponent)),
    import0.ɵdid(4284416,(null as any),0,import1.ResizeSensorComponent,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ResizeSensorComponentNgFactory:import0.ComponentFactory<import1.ResizeSensorComponent> = import0.ɵccf('kendo-resize-sensor',import1.ResizeSensorComponent,View_ResizeSensorComponent_Host_0,{rateLimit: 'rateLimit'},{resize: 'resize'},([] as any[]));
