// @progress/kendo-angular-inputs/dist/es/maskedtextbox/maskedtextbox.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@progress/kendo-angular-inputs/dist/es/maskedtextbox/maskedtextbox.component';
import * as import2 from '@progress/kendo-angular-inputs/dist/es/maskedtextbox/masking.service';
import * as import3 from '@progress/kendo-angular-l10n/dist/es/rtl';
import * as import4 from '@angular/forms';
const styles_MaskedTextBoxComponent:any[] = ([] as any[]);
export const RenderType_MaskedTextBoxComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_MaskedTextBoxComponent,
  data: {}
}
);
export function View_MaskedTextBoxComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{input: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'input',
        1
      ]

    ]
    ,(null as any),0,'input',[
      [
        'autocapitalize',
        'off'
      ]
      ,
      [
        'autocomplete',
        'off'
      ]
      ,
      [
        'autocorrect',
        'off'
      ]
      ,
      [
        'class',
        'k-textbox'
      ]
      ,
      [
        'spellcheck',
        'false'
      ]
      ,
      [
        'type',
        'text'
      ]

    ]
    ,[
      [
        8,
        'tabIndex',
        0
      ]
      ,
      [
        1,
        'title',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'dragstart'
      ]
      ,
      [
        (null as any),
        'drop'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.MaskedTextBoxComponent = v.component;
      if (('focus' === en)) {
        const pd_0:any = ((<any>co.handleFocus($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.handleBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('dragstart' === en)) {
        const pd_2:any = ((<any>co.handleDragDrop()) !== false);
        ad = (pd_2 && ad);
      }
      if (('drop' === en)) {
        const pd_3:any = ((<any>co.handleDragDrop()) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.MaskedTextBoxComponent = v.component;
    const currVal_0:any = co.tabIndex;
    const currVal_1:any = co.title;
    const currVal_2:any = co.disabled;
    ck(v,2,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_MaskedTextBoxComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'kendo-maskedtextbox',([] as any[]),[
      [
        1,
        'dir',
        0
      ]
      ,
      [
        2,
        'k-widget',
        (null as any)
      ]
      ,
      [
        2,
        'k-maskedtextbox',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-focused',
        (null as any)
      ]
      ,
      [
        2,
        'k-state-disabled',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'paste'
      ]
      ,
      [
        (null as any),
        'input'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('paste' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,2).pasteHandler($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('input' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,2).inputHandler($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_MaskedTextBoxComponent_0,RenderType_MaskedTextBoxComponent)),
    import0.ɵprd(256,(null as any),import2.MaskingService,import2.MaskingService,([] as any[])),
    import0.ɵdid(286720,(null as any),0,import1.MaskedTextBoxComponent,[
      import2.MaskingService,
      import0.Renderer,
      [
        2,
        import3.RTL
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(2560,(null as any),import4.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import1.MaskedTextBoxComponent]),
    import0.ɵprd(2560,(null as any),import4.NG_VALIDATORS,(p0_0:any) => {
      return [p0_0];
    },[import1.MaskedTextBoxComponent])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).direction;
    const currVal_1:any = import0.ɵnov(v,2).hostClasses;
    const currVal_2:any = import0.ɵnov(v,2).hostClasses;
    const currVal_3:any = import0.ɵnov(v,2).hostFocusedClass;
    const currVal_4:any = import0.ɵnov(v,2).hostDisabledClass;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export const MaskedTextBoxComponentNgFactory:import0.ComponentFactory<import1.MaskedTextBoxComponent> = import0.ɵccf('kendo-maskedtextbox',import1.MaskedTextBoxComponent,View_MaskedTextBoxComponent_Host_0,{
  disabled: 'disabled',
  title: 'title',
  mask: 'mask',
  value: 'value',
  rules: 'rules',
  prompt: 'prompt',
  promptPlaceholder: 'promptPlaceholder',
  includeLiterals: 'includeLiterals',
  maskValidation: 'maskValidation',
  tabIndex: 'tabIndex'
}
,{
  onFocus: 'focus',
  onBlur: 'blur',
  valueChange: 'valueChange'
}
,([] as any[]));
